<!DOCTYPE html>
<html><head>
<meta http-equiv="content-type" content="text/html; charset=UTF-8"><meta name="viewport" content="width=device-width"><title>https://biomedicalhub.github.io/genomics/singlepage.html</title><link rel="stylesheet" type="text/css" href="resource://content-accessible/viewsource.css"></head><body id="viewsource" class="highlight" style="tab-size: 4"><pre id="line1"><span></span><span class="doctype">&lt;!DOCTYPE html&gt;</span><span>
<span id="line2"></span></span><span>&lt;<span class="start-tag">html</span> <span class="attribute-name">lang</span>="<a class="attribute-value">en-us</a>"&gt;</span><span>
<span id="line3"></span>
<span id="line4"></span>  </span><span>&lt;<span class="start-tag">head</span>&gt;</span><span>
<span id="line5"></span>  </span><span>&lt;<span class="start-tag">meta</span> <span class="attribute-name">http-equiv</span>="<a class="attribute-value">X-UA-Compatible</a>" <span class="attribute-name">content</span>="<a class="attribute-value">IE=edge</a>"&gt;</span><span>
<span id="line6"></span>  </span><span>&lt;<span class="start-tag">meta</span> <span class="attribute-name">http-equiv</span>="<a class="attribute-value">content-type</a>" <span class="attribute-name">content</span>="<a class="attribute-value">text/html; charset=utf-8</a>"&gt;</span><span>
<span id="line7"></span>
<span id="line8"></span>  </span><span class="comment">&lt;!-- Enable responsiveness on mobile devices--&gt;</span><span>
<span id="line9"></span>  </span><span>&lt;<span class="start-tag">meta</span> <span class="attribute-name">name</span>="<a class="attribute-value">viewport</a>" <span class="attribute-name">content</span>="<a class="attribute-value">width=device-width, initial-scale=1.0, maximum-scale=1</a>"&gt;</span><span>
<span id="line10"></span>
<span id="line11"></span>  </span><span>&lt;<span class="start-tag">title</span>&gt;</span><span>
<span id="line12"></span>    
<span id="line13"></span>       <span class="entity"><span>&amp;</span>middot;</span> Introduction to Genomics
<span id="line14"></span>    
<span id="line15"></span>  </span><span>&lt;/<span class="end-tag">title</span>&gt;</span><span>
<span id="line16"></span>
<span id="line17"></span>  </span><span class="comment">&lt;!-- CSS --&gt;</span><span>
<span id="line18"></span>  </span><span>&lt;<span class="start-tag">link</span> <span class="attribute-name">rel</span>="<a class="attribute-value">stylesheet</a>" <span class="attribute-name">href</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/public/css/poole.css">/genomics/public/css/poole.css</a>"&gt;</span><span>
<span id="line19"></span>  </span><span>&lt;<span class="start-tag">link</span> <span class="attribute-name">rel</span>="<a class="attribute-value">stylesheet</a>" <span class="attribute-name">href</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/public/css/syntax.css">/genomics/public/css/syntax.css</a>"&gt;</span><span>
<span id="line20"></span>  </span><span>&lt;<span class="start-tag">link</span> <span class="attribute-name">rel</span>="<a class="attribute-value">stylesheet</a>" <span class="attribute-name">href</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/public/css/lanyon.css">/genomics/public/css/lanyon.css</a>"&gt;</span><span>
<span id="line21"></span>  </span><span>&lt;<span class="start-tag">link</span> <span class="attribute-name">rel</span>="<a class="attribute-value">stylesheet</a>" <span class="attribute-name">href</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/public/css/custom.css">/genomics/public/css/custom.css</a>"&gt;</span><span>
<span id="line22"></span>  </span><span>&lt;<span class="start-tag">link</span> <span class="attribute-name">rel</span>="<a class="attribute-value">stylesheet</a>" <span class="attribute-name">href</span>="<a class="attribute-value" href="view-source:https://fonts.googleapis.com/css?family=PT+Serif:400,400italic,700%7CPT+Sans:400">https://fonts.googleapis.com/css?family=PT+Serif:400,400italic,700%7CPT+Sans:400</a>"&gt;</span><span>
<span id="line23"></span>  </span><span>&lt;<span class="start-tag">link</span> <span class="attribute-name">rel</span>="<a class="attribute-value">stylesheet</a>" <span class="attribute-name">href</span>="<a class="attribute-value" href="view-source:https://maxcdn.bootstrapcdn.com/font-awesome/4.1.0/css/font-awesome.min.css">//maxcdn.bootstrapcdn.com/font-awesome/4.1.0/css/font-awesome.min.css</a>"&gt;</span><span>
<span id="line24"></span>
<span id="line25"></span>  </span><span class="comment">&lt;!-- Icons --&gt;</span><span>
<span id="line26"></span></span><span class="comment">&lt;!--   &lt;link rel="apple-touch-icon-precomposed" sizes="144x144" href="/genomicspublic/apple-touch-icon-precomposed.png"&gt; --&gt;</span><span>
<span id="line27"></span>  </span><span class="comment">&lt;!--link rel="shortcut icon" href="/genomics/public/favicon.ico"--&gt;</span><span>
<span id="line28"></span>
<span id="line29"></span>  </span><span class="comment">&lt;!-- RSS --&gt;</span><span>
<span id="line30"></span>  </span><span class="comment">&lt;!--link rel="alternate" type="application/rss+xml" title="RSS" href="/atom.xml"--&gt;</span><span>
<span id="line31"></span></span><span>&lt;/<span class="end-tag">head</span>&gt;</span><span>
<span id="line32"></span>
<span id="line33"></span>  </span><span>&lt;<span class="start-tag">body</span>&gt;</span><span>
<span id="line34"></span>
<span id="line35"></span>    </span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">masthead</a>"&gt;</span><span>
<span id="line36"></span>      </span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">container</a>"&gt;</span><span>
<span id="line37"></span>        </span><span>&lt;<span class="start-tag">h3</span> <span class="attribute-name">class</span>="<a class="attribute-value">masthead-title</a>"&gt;</span><span>
<span id="line38"></span>          </span><span>&lt;<span class="start-tag">a</span> <span class="attribute-name">href</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/">/genomics/</a>" <span class="attribute-name">title</span>="<a class="attribute-value">Home</a>"&gt;</span><span>Introduction to Genomics</span><span>&lt;/<span class="end-tag">a</span>&gt;</span><span>
<span id="line39"></span>          </span><span>&lt;<span class="start-tag">small</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">small</span>&gt;</span><span>
<span id="line40"></span>        </span><span>&lt;/<span class="end-tag">h3</span>&gt;</span><span>
<span id="line41"></span>      </span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span>
<span id="line42"></span>    </span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span>
<span id="line43"></span>
<span id="line44"></span>    </span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">container content</a>"&gt;</span><span>
<span id="line45"></span>      
<span id="line46"></span>
<span id="line47"></span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">singlepage</a>"&gt;</span><span>
<span id="line48"></span>  
<span id="line49"></span>    </span><span>&lt;<span class="start-tag">h1</span> <span class="attribute-name">class</span>="<a class="attribute-value">page-title</a>"&gt;</span><span>Welcome to Introduction to Genomics Course</span><span>&lt;/<span class="end-tag">h1</span>&gt;</span><span>
<span id="line50"></span>    </span><span>&lt;<span class="start-tag">h2</span> <span class="attribute-name">id</span>="<a class="attribute-value">introduction</a>"&gt;</span><span>Introduction</span><span>&lt;/<span class="end-tag">h2</span>&gt;</span><span>
<span id="line51"></span>
<span id="line52"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Welcome to the Genomics workshop. Generating reams of data in Biology is easy these days. In little more than a fortnight we can generate more data than the entire human genome project
<span id="line53"></span>generated in over a decade of work. Making biological sense out of that data, understanding its
<span id="line54"></span>limitations and how the analysis algorithms work is now the major challenge for researchers. The
<span id="line55"></span>aim of this workshop is to take you through a few example projects and tasks. On the way you will learn how to evaluate the quality of data as provided by a sequencing facility, how to align the data against a
<span id="line56"></span>known and annotated reference genome and how to perform a de-novo assembly. In addition you will also learn how to compare results between different samples.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line57"></span>
<span id="line58"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>This workshop is broken into 6 parts. You should feel free to take as long as you like on each part. It is much more important that you have a thorough understanding of each part, rather than try to race through the entire workshop.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line59"></span>
<span id="line60"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>The five parts are:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line61"></span>
<span id="line62"></span></span><span>&lt;<span class="start-tag">ul</span>&gt;</span><span>
<span id="line63"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>Introduction</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line64"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>Remapping a strain of </span><span>&lt;<span class="start-tag">em</span>&gt;</span><span>E.coli</span><span>&lt;/<span class="end-tag">em</span>&gt;</span><span> to a reference sequence</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line65"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>Assembly of unmapped reads</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line66"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>Complete </span><span>&lt;<span class="start-tag">em</span>&gt;</span><span>de-novo</span><span>&lt;/<span class="end-tag">em</span>&gt;</span><span> assembly of all reads</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line67"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>Repeating on strains of </span><span>&lt;<span class="start-tag">em</span>&gt;</span><span>V.parahaemolyticus</span><span>&lt;/<span class="end-tag">em</span>&gt;</span><span> and comparing them</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line68"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>Lon Read de-novo assembly.</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line69"></span></span><span>&lt;/<span class="end-tag">ul</span>&gt;</span><span>
<span id="line70"></span>
<span id="line71"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>For this workshop we will assume little background knowledge, except a basic familiarity with the Linux operating system and the Openstack cloud. We will cover the basics of how genomic DNA libraries are generated and sequenced, and the principles behind short read paired-end sequencing. We will look at why data can vary in quality, why adaptor sequences need to be filtered out and how to quality control data.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line72"></span>
<span id="line73"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>In the second part we will take the plunge and
<span id="line74"></span>align the filtered reads to a reference genome, call variants and compare them against the published genome to identify missing, truncated or altered genes. This will involve the use of a publicly available set of bacterial </span><span>&lt;<span class="start-tag">em</span>&gt;</span><span>E.coli</span><span>&lt;/<span class="end-tag">em</span>&gt;</span><span> Illumina reads and reference genome.
<span id="line75"></span>In parts 3 and 4 we will look at how one can identify novel sequences which are not present in the reference genome. In part 5, you will be asked to repeat the steps 2, 3 and 4 on other data sets and to compare the results.
<span id="line76"></span>In part 6 we will look at an assembly process using only long reads.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line77"></span>
<span id="line78"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>A word on notation. If you see something like this:</span><span>&lt;<span class="start-tag">br</span> <span>/</span>&gt;</span><span>
<span id="line79"></span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>cd ~/genomics_tutorial/reference/sequence</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;<span class="start-tag">br</span> <span>/</span>&gt;</span><span>
<span id="line80"></span>It means, type the highlighted text into your terminal.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line81"></span>
<span id="line82"></span>  
<span id="line83"></span>    </span><span>&lt;<span class="start-tag">h1</span> <span class="attribute-name">class</span>="<a class="attribute-value">page-title</a>"&gt;</span><span>Part 1. Short Read Genomics: Introduction</span><span>&lt;/<span class="end-tag">h1</span>&gt;</span><span>
<span id="line84"></span>    </span><span>&lt;<span class="start-tag">h2</span> <span class="attribute-name">id</span>="<a class="attribute-value">principles-of-illumina-based-sequencing</a>"&gt;</span><span>Principles of Illumina-based sequencing:</span><span>&lt;/<span class="end-tag">h2</span>&gt;</span><span>
<span id="line85"></span>
<span id="line86"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>There are several second generation sequencers currently on the market. These include the Ion Torrent, and many Illumina system. Other (now obsolescent) platforms included Life Tech SoLID and Roche 454. All of these systems rely on making hundreds of thousands of clonal copies of a fragment of DNA and sequencing the ensemble of fragments using DNA polymerase or in the case of the SOLiD via ligation. This is simply because the detectors (basically souped-up digital cameras), cannot detect fluorescence  changes from a single molecule.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line87"></span>
<span id="line88"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>The ‘third-generation’ Pacific Biosciences SMRT (Single Molecule Real Time) Sequel sequencers are able to detect fluorescence from a single molecule of DNA. However, the machines are very large and produce less than a tenth of the data of an Illumina MiSeq run and for long reads <span class="entity"><span>&amp;</span>gt;</span> 0kb error rates are generally around 10-12%.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line89"></span>
<span id="line90"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>The Oxford Nanopore MinION is another ‘third-generation’ single-molecule system which
<span id="line91"></span>measures changes in electrical current through a Nanopore as a single molecule is ratcheted through
<span id="line92"></span>it. Although error rates are higher (5-10%) and per-base costs are higher the technology has improved
<span id="line93"></span>rapidly and will probably replace second generation systems over the next few years. Currently
<span id="line94"></span>however, the second generation sequencers are dominating the sequencing world.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line95"></span>
<span id="line96"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>We will mainly look at the Illumina sequencing pipeline here, but the basic principles apply to other second-generation sequencers. If you would like further details on other platforms then I recommend reading:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line97"></span>
<span id="line98"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">em</span>&gt;</span><span>Mardis ER. Next-generation DNA sequencing methods. Annual Reviews Genomics Hum Genet 2008; 9 :387-402.</span><span>&lt;/<span class="end-tag">em</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line99"></span>
<span id="line100"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">em</span>&gt;</span><span>Goodwin S.  Coming of age: ten years of next-generation sequencing technologies. Nat Rev Genet. 2016; 17:333-51.</span><span>&lt;/<span class="end-tag">em</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line101"></span>
<span id="line102"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>A typical sequencing run would begin with the user supplying 1ng - 1ug of genomic DNA to a facility along with quality control information in the form of an automated electrophoresis output (e.g. Agilent Bioanalyser/Tapestation trace) or gel image and quantification information.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line103"></span>
<span id="line104"></span></span><span>&lt;<span class="start-tag">h2</span> <span class="attribute-name">id</span>="<a class="attribute-value">dna-library-preparation</a>"&gt;</span><span>DNA Library preparation</span><span>&lt;/<span class="end-tag">h2</span>&gt;</span><span>
<span id="line105"></span>
<span id="line106"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>For most sequencing applications, paired-end libraries are generated. Genomic DNA is sheared into 300-800 bp fragments (usually via sonication) and size-selected accordingly. Ends are repaired and an overhanging adenine base is added, after which oligonucleotide adaptors are ligated. In many cases the adaptors contain unique DNA sequences of 6-12bp which can be used to identify the sample if they are ‘multiplexed’ together for sequencing. This type of sequencing is used extensively when sequencing small genomes such as those of bacteria because it lowers the overall per-genome cost.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line107"></span>
<span id="line108"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part1_library_prep_workflow.jpg">/genomics/images/part1_library_prep_workflow.jpg</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line109"></span>
<span id="line110"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>A) Steps a through e explain the main steps in Illumina sample preparation.
<span id="line111"></span>B) Overview of the automated the size selection protocol.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line112"></span>
<span id="line113"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>*Borgstrom E, Lundin S, Lundeberg J, 2011 Large Scale Library Generation for High Throughput Sequencing. PLoS
<span id="line114"></span>ONE 6:e19119. doi:10.1371/journal.pone.0019119 *</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line115"></span>
<span id="line116"></span></span><span>&lt;<span class="start-tag">hr</span> <span>/</span>&gt;</span><span>
<span id="line117"></span>
<span id="line118"></span></span><span>&lt;<span class="start-tag">h3</span> <span class="attribute-name">id</span>="<a class="attribute-value">sequencing</a>"&gt;</span><span>Sequencing</span><span>&lt;/<span class="end-tag">h3</span>&gt;</span><span>
<span id="line119"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>adapted from: </span><span>&lt;<span class="start-tag">em</span>&gt;</span><span>(Next-Generation DNA Sequencing Methods Mardis, E.R. Annu. Rev. Genomics Hum. Genet. 2008. 9:387-402)</span><span>&lt;/<span class="end-tag">em</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line120"></span>
<span id="line121"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Once sufficient libraries have been prepared, the task is to amplify single strands of DNA to form
<span id="line122"></span>monoclonal clusters. The single molecule amplification step for the Illumina HiSeq 2500 starts with an
<span id="line123"></span>Illumina-specific adapter library and takes place on the oligo-derivatized surface of a flow cell, and is
<span id="line124"></span>performed by an automated device called a cBot Cluster Station. The flow cell is either a 2 or 8 channel sealed glass microfabricated device that allows bridge amplification of fragments on its surface, and uses DNA polymerase to produce multiple DNA copies, or clusters, that each represent the single molecule that initiated the cluster amplification.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line125"></span>
<span id="line126"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Separate or multiple libraries can be added to each of the eight channels, or the same library can be used in all eight, or combinations thereof. Each cluster contains approximately one million copies of the original fragment, which is sufficient for reporting incorporated bases at the required signal intensity for detection during sequencing.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line127"></span>
<span id="line128"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>The Illumina system utilizes a sequencing- by-synthesis approach in which all four nucleotides are added simultaneously to the flow cell channels, along with DNA polymerase, for incorporation into the oligo-primed cluster fragments (see figure below for details). Specifically, the nucleotides carry a base-unique fluorescent label and the 3 -OH group is chemically blocked such that each incorporation is a unique event.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line129"></span>
<span id="line130"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>An imaging step follows each base incorporation step, during which each flow cell lane is imaged in three 100-tile segments by the instrument optics at a cluster density of 600,000-800,000 per mm2. After each imaging step, the 3’ blocking group is chemically removed to prepare each strand for the next incorporation by DNA polymerase. This series of steps continues for a specific number of cycles, as determined by user-defined instrument settings, which permits discrete read lengths of 50-300 bases. A base-calling algorithm assigns sequences and associated quality values to each read and a quality checking pipeline evaluates the Illumina data from each run.</span><span>&lt;<span class="start-tag">br</span> <span>/</span>&gt;</span><span>
<span id="line131"></span>The figure below summarises the process:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line132"></span>
<span id="line133"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part1_sequencing_01.jpg">/genomics/images/part1_sequencing_01.jpg</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span>
<span id="line134"></span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part1_sequencing_02.jpg">/genomics/images/part1_sequencing_02.jpg</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line135"></span>
<span id="line136"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>The Illumina sequencing-by-synthesis approach:</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span> Cluster strands created by bridge amplification are primed and all four fluorescently labelled, 3-OH blocked nucleotides are added to the flow cell with DNA polymerase. The cluster strands are extended by one nucleotide. Following the incorporation step, the unused nucleotides and DNA polymerase molecules are washed away, a scan buffer is added to the flow cell, and the optics system scans each lane of the flow cell by imaging units called tiles. Once imaging is completed, chemicals that effect cleavage of the fluorescent labels and the 3 -OH blocking groups are added to the flow cell, which prepares the cluster strands for another round of fluorescent nucleotide incorporation.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line137"></span>
<span id="line138"></span></span><span>&lt;<span class="start-tag">hr</span> <span>/</span>&gt;</span><span>
<span id="line139"></span>
<span id="line140"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Base-calling</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span>  involves evaluating the raw intensity values for each fluorophore and comparing them to determine which base is actually present at a given position during a cycle. To call bases on the Illumina HiSeq or SOLiD platform, the positions of clusters need to be identified during the first few cycles. This is because they are formed in random positions on the flowcell as the annealing process is stochastic. This is in contrast to the 454 system and later Illumina models where the position of each cluster is defined by a manufactured pattern in which the reaction takes place.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line141"></span>
<span id="line142"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>If there are too many clusters, the edges of the clusters will begin to merge and the image analysis algorithms will not be able to distinguish one cluster from another (remember, the software is dealing with upwards of half a million clusters per square millimeter - that’s a lot of dots!).</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line143"></span>
<span id="line144"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part1_sequencing_03.jpg">/genomics/images/part1_sequencing_03.jpg</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line145"></span>
<span id="line146"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>The above figure illustrates the principles of base-calling from cycles 1 to 9. If we focus on the highlighted cluster, one can observe that the colour (wavelength) of light observed at each cycle changes along with the brightness (intensity). This is due to the incorporation of complementary ddNTPs containing fluorophores. So at cycle 1 we have a T base, at 2 a G base and so on. If the colour or intensity is ambiguous the sequencer will mark it as an N. Other clusters are also visible in the images; these will represent different monoclonal clusters with different sequences.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line147"></span>
<span id="line148"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>The base calling algorithms turn the raw intensity values into T,G,C,A or N base calls. There are a variety of methods to do this and the one mentioned here is by no means the only one available, but it is often used as the default method on the Illumina systems, and will only call a base if the intensity divided by the sum of the highest and second highest intensity is less than a given threshold (usually 0.6). Otherwise the base is marked with an N. In addition the standard Illumina pipeline will reject an entire read if two or more of these failures occur in the first 4 bases of a read (it uses these cycles to determine the boundary of a cluster).</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line149"></span>
<span id="line150"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">em</span>&gt;</span><span>These processes are carried out at the sequencing facility and you will not need to perform any of these tasks under normal circumstances. They are explained here as useful background information.</span><span>&lt;/<span class="end-tag">em</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line151"></span>
<span id="line152"></span></span><span>&lt;<span class="start-tag">hr</span> <span>/</span>&gt;</span><span>
<span id="line153"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Paired-end sequencing</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span> is a remarkably simple and powerful modification to the standard sequencing protocol. It is nearly always worth obtaining paired-end reads if performing genomic sequencing. Typically sequencers of any type are only able to sequence a portion of DNA (typically 125bp in the case of Illumina) before the fidelity of the enzyme and de-phasing of clusters (see later) increase the error rate beyond tolerable levels. As a result, on the Illumina system, a fragment which is 500bp long will only have the first 125bp sequenced.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line154"></span>
<span id="line155"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>If the size selection is tight enough and you know that nearly all the fragments are close to 500bp long, you can repeat the sequencing reaction from the other end of the fragment. This will yield two reads for each DNA fragment separated by a known distance. In the figure below the dashed regions represent the complete DNA fragment and the solid lines the regions we are able to sequence:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line156"></span>
<span id="line157"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>The added information gained by knowing the distance between the two reads can be invaluable for spanning repetitive regions. In the figure below, the light coloured regions indicate repetitive sections of DNA. If a read contains only repetitive DNA, an alignment algorithm will be able to map the read to many locations in a reference genome. However, with paired-end reads, there is a greater chance that</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line158"></span>
<span id="line159"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part1_paired_end_reads.jpg">/genomics/images/part1_paired_end_reads.jpg</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line160"></span>
<span id="line161"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>at least one of the two reads will map to a unique region of DNA. In this way one of the reads can be used to anchor the other read in the pair and help resolve the repetitive region. Paired-end reads are often used when performing de-novo genome sequencing (i.e. when a reference is not available to align against) because they enable contiguous regions of DNA to be ordered, or when characterizing variants such as large insertions or deletions.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line162"></span>
<span id="line163"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Other forms of paired-end sequencing with much larger distances (e.g. 10kb) are possible with so called ‘mate-pair’ libraries. These are usually used in specific projects to help order contigs in de-novo sequencing projects. We will not cover them here, but the principles behind them are similar.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line164"></span>
<span id="line165"></span></span><span>&lt;<span class="start-tag">hr</span> <span>/</span>&gt;</span><span>
<span id="line166"></span>
<span id="line167"></span></span><span>&lt;<span class="start-tag">h2</span> <span class="attribute-name">id</span>="<a class="attribute-value">inherent-sources-of-error</a>"&gt;</span><span>Inherent sources of error</span><span>&lt;/<span class="end-tag">h2</span>&gt;</span><span>
<span id="line168"></span>
<span id="line169"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>No measurement is without a certain degree of error. This is true in sequencing. As such there is a finite probability that a base will not be called correctly. There are several possible sources:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line170"></span>
<span id="line171"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Frequency cross-talk and normalisation errors:</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line172"></span>
<span id="line173"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>When reading an A base, a small amount of C will also be measured due to frequency overlap and vice-versa. Similarly with G and T bases. Additionally, from the figure below, it should be clear that the extent to which the dyes fluoresce differs. As such it is necessary to normalize the intensities. This normalisation process can also introduce errors.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line174"></span>
<span id="line175"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Frequency response curve for A and C dyes
<span id="line176"></span> </span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part1_crosstalk.jpg">/genomics/images/part1_crosstalk.jpg</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line177"></span>
<span id="line178"></span></span><span>&lt;<span class="start-tag">hr</span> <span>/</span>&gt;</span><span>
<span id="line179"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">em</span>&gt;</span><span>(Intensity y-axis and frequency on the x-axis)</span><span>&lt;/<span class="end-tag">em</span>&gt;</span><span>
<span id="line180"></span> **</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line181"></span>
<span id="line182"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Phasing/Pre-phasing:</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line183"></span>
<span id="line184"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>his occurs when a strand of DNA lags or leads the other DNA strands within a cluster. This introduces additional background noise into the signal and reduces the intensity of the true base. In the example below we have a cluster with 7 strands of DNA (very small, but this is just an example). Five strands are on a C-base, whilst 1 is lagging behind (called phasing) on a G base and the remaining strand is running ahead of the pack (confusingly called pre-phasing) on an A base. As such the C signal will be reduced and A and G boosted for the rest of the sequencing run. Too much phasing or pre-phasing (i.e. <span class="entity"><span>&amp;</span>gt;</span> 15-20%) usually causes problems for the base calling algorithm and result in clusters being filtered out.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line185"></span>
<span id="line186"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part1_phasing.jpg">/genomics/images/part1_phasing.jpg</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line187"></span>
<span id="line188"></span></span><span>&lt;<span class="start-tag">hr</span> <span>/</span>&gt;</span><span>
<span id="line189"></span>
<span id="line190"></span></span><span>&lt;<span class="start-tag">h2</span> <span class="attribute-name">id</span>="<a class="attribute-value">other-issues</a>"&gt;</span><span>Other issues:</span><span>&lt;/<span class="end-tag">h2</span>&gt;</span><span>
<span id="line191"></span>
<span id="line192"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Biases introduced by sample preparation</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span> - your sequencing is only as good as your  experimental design and DNA extraction. Remember that your sample will be put through several cycles of PCR before sequencing which is also introduces a potential source of bias.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line193"></span>
<span id="line194"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>High AT or GC content sequences</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span> - reduces the complexity of the sequence and can result in higher error rates.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line195"></span>
<span id="line196"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Homopolymeric sequences</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span> - long stretches of a single base can make it difficult to determine  phasing and pre-phasing rates. This can introduce errors in determining the precise length of a hompolymeric stretch of sequence. This much more of a problem on the 454 and Ion Torrent than Illumina platforms but still worth bearing in mind. Especially if you encounter indels which have been called in homopolymeric tracts.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line197"></span>
<span id="line198"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Certain motifs</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span> can cause loops and other steric clashes </span><span>&lt;<span class="start-tag">em</span>&gt;</span><span>See Nakamura et al, Sequence-specific error profile of Illumina sequencers Nuc. Acid Res. first published online May 16, 2011 doi:10.1093/nar/gkr344</span><span>&lt;/<span class="end-tag">em</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line199"></span>
<span id="line200"></span></span><span>&lt;<span class="start-tag">h2</span> <span class="attribute-name">id</span>="<a class="attribute-value">understanding-quality-scores</a>"&gt;</span><span>Understanding Quality scores</span><span>&lt;/<span class="end-tag">h2</span>&gt;</span><span>
<span id="line201"></span>
<span id="line202"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>To account for the possible errors and provide an estimate of confidence in a given base-call, the Illumina sequencing pipeline assigns a quality score to each base called. Most quality scores are calculated using the Phred scale. Each base call has an associated base call quality which estimates the chance that the base call is incorrect.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line203"></span>
<span id="line204"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Q10 = 1 in 10 chance of incorrect base call</span><span>&lt;<span class="start-tag">br</span> <span>/</span>&gt;</span><span>
<span id="line205"></span>Q20 = 1 in 100 chance of incorrect base call</span><span>&lt;<span class="start-tag">br</span> <span>/</span>&gt;</span><span>
<span id="line206"></span>Q30 = 1 in 1000 chance of incorrect base call</span><span>&lt;<span class="start-tag">br</span> <span>/</span>&gt;</span><span>
<span id="line207"></span>Q40 = 1 in 10,000 chance of incorrect base call</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line208"></span>
<span id="line209"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>For most 454, SOLiD and Illumina runs you should see quality scores between Q20 and Q40. Note that these as only estimates of base-quality based on calibration runs performed by the manufacturer against a sample of known sequence with (typically) a GC content of 50%. Extreme GC bias and/or particular motifs or homopolymers can cause the quality scores to become unreliable.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line210"></span>
<span id="line211"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Accurate base qualities are an essential part in ensuring variant calls are correct. As a rough and ready rule we generally assume that with Illumina data anything less than Q20 is not useful data and should be excluded from the analysis.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line212"></span>
<span id="line213"></span></span><span>&lt;<span class="start-tag">h2</span> <span class="attribute-name">id</span>="<a class="attribute-value">reads-containing-adaptors</a>"&gt;</span><span>Reads containing adaptors</span><span>&lt;/<span class="end-tag">h2</span>&gt;</span><span>
<span id="line214"></span>
<span id="line215"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Some reads will contain adaptor sequences after sequencing, usually at the end of the read. This is usually because of short sample DNA fragments, which result in the polymerase reading into the adaptor region. Occasionally this can also happen because of mis-priming. It is important to remove or trim sequences containing these reads as the adaptor sequences can prevent reads mapping to a reference sequence and will adversely affect de-novo assembly</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line216"></span>
<span id="line217"></span>  
<span id="line218"></span>    </span><span>&lt;<span class="start-tag">h1</span> <span class="attribute-name">class</span>="<a class="attribute-value">page-title</a>"&gt;</span><span>Part 2. Short Read Genomics: Remapping</span><span>&lt;/<span class="end-tag">h1</span>&gt;</span><span>
<span id="line219"></span>    </span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>In this section of the workshop we will be analysing a strain of </span><span>&lt;<span class="start-tag">em</span>&gt;</span><span>E.coli</span><span>&lt;/<span class="end-tag">em</span>&gt;</span><span> which was sequenced at Exeter.  It is closely related to the K-12 substrain MG1655 </span><span>&lt;<span class="start-tag">a</span> <span class="attribute-name">href</span>="<a class="attribute-value" href="view-source:http://www.ncbi.nlm.nih.gov/nuccore/U00096">http://www.ncbi.nlm.nih.gov/nuccore/U00096</a>"&gt;</span><span>(http://www.ncbi.nlm.nih.gov/nuccore/U00096).</span><span>&lt;/<span class="end-tag">a</span>&gt;</span><span>  We want to obtain a list of single nucleotide polymorphisms (SNPs), insertions/deletions (Indels) and  any genes which have been deleted.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line220"></span>
<span id="line221"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Quality control</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line222"></span>
<span id="line223"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>In this section of the workshop we will be learning about evaluating the quality of an Illumina MiSeq sequencing run. Sequencing data is usually delivered in FASTQ format and the process described here can be used with any FASTQ formatted file from any platform (e.g 454, Illumina, Ion Torrent, PacBio etc).</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line224"></span>
<span id="line225"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>2nd (and 3rd) generation sequencers produce vast quantities
<span id="line226"></span>of data. A single Illumina MiSeq lane will produce over 10-Gbases of data. However, the error rates of these platforms are 10-100x higher than Sanger sequencing. They also have very different error profiles. Unlike Sanger sequencing, where the most reliable sequences tend to be in the middle, NGS platforms tend to be most reliable near the beginning of each read.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line227"></span>
<span id="line228"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Quality control usually involves:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line229"></span></span><span>&lt;<span class="start-tag">ul</span>&gt;</span><span>
<span id="line230"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>Calculating the number of reads before quality control</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line231"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>Calculating GC content, identifying over-represented sequences</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line232"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>Remove or trim reads containing adaptor sequences</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line233"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>Remove or trim reads containing low quality bases</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line234"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>Calculating the number of reads after quality control</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line235"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>Rechecking GC content, identifying over-represented sequences</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line236"></span></span><span>&lt;/<span class="end-tag">ul</span>&gt;</span><span>
<span id="line237"></span>
<span id="line238"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Quality control is necessary because:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line239"></span></span><span>&lt;<span class="start-tag">ul</span>&gt;</span><span>
<span id="line240"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>CPU time required for alignment and assembly is reduced</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line241"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>Data storage requirements are reduced</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line242"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>Reduce potential for bias in variant calling and/or de-novo assembly</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line243"></span></span><span>&lt;/<span class="end-tag">ul</span>&gt;</span><span>
<span id="line244"></span>
<span id="line245"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Quality scores</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line246"></span>
<span id="line247"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Most quality scores are calculated using the Phred scale (</span><span>&lt;<span class="start-tag">em</span>&gt;</span><span>Ewing B, Green P</span><span>&lt;<span class="start-tag">a</span> <span class="attribute-name">href</span>="<a class="attribute-value" href="view-source:http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve&amp;db=pubmed&amp;dopt=Abstract&amp;list_uids=9521922">http://www.ncbi.nlm.nih.gov/entrez/query.fcgi?cmd=Retrieve<span class="entity"><span>&amp;</span>amp;</span>db=pubmed<span class="entity"><span>&amp;</span>amp;</span>dopt=Abstract<span class="entity"><span>&amp;</span>amp;</span>list_uids=9521922</a>"&gt;</span><span>: Basecalling of automated sequencer traces using phred. II. Error probabilities. </span><span>&lt;/<span class="end-tag">a</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">em</span>&gt;</span><span> </span><span>&lt;<span class="start-tag">em</span>&gt;</span><span>Genome Research 8:186-194 (1998)</span><span>&lt;/<span class="end-tag">em</span>&gt;</span><span>).</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line248"></span>
<span id="line249"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Each base call has an associated base call quality which estimates chance that the base call is incorrect.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line250"></span>
<span id="line251"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Q10 = 1 in 10 chance of incorrect base call</span><span>&lt;<span class="start-tag">br</span> <span>/</span>&gt;</span><span>
<span id="line252"></span>Q20 = 1 in 100 chance of incorrect base call</span><span>&lt;<span class="start-tag">br</span> <span>/</span>&gt;</span><span>
<span id="line253"></span>Q30 = 1 in 1000 chance of incorrect base call</span><span>&lt;<span class="start-tag">br</span> <span>/</span>&gt;</span><span>
<span id="line254"></span>Q40 = 1 in 10,000 chance of incorrect base call</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line255"></span>
<span id="line256"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>For most 454, SoLID and Illumina runs you should see quality scores between Q20 and Q40. Note that these as only estimates of base-quality based on calibration runs performed by the manufacturer against a sample of known sequence with (typically) a GC content of 50%. Extreme GC biases and/or particular motifs or homopolymers can cause the quality scores to
<span id="line257"></span>become unreliable. Accurate base qualities are an essential part in ensuring variant calls are correct. As a rough and ready rule we generally assume that with Illumina data anything less than Q20 is not useful data and should be excluded.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line258"></span>
<span id="line259"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Once you understand the FASTQ format try to work out what is happing to the quality scores here and why - A FASTQ entry consists of 4 lines:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line260"></span>
<span id="line261"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part2_fastq_01.png">/genomics/images/part2_fastq_01.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line262"></span>
<span id="line263"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>line 1</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span> - A header line beginning with  ‘@’ containing information about the name of the sequencer, and  the position at which the originating cluster was
<span id="line264"></span>located and whether it passed purity filters.</span><span>&lt;<span class="start-tag">br</span> <span>/</span>&gt;</span><span>
<span id="line265"></span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>line 2</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span> - The DNA sequence of the read</span><span>&lt;<span class="start-tag">br</span> <span>/</span>&gt;</span><span>
<span id="line266"></span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>line 3</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span> - A header line or line beginning with just ‘+’</span><span>&lt;<span class="start-tag">br</span> <span>/</span>&gt;</span><span>
<span id="line267"></span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>line 4</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span> - Quality scores for each base encoded in ASCII format</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line268"></span>
<span id="line269"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>To reduce storage requirements, the FASTQ quality scores are stored as single characters and converted to numbers by obtaining the ASCII quality score and subtracting either 33 (or 64 for much older datasets). For example, the above FASTQ file is Sanger formatted and the character ! has an ASCII value of 33. Therefore the corresponding base would have a Phred quality score of 33-33=Q0 (i.e. totally unreliable). On the other hand a base with a quality score denoted by @ has an ASCII value of 64 and would have a Phred quality score of 64-33=Q31 (i.e. less than 1/1000 chance of being incorrect).</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line270"></span>
<span id="line271"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Happily the community has pretty much standardized on base 33 encoding, but do be aware that older datasets may use a different offset for encoding.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line272"></span>
<span id="line273"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>The first task when one receives sequencing data is to evaluate its quality and determine whether all the cash you have handed over was well-spent! To do this we will use the FastQC toolkit </span><span>&lt;<span class="start-tag">a</span> <span class="attribute-name">href</span>="<a class="attribute-value" href="view-source:https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">https://www.bioinformatics.babraham.ac.uk/projects/fastqc/</a>"&gt;</span><span>(http://www.bioinformatics.bbsrc.ac.uk/projects/fastqc/).</span><span>&lt;/<span class="end-tag">a</span>&gt;</span><span> FastQC offers a graphical visualisation of QC metrics, but </span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>does not</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span> have the ability to filter data.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line274"></span>
<span id="line275"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Task 1</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span>
<span id="line276"></span>From your home directory change into the workshop_materials/genomics_tutorial/data/sequencing/ecoli_exeter/ directory and list the directory contents. E.g.:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line277"></span>
<span id="line278"></span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">pre</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">code</span>&gt;</span><span>cd ~/workshop_materials/genomics_tutorial/data/sequencing/ecoli_exeter
<span id="line279"></span>ls -l
<span id="line280"></span></span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">pre</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span>
<span id="line281"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part2_task1_01.png">/genomics/images/part2_task1_01.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line282"></span>
<span id="line283"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Note that this is a paired-end run. As such there are two files:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line284"></span></span><span>&lt;<span class="start-tag">ul</span>&gt;</span><span>
<span id="line285"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>one for read 1 (E_Coli_CGATGT_L001_R1_001.fastq)</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line286"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>and the other for the reverse read 2 (E_Coli_CGATGT_L001_R2_001.fastq)</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line287"></span></span><span>&lt;/<span class="end-tag">ul</span>&gt;</span><span>
<span id="line288"></span>
<span id="line289"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Reads from the same pair can be identified because they have the same header. Programs which use FASTQ files generally require that the read 1 and read 2 files have the reads in the same order. To view the first few headers we can use the head and grep commands:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line290"></span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">pre</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">code</span>&gt;</span><span>head E_Coli_CGATGT_L001_R1_001.fastq | grep MISEQ
<span id="line291"></span>head E_Coli_CGATGT_L001_R2_001.fastq | grep MISEQ  
<span id="line292"></span></span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">pre</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span>
<span id="line293"></span>
<span id="line294"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part2_task1_02.png">/genomics/images/part2_task1_02.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line295"></span>
<span id="line296"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>The only difference in the headers for the two reads is the read number. Of course this is no guarantee that all the headers in the file are consistent. To get some more confidence repeat the above commands using ‘tail’ instead of ‘head’ to compare reads at the end of the files.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line297"></span>
<span id="line298"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>You can also check that there is an identical number of reads in each file using cat, grep and wc -l:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line299"></span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">pre</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">code</span>&gt;</span><span>cat E_Coli_CGATGT_L001_R1_001.fastq | grep MISEQ | wc -l
<span id="line300"></span>cat E_Coli_CGATGT_L001_R2_001.fastq | grep MISEQ | wc -l
<span id="line301"></span></span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">pre</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span>
<span id="line302"></span>
<span id="line303"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Task 2</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line304"></span>
<span id="line305"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Now, let’s start the fastqc program.</span><span>&lt;<span class="start-tag">br</span> <span>/</span>&gt;</span><span>
<span id="line306"></span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>fastqc</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line307"></span>
<span id="line308"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part2_task2_01.png">/genomics/images/part2_task2_01.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line309"></span>
<span id="line310"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Load the E_Coli_CGATGT_L001_R1_001.fastq file from the
<span id="line311"></span>workshop_materials/genomics_tutorial/data/sequencing/ecoli_exeter directory.
<span id="line312"></span> </span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part2_task2_02.png">/genomics/images/part2_task2_02.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line313"></span>
<span id="line314"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>After a few minutes the program should finish analysing the FASTQ file.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line315"></span>
<span id="line316"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part2_task2_03.png">/genomics/images/part2_task2_03.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line317"></span>
<span id="line318"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>The fastqc program performs a number of tests which determines whether a green tick (pass), exclamation mark (warning) or red cross (fail) is displayed. However it is important to realise that fastqc has no knowledge of what your library is or should look like.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line319"></span>
<span id="line320"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>All of its tests are based on a completely random library with 50% GC content. Therefore if you have a sample which does not match these assumptions, it may ‘fail’ the library. For example, if you have a high AT or high GC organism it may fail the per sequence GC content. If you have any barcodes or low complexity libraries (e.g. small RNA libraries) they may also fail some of the sequence complexity tests.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line321"></span>
<span id="line322"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>The bottom line is that you need to be aware of what your library is and whether what fastqc is reporting makes sense for that type of library. In this case we have a number of errors and warnings which at first sight suggest there has been a problem - but don’t worry too much yet. Let’s go through them in turn.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line323"></span>
<span id="line324"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part2_task2_04.jpg">/genomics/images/part2_task2_04.jpg</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line325"></span>
<span id="line326"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Per Base Sequence Quality</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span> is one of the most important metrics. This view shows an overview of the range of quality values across all bases at each position in the FASTQ file.  Generally anything with a median quality score greater than Q20 is regarded as acceptable; anything above Q30 is regarded as ‘good’. For more details, see the help documentation in fastqc.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line327"></span>
<span id="line328"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>In this case this check is red - and it is true that the quality drops off at the end of the reads. It is normal for read quality to get worse towards the end of the read. You can see that at 250 bases the quality is still very good, we will later trim off the low quality bases so reserve judgment for now.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line329"></span>
<span id="line330"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part2_task2_05.jpg">/genomics/images/part2_task2_05.jpg</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span>
<span id="line331"></span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Per tile sequence quality</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span> is a purely technical view on the sequencing run, it is more important for the team running the sequencer. The sequencing flowcell is divided up into areas called cells. You can see that the read quality drops off in some cells faster than others. This maybe because of the way the sample flowed over the flowcell or a mark or smear on the lens of the optics.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line332"></span>
<span id="line333"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part2_task2_06.png">/genomics/images/part2_task2_06.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span>
<span id="line334"></span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Per base sequence content</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span>
<span id="line335"></span>For a completely randomly generated library with a GC content of 50% one expects that at any given position within a read there will be a 25% chance of finding an A, C, T or G base. Here we can see that our library satisfies these criteria, although there appears to be some minor bias at the beginning of the read. This may be due to PCR duplicates during amplification or during library preparation. It is
<span id="line336"></span>unlikely that one will ever see a perfectly uniform distribution. See </span><span>&lt;<span class="start-tag">a</span> <span class="attribute-name">href</span>="<a class="attribute-value" href="view-source:http://sequencing.exeter.ac.uk/guide-to-your-data/quality-control">http://sequencing.exeter.ac.uk/guide-to-your-data/quality-control</a>"&gt;</span><span>http://sequencing.exeter.ac.uk/guide-to-your-data/quality-control </span><span>&lt;/<span class="end-tag">a</span>&gt;</span><span>
<span id="line337"></span>for examples of good vs bad runs as well as the fastqc help for more details.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line338"></span>
<span id="line339"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part2_task2_06.png">/genomics/images/part2_task2_06.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span>
<span id="line340"></span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Sequence Duplication Levels</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span> In a library that covers a whole genome uniformly most sequences will occur only once in the final set. A low level of duplication may indicate a very high level of coverage of the target sequence, but a high level of duplication is more likely to indicate some kind of enrichment bias (e.g. PCR over- amplification).  This module counts the degree of duplication for every sequence in the set and creates a plot showing the relative number of sequences with different degrees of duplication.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line341"></span>
<span id="line342"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part2_task2_07.png">/genomics/images/part2_task2_07.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span>
<span id="line343"></span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Overrepresented Sequences</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span>
<span id="line344"></span> This check for sequences that occur more frequently than expected in your data. It also checks any sequences it finds against a small database of known sequences. In this case it has found that a small number of reads 4000 out of 600000 appear to contain a sequence used in the preparation for the library. A typical cause is that the original DNA was shorter than the length of the read - so the sequencing overruns the actual DNA and runs in to the adaptors used to bind it to the flowcell.  At this level there is nothing to worry about - they will be trimmed in later stages.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line345"></span>
<span id="line346"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>There are other reports available -  Have a look at them and at what the author of FastQC has to say.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line347"></span>
<span id="line348"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">a</span> <span class="attribute-name">href</span>="<a class="attribute-value" href="view-source:https://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/3%20Analysis%20Modules//">https://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/3%20Analysis%20Modules//</a>"&gt;</span><span>https://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/3%20Analysis%20Modules/ </span><span>&lt;/<span class="end-tag">a</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line349"></span>
<span id="line350"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Remember the error and warning flags are his (albeit experienced) judgement of what typical data should look like. It is up to you to use some initiative and understand whether what you are seeing is typical for your dataset and how that might affect any analysis you are performing.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line351"></span>
<span id="line352"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Do the same for read 2 as we have for read 1. Open fastqc and analyse the read 2 file. Look at the various plots and metrics which are generated. How similar are they? Note that the number of reads reported in both files is identical. This is because if one read fails to pass the Illumina chastity filter, its partner is automatically excluded too.
<span id="line353"></span>Overall, both read 1 and read 2 can be regarded as ‘good’ data-sets.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line354"></span>
<span id="line355"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Quality filtering of Illumina data</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line356"></span>
<span id="line357"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>In this section we will be filtering the data to ensure any low quality reads are removed and that any sequences containing adaptor sequences are either trimmed or removed altogether. To do this we will use the fastq-mcf program from the ea-utils package (available at </span><span>&lt;<span class="start-tag">a</span> <span class="attribute-name">href</span>="<a class="attribute-value" href="view-source:https://expressionanalysis.github.io/ea-utils/">https://expressionanalysis.github.io/ea-utils/</a>"&gt;</span><span>https://expressionanalysis.github.io/ea-utils/</span><span>&lt;/<span class="end-tag">a</span>&gt;</span><span>).
<span id="line358"></span>This package is remarkably fast and ensures that after filtering both read 1 and read 2 files are in the correct order.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line359"></span>
<span id="line360"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">em</span>&gt;</span><span>Note: Typically when submitting raw Illumina data to NCBI or EBI you would submit unfiltered data, so don’t delete your original fastq files!</span><span>&lt;/<span class="end-tag">em</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line361"></span>
<span id="line362"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>A note on checking for contaminants:</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line363"></span>
<span id="line364"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>A number of tools are available now which also enable to you to quickly search reads and assign them to particular species or taxonomic groups. These can serve as a quick check to make sure your samples or libraries are not contaminated with DNA from other sources. If you are performing a de-novo assembly for example and have unwittingly have DNA sequence present from multiple organisms, you will risk poor results and chimeric contigs.
<span id="line365"></span>Some ‘contaminants’ can turn out to be inevitable by-products of sampling and DNA extraction. This is often the case with algae or other symbionts. In addition some groups have made some amazing discoveries such as the discovery of a third symbiont (which turned out to be a yeast) in lichen. </span><span>&lt;<span class="start-tag">a</span> <span class="attribute-name">href</span>="<a class="attribute-value" href="view-source:http://science.sciencemag.org/content/353/6298/488.full">http://science.sciencemag.org/content/353/6298/488.full</a>"&gt;</span><span>http://science.sciencemag.org/content/353/6298/488.full </span><span>&lt;/<span class="end-tag">a</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line366"></span>
<span id="line367"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Some tools you can use to check the taxonomic classification of reads include We won’t do this now but will do an example in the final section of the workshop on long reads.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line368"></span></span><span>&lt;<span class="start-tag">ul</span>&gt;</span><span>
<span id="line369"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span></span><span>&lt;<span class="start-tag">a</span> <span class="attribute-name">href</span>="<a class="attribute-value" href="view-source:https://ccb.jhu.edu/software/kraken/">https://ccb.jhu.edu/software/kraken/</a>"&gt;</span><span>Kraken</span><span>&lt;/<span class="end-tag">a</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line370"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span></span><span>&lt;<span class="start-tag">a</span> <span class="attribute-name">href</span>="<a class="attribute-value" href="view-source:https://ccb.jhu.edu/software/centrifuge/manual.shtml">https://ccb.jhu.edu/software/centrifuge/manual.shtml</a>"&gt;</span><span>Centrifuge</span><span>&lt;/<span class="end-tag">a</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line371"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span></span><span>&lt;<span class="start-tag">a</span> <span class="attribute-name">href</span>="<a class="attribute-value" href="view-source:https://github.com/blaxterlab/blobology">https://github.com/blaxterlab/blobology</a>"&gt;</span><span>Blobology</span><span>&lt;/<span class="end-tag">a</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line372"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>Blast (in conjunction with sub-sampling your reads) and Krona to plot results</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line373"></span></span><span>&lt;/<span class="end-tag">ul</span>&gt;</span><span>
<span id="line374"></span>
<span id="line375"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Task 3</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line376"></span>
<span id="line377"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Make sure you are in the correct directory:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line378"></span>
<span id="line379"></span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">pre</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">code</span>&gt;</span><span>cd ~/workshop_materials/genomics_tutorial/data/sequencing/ecoli_exeter/
<span id="line380"></span></span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">pre</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span>
<span id="line381"></span>
<span id="line382"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>We will execute the fastq-mcf program which performs both adaptor sequence trimming and low quality bases. To remove adaptor sequences, we need to supply the adaptor sequences to the program. A list of the most common adaptors used is given in the file:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line383"></span>
<span id="line384"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>~/workshop_materials/genomics_tutorial/data/reference/adaptors/adaptors.fasta</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line385"></span>
<span id="line386"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>View it by typing:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line387"></span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">pre</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">code</span>&gt;</span><span>less ~/workshop_materials/genomics_tutorial/data/reference/adaptors/adaptors.fasta  
<span id="line388"></span></span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">pre</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span>
<span id="line389"></span>
<span id="line390"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part2_task3_01.png">/genomics/images/part2_task3_01.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line391"></span>
<span id="line392"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Now to run the fastq-mcf program, type the following (all on one line):</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line393"></span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">pre</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">code</span>&gt;</span><span>fastq-mcf ../../reference/adaptors/adaptors.fasta E_Coli_CGATGT_L001_R1_001.fastq E_Coli_CGATGT_L001_R2_001.fastq -o E_Coli_CGATGT_L001_R1_001.filtered.fastq -o E_Coli_CGATGT_L001_R2_001.filtered.fastq -C 1000000 -q 20 -p 10 -u -x 0.01
<span id="line394"></span></span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">pre</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span>
<span id="line395"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part2_task3_02.png">/genomics/images/part2_task3_02.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line396"></span>
<span id="line397"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>While this is running enter the command fastq-mcf in another terminal and try to understand what the options we are using do. We have found that these parameters generally work well for Illumina data. If you would like to learn more about these options, you can look at the manual here: </span><span>&lt;<span class="start-tag">a</span> <span class="attribute-name">href</span>="<a class="attribute-value" href="view-source:https://github.com/ExpressionAnalysis/ea-utils/blob/wiki/FastqMcf.md">https://github.com/ExpressionAnalysis/ea-utils/blob/wiki/FastqMcf.md</a>"&gt;</span><span>https://github.com/ExpressionAnalysis/ea-utils/blob/wiki/FastqMcf.md</span><span>&lt;/<span class="end-tag">a</span>&gt;</span><span>.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line398"></span>
<span id="line399"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>In short we are using 1 million reads to form a model of the sequence quality and then applying the filters which remove bases with q-score less than 20, trims adaptors allowing up to 10% mismatch in the adaptor sequence, allowing only pass-filter reads (virtually all sequencing data is pass-filter these days, so this is just included to be safe) and trims back reads which contain more than 1% Ns until they contain 1% or less Ns.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line400"></span>
<span id="line401"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>After a few minutes the filtering should be complete and you should see something similar to:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line402"></span>
<span id="line403"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part2_task3_03.png">/genomics/images/part2_task3_03.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line404"></span>
<span id="line405"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>You can see that the trimming has been harsher on the R2 reads than on the R1 - this is generally to be expected in Illumina paired end runs. If we look at the sizes of the files produced:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line406"></span>
<span id="line407"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>ls -l</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line408"></span>
<span id="line409"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part2_task3_04.png">/genomics/images/part2_task3_04.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line410"></span>
<span id="line411"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>You can see that the original files are exactly the same size, but the R2 filtered file is smaller than R1. Now count the lines in all the files:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line412"></span>
<span id="line413"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>wc -l *.filtered.fastq</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line414"></span>
<span id="line415"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part2_task3_05.png">/genomics/images/part2_task3_05.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line416"></span>
<span id="line417"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Although the reads have been trimmed differently - the number of reads in the R1 and R2 files are identical. This is required for all the tools we will use to analyse paired end data.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line418"></span>
<span id="line419"></span></span><span>&lt;<span class="start-tag">hr</span> <span>/</span>&gt;</span><span>
<span id="line420"></span>
<span id="line421"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Task 4</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line422"></span>
<span id="line423"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Check the quality scores and sequence distribution in the fastqc program for the two filtered fastq files. You should notice very little change (since comparatively few reads were filtered).</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line424"></span>
<span id="line425"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>However, you should notice a significant improvement in quality and the absence of adaptor sequences.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line426"></span>
<span id="line427"></span></span><span>&lt;<span class="start-tag">hr</span> <span>/</span>&gt;</span><span>
<span id="line428"></span>
<span id="line429"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Task 5</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line430"></span>
<span id="line431"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>We can perform a quick check (although this by no means guarantees) that the sequences in read 1 and read 2 are in the same order by checking the ends of the two files and making sure that the headers are the same.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line432"></span>
<span id="line433"></span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">pre</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">code</span>&gt;</span><span>head E_Coli_CGATGT_L001_R1_001.filtered.fastq | grep MISEQ
<span id="line434"></span>head E_Coli_CGATGT_L001_R2_001.filtered.fastq | grep MISEQ
<span id="line435"></span>tail E_Coli_CGATGT_L001_R1_001.filtered.fastq | grep MISEQ
<span id="line436"></span>tail E_Coli_CGATGT_L001_R2_001.filtered.fastq | grep MISEQ
<span id="line437"></span></span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">pre</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span>
<span id="line438"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part2_task5_01.png">/genomics/images/part2_task5_01.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line439"></span>
<span id="line440"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Check the number of reads in each filtered file. They should be the same. To do this use the grep command to search for the number of times the header appears. E.g:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line441"></span>
<span id="line442"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>grep -c MISEQ E_Coli_CGATGT_L001_R1_001.filtered.fastq</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line443"></span>
<span id="line444"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Do the same for the E_Coli_CGATGT_L001_R2_001.filtered.fastq file.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line445"></span>
<span id="line446"></span></span><span>&lt;<span class="start-tag">hr</span> <span>/</span>&gt;</span><span>
<span id="line447"></span>
<span id="line448"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Task 6</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line449"></span>
<span id="line450"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Aligning Illumina data to a reference sequence</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line451"></span>
<span id="line452"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Now that we have checked the quality of our raw data, we can begin to align the reads against a reference sequence. In this way we can compare how the reference sequence and the strain we have sequenced compare.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line453"></span>
<span id="line454"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>To do this we will be using a program called BWA (Burrows Wheeler Aligner </span><span>&lt;<span class="start-tag">em</span>&gt;</span><span>Li H. and Durbin R. (2009) Fast and accurate short read alignment with Burrows-Wheeler Transform. Bioinformatics, 25:1754-60.</span><span>&lt;/<span class="end-tag">em</span>&gt;</span><span> ). This uses an algorithm called (unsurprisingly) Burrows Wheeler to rapidly map reads to the reference genome. BWA also allows for a certain number of mismatches to account for variants which may be present in strain 1 vs the reference genome. Unlike other alignment packages such as Bowtie (version 1) BWA allows for insertions or deletions as well.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line455"></span>
<span id="line456"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>By mapping reads against a reference, what we mean is that we want to go from a FASTQ file listing lots of reads, to another type of file (which we’ll describe later) which lists the reads AND where/if it maps against the reference genome.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line457"></span>
<span id="line458"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>The figure below illustrates what we are trying to achieve here. Along the top in grey is the reference sequence. The coloured sequences below indicate individual sequences and how they map to the reference. If there is a real variant in a bacterial genome we would expect that (nearly) all the reads would contain the variant at the relevant position rather than the same base as the reference genome. Remember that error rates for any single read on second generation platforms tend to be around 0.5-1%. Therefore a 300bp read is on average likely to contain at 2-3 errors.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line459"></span>
<span id="line460"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Let’s look at 2 potential sources of artefacts.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line461"></span>
<span id="line462"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Sequencing error:</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line463"></span>
<span id="line464"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part2_task6_01.png">/genomics/images/part2_task6_01.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line465"></span>
<span id="line466"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>The region highlighted in green above shows that most reads agree with the reference sequence (i.e. C-base). However, 2 reads near the bottom show an A-base. In this situation we can safely assume that the A-bases are due to a sequencing error rather than a genuine variant since the ‘variant’ has only one read supporting it. If this occurred at a higher frequency however, we would struggle to determine whether it was a genuine variant or an error.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line467"></span>
<span id="line468"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>PCR duplication:</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line469"></span>
<span id="line470"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>The highlighted region red above shows where there appears to be a variant. A C-base is present in the reference and half the reads, whilst an A-base is present in a set of reads which all start at the same position.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line471"></span>
<span id="line472"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Is this a genuine difference or a sequencing or sample prep error? What do you think? If this was a real sample, would you expect all the reads containing an A to start at the same location?</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line473"></span>
<span id="line474"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>The answer is probably not. This ‘SNP’ is in fact probably an artefact of PCR duplication. I.e. the same fragment of DNA has been replicated many times more than the average and happens to contain an error at the first position. We can filter out such reads during after alignment to the reference (see later).</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line475"></span>
<span id="line476"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Note that the entire region above seems to contain lots of PCR duplicates with reads starting at the same location. In the case of the region highlighted in red, this will likely cause a false SNP call. The area in green also contains PCR duplicates the As at these positions are probably either sequencing errors or errors introduced during PCR.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line477"></span>
<span id="line478"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>It’s always important to think critically about any finding - don’t assume that whatever bioinformatic tools you are using are perfect. Or that you have used them perfectly.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line479"></span>
<span id="line480"></span></span><span>&lt;<span class="start-tag">hr</span> <span>/</span>&gt;</span><span>
<span id="line481"></span>
<span id="line482"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Indexing a reference genome:</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line483"></span>
<span id="line484"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Before we can start aligning reads to a reference genome, the genome sequence needs to be indexed. This means sorting the genome into easily searched chunks.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line485"></span>
<span id="line486"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Task 7</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line487"></span>
<span id="line488"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Generating an index file from the reference sequence</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line489"></span>
<span id="line490"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Change directory to the reference directory, and list the files:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line491"></span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">pre</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">code</span>&gt;</span><span>cd ~/workshop_materials/genomics_tutorial/data/reference/U00096/
<span id="line492"></span>ls -l
<span id="line493"></span></span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">pre</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span>
<span id="line494"></span>
<span id="line495"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part2_task7_01.png">/genomics/images/part2_task7_01.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line496"></span>
<span id="line497"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>In this directory we have 2 files. U00096.fna is a FASTA file which contains the reference genome sequence. The U00096.gff file contains the annotation for this genome. We will use this later.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line498"></span>
<span id="line499"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>First, let’s looks at the bwa command itself. Type:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line500"></span>
<span id="line501"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>bwa</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line502"></span>
<span id="line503"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>This should yield something like:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line504"></span>
<span id="line505"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part2_task7_02.png">/genomics/images/part2_task7_02.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line506"></span>
<span id="line507"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>BWA is actually a suite of programs which all perform different functions. We are only going to use two during this workshop, bwa index, bwa mem</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line508"></span>
<span id="line509"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>If we type:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line510"></span>
<span id="line511"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>bwa index</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line512"></span>
<span id="line513"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>We can see more options for the bwa index command.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line514"></span>
<span id="line515"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part2_task7_03.png">/genomics/images/part2_task7_03.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line516"></span>
<span id="line517"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>By default bwa index will use the IS algorithm to produce the index. This works well for most genomes, but for very large ones (e.g. vertebrate) you may need to use bwtsw. For bacterial genomes the default algorithm will work fine.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line518"></span>
<span id="line519"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Now we will create a reference index for the genome using BWA:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line520"></span>
<span id="line521"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>bwa index U00096.fna</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line522"></span>
<span id="line523"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part2_task7_04.png">/genomics/images/part2_task7_04.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line524"></span>
<span id="line525"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>If you now list the directory contents using the ‘ls’ command, you will notice that the BWA index program has created a set of new files. These are the index files BWA needs.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line526"></span>
<span id="line527"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part2_task7_05.png">/genomics/images/part2_task7_05.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line528"></span>
<span id="line529"></span></span><span>&lt;<span class="start-tag">hr</span> <span>/</span>&gt;</span><span>
<span id="line530"></span>
<span id="line531"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Task 8: Aligning reads to the indexed reference sequence.</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line532"></span>
<span id="line533"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Now we can begin to align read 1 and read 2 to the reference genome. First of all change back into the ~/workshop_materials/genomics_tutorial/data/sequencing/ecoli_exeter/ directory and create a subdirectory to contain our remapping results.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line534"></span>
<span id="line535"></span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">pre</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">code</span>&gt;</span><span>cd ~/workshop_materials/genomics_tutorial/data/sequencing/ecoli_exeter/
<span id="line536"></span>mkdir remapping_to_reference
<span id="line537"></span>cd remapping_to_reference
<span id="line538"></span></span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">pre</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span>
<span id="line539"></span>
<span id="line540"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Let’s explore the alignment options BWA MEM has to offer. Type:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line541"></span>
<span id="line542"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>bwa mem</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line543"></span>
<span id="line544"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part2_task8_01.png">/genomics/images/part2_task8_01.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line545"></span>
<span id="line546"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>The basis format of the command is:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line547"></span>
<span id="line548"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Usage: bwa mem [options] </span><span>&lt;<span class="start-tag">idxbase</span>&gt;</span><span> </span><span>&lt;<span class="start-tag">in1.fq</span>&gt;</span><span> [in2.fq]</span><span>&lt;/<span class="end-tag">in1.fq</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">idxbase</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line549"></span>
<span id="line550"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>We can see that we need to provide BWA with a FASTQ files containing the raw reads (denoted by </span><span>&lt;<span class="start-tag">in.fq</span>&gt;</span><span> and </span><span>&lt;<span class="start-tag">in2.fq</span>&gt;</span><span>) to align to a reference file (unhelpfully this is listed as </span><span>&lt;<span class="start-tag">idxbase</span>&gt;</span><span>).</span><span>&lt;/<span class="end-tag">idxbase</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">in2.fq</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">in.fq</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line551"></span>
<span id="line552"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>There are also a number of options. The most important are</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line553"></span></span><span>&lt;<span class="start-tag">ul</span>&gt;</span><span>
<span id="line554"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>the maximum number of differences in the seed (-k i.e. the first 32 bp of the sequence vs the reference),</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line555"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>the number of processors the program should use (-t - our machine has 8 processors).</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line556"></span></span><span>&lt;/<span class="end-tag">ul</span>&gt;</span><span>
<span id="line557"></span>
<span id="line558"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Our reference sequence is in
<span id="line559"></span>~/workshop_materials/genomics_tutorial/data/reference/U00096/U00096.fna</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line560"></span>
<span id="line561"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Our filtered reads in:</span><span>&lt;<span class="start-tag">br</span> <span>/</span>&gt;</span><span>
<span id="line562"></span>~/workshop_materials/genomics_tutorial/data/sequencing/ecoli_exeter/E_Coli_CGATGT_L001_R1_001.filtered.fastq</span><span>&lt;<span class="start-tag">br</span> <span>/</span>&gt;</span><span>
<span id="line563"></span>~/workshop_materials/genomics_tutorial/data/sequencing/ecoli_exeter/E_Coli_CGATGT_L001_R2_001.filtered.fastq</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line564"></span>
<span id="line565"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>So to align our paired reads using processors and output to file E_Coli_CGATGT_L001_filtered.sam:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line566"></span>
<span id="line567"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>type, all on one line:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line568"></span>
<span id="line569"></span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">pre</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">code</span>&gt;</span><span>bwa mem -t 8 ~/workshop_materials/genomics_tutorial/data/reference/U00096/U00096.fna ~/workshop_materials/genomics_tutorial/data/sequencing/ecoli_exeter/E_Coli_CGATGT_L001_R1_001.filtered.fastq ~/workshop_materials/genomics_tutorial/data/sequencing/ecoli_exeter/E_Coli_CGATGT_L001_R2_001.filtered.fastq <span class="entity"><span>&amp;</span>gt;</span> E_Coli_CGATGT_L001_filtered.sam
<span id="line570"></span></span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">pre</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span>
<span id="line571"></span>
<span id="line572"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>There will be quite a lot of output but the end should look like:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line573"></span>
<span id="line574"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part2_task8_02.png">/genomics/images/part2_task8_02.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line575"></span>
<span id="line576"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Once the alignment is complete, list the directory contents and check that the alignment file is present.
<span id="line577"></span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>ls -lh</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line578"></span>
<span id="line579"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part2_task8_03.png">/genomics/images/part2_task8_03.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line580"></span>
<span id="line581"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Note: </span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>ls -lh</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span> outputs the size of the file in human readable format (780Mb in this case - don’t worry if yours are slightly different.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line582"></span>
<span id="line583"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>The raw alignment is stored in what is called SAM format (Simple AlignMent format). It is in plain text format and you can view it if you wish using the ‘less’ command. Do not try to open the whole file in a text editor as you will likely run out of memory!</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line584"></span>
<span id="line585"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>less E_Coli_CGATGT_L001_filtered.sam</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line586"></span>
<span id="line587"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part2_task8_04.png">/genomics/images/part2_task8_04.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line588"></span>
<span id="line589"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Each alignment line has 11 mandatory fields for essential alignment information
<span id="line590"></span>such as mapping position, and a variable number of optional fields for flexible or aligner specific information. For further details as to what each field means see</span><span>&lt;<span class="start-tag">a</span> <span class="attribute-name">href</span>="<a class="attribute-value" href="view-source:http://samtools.sourceforge.net/SAM1.pdf">http://samtools.sourceforge.net/SAM1.pdf</a>"&gt;</span><span> http://samtools.sourceforge.net/SAM1.pdf </span><span>&lt;/<span class="end-tag">a</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line591"></span>
<span id="line592"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Task 9: Convert SAM to BAM file</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line593"></span>
<span id="line594"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Before we can visualise the alignment however, we need to convert the SAM file to a BAM (Binary AlignMent format) which can be read by most software analysis packages. To do this we will use another suite of programs called samtools. Type:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line595"></span>
<span id="line596"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>samtools view</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line597"></span>
<span id="line598"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part2_task9_01.png">/genomics/images/part2_task9_01.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line599"></span>
<span id="line600"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>We can see that we need to provide samtools view with a reference genome in FASTA format file (-T), the -b and -S flags to say that the output should be in BAM format and the input in SAM, plus the alignment file.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line601"></span>
<span id="line602"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Remember our reference sequence is in
<span id="line603"></span>~/workshop_materials/genomics_tutorial/data/reference/U00096/U00096.fna</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line604"></span>
<span id="line605"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Type (all on one line):</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line606"></span>
<span id="line607"></span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">pre</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">code</span>&gt;</span><span>samtools view -bS --threads 8 -T  ~/workshop_materials/genomics_tutorial/data/reference/U00096/U00096.fna E_Coli_CGATGT_L001_filtered.sam <span class="entity"><span>&amp;</span>gt;</span> E_Coli_CGATGT_L001_filtered.bam
<span id="line608"></span></span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">pre</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span>
<span id="line609"></span>
<span id="line610"></span></span><span>&lt;<span class="start-tag">blockquote</span>&gt;</span><span>
<span id="line611"></span>  </span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Aside: Many commands can run in multiple processes often called threads, which will reduce the runtime of many commands - for samtools it is the the --threads option. Out of interest try a few different values. You can prefix your command with </span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>time</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span> to capture the runtime easily.</span><span>&lt;<span class="start-tag">br</span> <span>/</span>&gt;</span><span>
<span id="line612"></span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>samtools view -bS -T --threads xx ..etc..</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;<span class="start-tag">br</span> <span>/</span>&gt;</span><span>
<span id="line613"></span>try a few different values e.g. 4,8,16 - do you understand the pattern?</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line614"></span></span><span>&lt;/<span class="end-tag">blockquote</span>&gt;</span><span>
<span id="line615"></span>
<span id="line616"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>ls -lh</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line617"></span>
<span id="line618"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part2_task9_02.png">/genomics/images/part2_task9_02.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line619"></span>
<span id="line620"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>It’s always good to check that your files have processed correctly if something goes wrong it’s better to catch it immediately.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line621"></span>
<span id="line622"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Note that the bam file is smaller than the sam file - this is to be expected as the binary format is more efficient.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line623"></span>
<span id="line624"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Task 10: Sort BAM file</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line625"></span>
<span id="line626"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Once this is complete we then need to sort the BAM file so that the reads are stored in the order they appear along the chromosomes (don’t ask me why this isn’t done automatically….). We can do this using the samtools sort command.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line627"></span>
<span id="line628"></span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">pre</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">code</span>&gt;</span><span>samtools sort --threads 8 E_Coli_CGATGT_L001_filtered.bam -o E_Coli_CGATGT_L001_filtered.sorted.bam
<span id="line629"></span></span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">pre</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span>
<span id="line630"></span>
<span id="line631"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>This will take another minute or so. Don’t forget to check the resulting file.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line632"></span>
<span id="line633"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part2_task10_01.png">/genomics/images/part2_task10_01.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line634"></span>
<span id="line635"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>A note on piping BWA and samtools commands:</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span>
<span id="line636"></span>In tasks 8-10 we aligned reads to the reference genome, converted SAM to BAM and then sorted the resulting BAM file. For clarity we have shown these as individual steps. However, in real-life, it is faster and easier to do these simultaneously using Unix pipes!</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line637"></span>
<span id="line638"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>E.g. (there is no need to do this)</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line639"></span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">pre</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">code</span>&gt;</span><span>bwa mem -t 2 ~/workshop_materials/genomics_tutorial/data/reference/U00096/U00096.fna ~/workshop_materials/genomics_tutorial/data/sequencing/ecoli_exeter/E_Coli_CGATGT_L001_R1_001.filtered.fastq ~/workshop_materials/genomics_tutorial/data/sequencing/ecoli_exeter/E_Coli_CGATGT_L001_R2_001.filtered.fastq <span class="entity"><span>&amp;</span>gt;</span> E_Coli_CGATGT_L001_filtered.sam | samtools sort -O bam -o E_Coli_CGATGT_L001_filtered.sorted.bam
<span id="line640"></span></span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">pre</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span>
<span id="line641"></span>
<span id="line642"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Task 11: Remove suspected PCR duplicates</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line643"></span>
<span id="line644"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Especially when using paired-end reads, samtools can do a reasonably good job of removing potential PCR duplicates (see the first part of this workshop if you are unsure what this means).</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line645"></span>
<span id="line646"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Again, samtools has a command to do this called rmdup.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line647"></span>
<span id="line648"></span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">pre</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">code</span>&gt;</span><span>samtools rmdup E_Coli_CGATGT_L001_filtered.sorted.bam E_Coli_CGATGT_L001_filtered.sorted.rmdup.bam
<span id="line649"></span></span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">pre</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span>
<span id="line650"></span>
<span id="line651"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part2_task11_01.png">/genomics/images/part2_task11_01.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line652"></span>
<span id="line653"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>You will notice some warnings about inconsistent BAM file for pair - this is just a warning that a pair of reads does not align together on the genome within the expected tolerance - it is normal to expect some of these, and you can ignore.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line654"></span>
<span id="line655"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Task 12: Index the BAM file</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line656"></span>
<span id="line657"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Most programs used to view BAM formatted data require an index file to locate the reads mapping to a particular location quickly. You can think of this as an index in a book, telling you where to go to find particular phrases or words. We’ll use the samtools index command to do this.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line658"></span>
<span id="line659"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>samtools index E_Coli_CGATGT_L001_filtered.sorted.rmdup.bam</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line660"></span>
<span id="line661"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>We should obtain a .bai file (known as a BAM-index file).
<span id="line662"></span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part2_task12_01.png">/genomics/images/part2_task12_01.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line663"></span>
<span id="line664"></span></span><span>&lt;<span class="start-tag">hr</span> <span>/</span>&gt;</span><span>
<span id="line665"></span>
<span id="line666"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Task 13: Obtain mapping statistics</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line667"></span>
<span id="line668"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Finally we can obtain some summary statistics.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line669"></span>
<span id="line670"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>samtools flagstat E_Coli_CGATGT_L001_filtered.sorted.rmdup.bam <span class="entity"><span>&amp;</span>gt;</span> mappingstats.txt</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line671"></span>
<span id="line672"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part2_task13_01.png">/genomics/images/part2_task13_01.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line673"></span>
<span id="line674"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>This should only take a few seconds. Once complete view the mappingstats.txt file using a text-editor (e.g. gedit or nano) or the ‘more’ command.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line675"></span>
<span id="line676"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part2_task13_02.png">/genomics/images/part2_task13_02.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line677"></span>
<span id="line678"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>So here we can see we have 1250574 reads in total, none of which failed QC.
<span id="line679"></span>71.88% of reads mapped to the reference genome and 71.55% mapped with the expected 500-600bp distance between them. 1414 reads could not have their read-pair mapped.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line680"></span>
<span id="line681"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>0 reads have mapped to a different chromosome than their pair (0 has a mapping quality <span class="entity"><span>&amp;</span>gt;</span> 5 - this is a Phred scaled quality score much as we say in the FASTQ files). If there were any such reads they would likely due to repetitive sequences (e.g phage insertion sites) or an insertion of plasmid or phage DNA into the main chromosome.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line682"></span>
<span id="line683"></span></span><span>&lt;<span class="start-tag">hr</span> <span>/</span>&gt;</span><span>
<span id="line684"></span>
<span id="line685"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Task 14: Cleanup</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span>
<span id="line686"></span>We have a number of leftover intermediate files which we can now remove to save space.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line687"></span>
<span id="line688"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Type (all on one line):</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line689"></span>
<span id="line690"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>rm E_Coli_CGATGT_L001_filtered.sam E_Coli_CGATGT_L001_filtered.bam E_Coli_CGATGT_L001_filtered.sorted.bam</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line691"></span>
<span id="line692"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>In case you get asked if you are sure to remove 3 arguments type in ‘yes’ and hit enter. You should now be left with the processed alignment file, the index file
<span id="line693"></span>and the mapping stats.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line694"></span>
<span id="line695"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Well done! You have now mapped, filtered and sorted your first whole
<span id="line696"></span>genome data-set! Let’s take a look at it!</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line697"></span>
<span id="line698"></span></span><span>&lt;<span class="start-tag">hr</span> <span>/</span>&gt;</span><span>
<span id="line699"></span>
<span id="line700"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Task 15: QualiMap</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line701"></span>
<span id="line702"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Qualimap (http://qualimap.bioinfo.cipf.es/) is a program that summarises the alignment in much more detail than the mapping stats file we
<span id="line703"></span>produced. It’s a technical tool which allows you to assess the sequencing for any problems and biases in the sequencing and the alignment rather than a tool to deduce biological features.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line704"></span>
<span id="line705"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>There are a few options to the program, We want to run bamqc. Type: </span><span>&lt;<span class="start-tag">br</span> <span>/</span>&gt;</span><span>
<span id="line706"></span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>qualimap bamqc</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;<span class="start-tag">br</span> <span>/</span>&gt;</span><span>
<span id="line707"></span>to get some help on this command.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line708"></span>
<span id="line709"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>To get the report, first make sure you are in the directory:
<span id="line710"></span>~/workshop_materials/genomics_tutorial/data/sequencing/ecoli_exeter/remapping_to_reference
<span id="line711"></span>then run the command:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line712"></span>
<span id="line713"></span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">pre</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">code</span>&gt;</span><span>qualimap bamqc -outdir bamqc -bam E_Coli_CGATGT_L001_filtered.sorted.rmdup.bam -gff ~/workshop_materials/genomics_tutorial/data/reference/U00096/U00096.gff
<span id="line714"></span></span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">pre</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span>
<span id="line715"></span>
<span id="line716"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>this creates a subfolder called bamqc</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line717"></span>
<span id="line718"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part2_task15_01.png">/genomics/images/part2_task15_01.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line719"></span>
<span id="line720"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>cd to this directory and run</span><span>&lt;<span class="start-tag">br</span> <span>/</span>&gt;</span><span>
<span id="line721"></span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>firefox qualimapReport.html</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line722"></span>
<span id="line723"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>There is a lot in the report so just a few highlights:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line724"></span>
<span id="line725"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part2_task15_02.jpg">/genomics/images/part2_task15_02.jpg</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line726"></span>
<span id="line727"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>This shows the numbe  of reads that ‘cover’ each section of the genome. The red line shows a
<span id="line728"></span>rolling average around 50x - this means that on average every part of the genome was sequenced 50X. It is important to have sufficient depth of coverage in order to be confident that any features you find in your data are real and not a result of sequencing errors.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line729"></span>
<span id="line730"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>What do you think the regions of low/zero coverage correspond to?</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line731"></span>
<span id="line732"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part2_task15_03.jpg">/genomics/images/part2_task15_03.jpg</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line733"></span>
<span id="line734"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>The Insert Size Histogram displays the range of sizes of the DNA fragments. It shows how well your DNA was size selected before sequencing. Note that the ‘insert’ refers to the DNA that was inserted between the sequencing adaptors, so equates to the size range of the DNA that was used. In this case we have 300 paired end reads and our insert size varies around 600 bases - so there should only be a small gap between the reads that was not sequenced.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line735"></span>
<span id="line736"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Have a look at some of the other graphs produced.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line737"></span>
<span id="line738"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Task 16: Load the Integrative Genomics Viewer</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line739"></span>
<span id="line740"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>The Integrative Genome Viewer (IGV) is a tool developed by the Broad Institute for browsing interactively the alignment data you produced. It has a wealth of features and we can only cover some basics to get you started. Go to http://www.broadinstitute.org/igv/ to get more information.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line741"></span>
<span id="line742"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>In your terminal, type </span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>igv.sh</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span> Or you can click the icon on the desktop.
<span id="line743"></span>IGV viewer should appear:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line744"></span>
<span id="line745"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part2_task16_01.png">/genomics/images/part2_task16_01.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line746"></span>
<span id="line747"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Notice that by default a human genome has been loaded.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line748"></span>
<span id="line749"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Task 17: Import the E.coli U0009 reference genome to IGV</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line750"></span>
<span id="line751"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>By default IGV does not contain our reference genome. We’ll need to
<span id="line752"></span>import it.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line753"></span>
<span id="line754"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Click on ‘Genomes -<span class="entity"><span>&amp;</span>gt;</span>Create .genome file…’</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line755"></span>
<span id="line756"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part2_task17_01.png">/genomics/images/part2_task17_01.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line757"></span>
<span id="line758"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Enter the information above and click on ‘OK’ .</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line759"></span>
<span id="line760"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>IGV will ask where it can save the genome file. Your home
<span id="line761"></span>directory will be fine.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line762"></span>
<span id="line763"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part2_task17_02.png">/genomics/images/part2_task17_02.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line764"></span>
<span id="line765"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Click ‘Save’ again.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line766"></span>
<span id="line767"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Note that the genome and the annotation have now been imported.
<span id="line768"></span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part2_task17_03.png">/genomics/images/part2_task17_03.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span>
<span id="line769"></span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part2_task17_04.png">/genomics/images/part2_task17_04.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line770"></span>
<span id="line771"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Task 18: Load the BAM file</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line772"></span>
<span id="line773"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Load the alignment file. Note that IGV requires the .bai index file to
<span id="line774"></span>also be in the same directory.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line775"></span>
<span id="line776"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Select File… and Load From File</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line777"></span>
<span id="line778"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part2_task18_01.png">/genomics/images/part2_task18_01.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line779"></span>
<span id="line780"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Select the bam file and click open</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line781"></span>
<span id="line782"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Once loaded your screen should look similar to the following. Note that you can load more BAM files if you wish to compare different samples or the results of different mapping programs.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line783"></span>
<span id="line784"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part2_task18_02.png">/genomics/images/part2_task18_02.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line785"></span>
<span id="line786"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Select the chromosome U00096.3 if it is not already selected</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line787"></span>
<span id="line788"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part2_task18_03.jpg">/genomics/images/part2_task18_03.jpg</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line789"></span>
<span id="line790"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Use the +/- keys to zoom in or use the zoom bar at the top right of the screen to zoom into about 1-2kbases as below:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line791"></span>
<span id="line792"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part2_task18_04.png">/genomics/images/part2_task18_04.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line793"></span>
<span id="line794"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Right click on the main area and select view as pairs</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line795"></span>
<span id="line796"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part2_task18_05.jpg">/genomics/images/part2_task18_05.jpg</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line797"></span>
<span id="line798"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>The gray graph at the top of the figure indicates the coverage of the genome:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line799"></span>
<span id="line800"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part2_task18_06.png">/genomics/images/part2_task18_06.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line801"></span>
<span id="line802"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>The more reads mapping to a certain location, the higher the peak on the graph. You’ll see a coloured line of blue, green or red in this coverage plot if there are any SNPs (single-nucleotide polymorphisms) present (there are none in the plot).</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line803"></span>
<span id="line804"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>If there are any regions in the genome which are not covered by the reads, you will see these as gaps in the coverage graph. Sometimes these gaps are caused by repetitive regions; others are caused by genuine insertions/deletions in your new strain with respect to the reference.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line805"></span>
<span id="line806"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Below the coverage graph is a representation of each read pair as it is mapped to the genome. One pair is highlighted.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line807"></span>
<span id="line808"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part2_task18_07.png">/genomics/images/part2_task18_07.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line809"></span>
<span id="line810"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>This pair consists of 2 reads with a gap (there may be no gap if the reads overlap) Any areas of mismatch either due to inconsistent distances between paired-end reads or due to differences between the reference and the read and are highlighted by a colour. The brighter the colour, the higher the base-calling quality is estimated to be. Differences in a single read are likely to be sequencing errors. Differences consistent in all reads are likely to be mutations.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line811"></span>
<span id="line812"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Hover over a read to get detailed information about the reads’ alignment:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line813"></span>
<span id="line814"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part2_task18_08.jpg">/genomics/images/part2_task18_08.jpg</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line815"></span>
<span id="line816"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>You don’t need to understand every value, but compare this to the SAM format to get an idea of what is there.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line817"></span>
<span id="line818"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>SNPs and Indels</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line819"></span>
<span id="line820"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>The following 3 tasks are open-ended. Please take your time with these. Read the examples on the following page if you get stuck.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line821"></span>
<span id="line822"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Task 19: Read about the alignment display format</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line823"></span>
<span id="line824"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Visit</span><span>&lt;<span class="start-tag">a</span> <span class="attribute-name">href</span>="<a class="attribute-value" href="view-source:http://www.broadinstitute.org/software/igv/AlignmentData">http://www.broadinstitute.org/software/igv/AlignmentData</a>"&gt;</span><span> http://www.broadinstitute.org/software/igv/AlignmentData
<span id="line825"></span></span><span>&lt;/<span class="end-tag">a</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line826"></span>
<span id="line827"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Task 20: Manually identify a region without any reads mapping.</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line828"></span>
<span id="line829"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>It can be quite difficult to find even with a very small genome. Zoom out as far as you can and still see the reads. Use the coverage plot from QualiMap to try to find it. Are there genes associated?</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line830"></span>
<span id="line831"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Because of the way IGV handles BAM files, it will not display coverage information if you zoom out too far. To get coverage information across the entire genome, regardless of how far you are zoomed out. You’ll  need to create a TDF file which contains a coverage information across windows of X number of bases across the genome. You can do this within IGV:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line832"></span>
<span id="line833"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Select Tools-<span class="entity"><span>&amp;</span>gt;</span>Run igvtools:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line834"></span>
<span id="line835"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part2_task20_01.png">/genomics/images/part2_task20_01.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line836"></span>
<span id="line837"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Now load the BAM alignment file in the Input field and click Run:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line838"></span>
<span id="line839"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part2_task20_02.png">/genomics/images/part2_task20_02.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line840"></span>
<span id="line841"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Once completed, you can load this TDF file as by:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line842"></span>
<span id="line843"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Select File -<span class="entity"><span>&amp;</span>gt;</span> Load from file</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line844"></span>
<span id="line845"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Select the TDF file you have just created.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line846"></span>
<span id="line847"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>You should then see the extra coverage track which remains visible
<span id="line848"></span>even after you zoom out.+</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line849"></span>
<span id="line850"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part2_task20_03.png">/genomics/images/part2_task20_03.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line851"></span>
<span id="line852"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Again try to use the QualiMap report to give you an idea. What is this region? Is there a gene close-by? What do you think this is? (Think about repetitive sequences, what does BWA do if a region in the genome has been duplicated)</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line853"></span>
<span id="line854"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Task 21: Identify SNPs and Indels manually</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line855"></span>
<span id="line856"></span></span><span>&lt;<span class="start-tag">ul</span>&gt;</span><span>
<span id="line857"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>Can you find any SNPs?</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line858"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>Which genes (if any) are they in?</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line859"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>How reliable do they look? (Hint: look at the number of reads mapping, their orientation - which strand they are on and how bright the base calls are).</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line860"></span></span><span>&lt;/<span class="end-tag">ul</span>&gt;</span><span>
<span id="line861"></span>
<span id="line862"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Zoom in to maximum magnification at the site of the SNP. Can you determine whether a SNP results in a synonymous (i.e. silent) or non-synonymous change in the amino acid? Can you use PDB </span><span>&lt;<span class="start-tag">a</span> <span class="attribute-name">href</span>="<a class="attribute-value" href="view-source:http://www.rcsb.org/pdb/home/home.do">http://www.rcsb.org/pdb/home/home.do</a>"&gt;</span><span>(http://www.rcsb.org/pdb/home/home.do)</span><span>&lt;/<span class="end-tag">a</span>&gt;</span><span> or other resources to determine whether or not this occurs in a catalytic site or other functionally crucial region? (Note this may not always be possible).</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line863"></span>
<span id="line864"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>What effect do you think this would have on the cell?</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line865"></span>
<span id="line866"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Here are some regions where there are differences in the organism sequenced and the reference: Can you interpret what has happened to the genome of our strain? Try to work out what is going on yourself before looking at the comment</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line867"></span>
<span id="line868"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Paste each of the genomic locations in this box and click go
<span id="line869"></span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part2_task21_01.png">/genomics/images/part2_task21_01.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line870"></span>
<span id="line871"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>U00096.3:2,108,392-2,133,153</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line872"></span>
<span id="line873"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>U00096.3:3,662,049-3,663,291</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line874"></span>
<span id="line875"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>U00096.3:4,296,249-4,296,510</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line876"></span>
<span id="line877"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>U00096.3:565,965-566,489</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line878"></span>
<span id="line879"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Region U00096.3:2,108,392-2,133,153</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line880"></span>
<span id="line881"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part2_task21_02.png">/genomics/images/part2_task21_02.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line882"></span>
<span id="line883"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>This area corresponds to the drop in coverage identified by Qualimap. It looks like a fairly large region of about 17K bases which was present in the reference and is missing from our sequenced genome. It looks like about 12 genes from the reference strain are not present in our strain - it this real or an artefact?</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line884"></span>
<span id="line885"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Well it is pretty conclusive we have coverage of about 60X either side of the deletion and nothing at all within. There are nice clean edges to the start and end of the deletion. We also have paired reads which span the deletion. This is exactly what you would expect if the two regions of coverage were actually joined together.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line886"></span>
<span id="line887"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Region U00096.3:3,662,049-3,663,291</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line888"></span>
<span id="line889"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part2_task21_03.png">/genomics/images/part2_task21_03.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line890"></span>
<span id="line891"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Zoom right in until you can see the reference sequence and protein sequence at the bottom of the display.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line892"></span>
<span id="line893"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>The first thing to note is that only discrepancies with respect to the reference are shown. If a read is entirely the same as its reference, it will appear entirely grey. Blue and red blocks indicate the presence of an ‘abnormal’ distance between paired-end reads. Note that unless this is consistent across most of the reads at a given position, it is not significant.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line894"></span>
<span id="line895"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Here we have a C-<span class="entity"><span>&amp;</span>gt;</span>T SNP. This changes the codon from CAG-<span class="entity"><span>&amp;</span>gt;</span>TAG (remember to check what strand the gene is on this one is on the forward strand, if it was on the reverse strand you would have to take the reverse complement of the codon to interpret the amino acid it codes for.) and results in a Gln-<span class="entity"><span>&amp;</span>gt;</span>Stop mutation in the final protein product which is very likely to change the effect of the protein product.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line896"></span>
<span id="line897"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Hover over the gene to get some more information from the annotation…</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line898"></span>
<span id="line899"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part2_task21_05.jpg">/genomics/images/part2_task21_05.jpg</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line900"></span>
<span id="line901"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Since it is a drug resistance protein it could be very significant.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line902"></span>
<span id="line903"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>One additional check is that the SNPs occur when reading the forward strand. We can check this by looking at the direction of the grey reads,or by hovering over the coverage graph - see previous diagram. We can see that approximately half of the bases reporting the C-<span class="entity"><span>&amp;</span>gt;</span>T mutation occur in read 1 (forward arrow), and half in read 2 (reverse arrow). This adds confidence to the base-call as it reduces the likelihood of this SNP being the result of a PCR duplication error.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line904"></span>
<span id="line905"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Note that sequencing errors in Illumina data are quite common (look at the odd bases showing up in the screen above. We rely on depth of sequencing to average out these errors. This is why people often mention that a minimum median coverage of 20-30x across the genome is required for accurate SNP-calling with Illumina data. This is not necessarily true for simple organisms such as prokaryotes, but for diploid and polyploid organisms it becomes important because each position may have one, two or many alleles changed.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line906"></span>
<span id="line907"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Region U00096.3:4,296,249-4,296,510</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line908"></span>
<span id="line909"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Much the same guidelines apply for indels as they do for SNPs. Here we have an insertion of two bases CG in our sample compared to the reference. Again, we can see how much confidence we have that the insertion is real by checking that the indel is found on both read 1 and read 2 and on both strands.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line910"></span>
<span id="line911"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>The insertion is signified by the presence of a purple bar. Hover your mouse over it to get more details as below</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line912"></span>
<span id="line913"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part2_task21_06.jpg">/genomics/images/part2_task21_06.jpg</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line914"></span>
<span id="line915"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>We can hover our mouse over the reference sequence to get details of the gene. We can see that it occurs in a repeat region and is unlikely to have very significant effects.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line916"></span>
<span id="line917"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>One can research the effect that a SNP or Indel may have by finding the relevant gene at </span><span>&lt;<span class="start-tag">a</span> <span class="attribute-name">href</span>="<a class="attribute-value" href="view-source:http://www.uniprot.org/">http://www.uniprot.org/</a>"&gt;</span><span>http://www.uniprot.org </span><span>&lt;/<span class="end-tag">a</span>&gt;</span><span>(or google ‘mdtF uniprot’ in the previous case).</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line918"></span>
<span id="line919"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>It should be clear from this quick exercise that trying to work out where SNPs and Indels are manually is a fairly tedious process if there are many mutations. As such, the next section will look at how to obtain spread-sheet friendly summary details of these.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line920"></span>
<span id="line921"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Region U00096.3:565,965-566,489</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line922"></span>
<span id="line923"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>This last region is more complex try to understand what genomic mutation could
<span id="line924"></span>account for this pattern - discuss with a colleague or an instructor.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line925"></span>
<span id="line926"></span></span><span>&lt;<span class="start-tag">hr</span> <span>/</span>&gt;</span><span>
<span id="line927"></span>
<span id="line928"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Recap: SNP/Indel identification</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line929"></span>
<span id="line930"></span></span><span>&lt;<span class="start-tag">ul</span>&gt;</span><span>
<span id="line931"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>Only changes from the reference sequence are displayed in IGV</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line932"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>Genuine SNPs/Indels should be present on both read 1 and read 2</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line933"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>Genuine SNPs/Indels should be present on both strand</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line934"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>Genuine SNPs/Indels should be supported by a good (i.e. 20-30x) depth of coverage</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line935"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>Very important mutations (i.e. ones relied upon in a paper) should be confirmed e.g. using PCR.</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line936"></span></span><span>&lt;/<span class="end-tag">ul</span>&gt;</span><span>
<span id="line937"></span>
<span id="line938"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Task 22: Identify SNPs and Indels using automated variant callers</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line939"></span>
<span id="line940"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Viewing alignments is useful when convincing yourself or others that a particular mutation is real rather than an artefact and for getting a feel for short read sequencing datasets. However, if we want to quickly and easily find variants we need to be able to generate lists of variants, in which gene they occur (if any) and what effect they have. We also need to know which (if any) genes are missing (i.e. have zero coverage).</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line941"></span>
<span id="line942"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>To call variants we can use a number of packages (e.g. VarScan, GATK). However here, we will show you how to use the bcftools package which comes with samtools. First we need to generate a pileup file which contains only locations with the variants and pass this to bcftools.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line943"></span>
<span id="line944"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Make sure you are in the correct directory.</span><span>&lt;<span class="start-tag">br</span> <span>/</span>&gt;</span><span>
<span id="line945"></span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>cd ~/workshop_materials/genomics_tutorial/data/sequencing/ecoli_exeter/remapping_to_reference</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;<span class="start-tag">br</span> <span>/</span>&gt;</span><span>
<span id="line946"></span>and type the following:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line947"></span>
<span id="line948"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>samtools mpileup</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line949"></span>
<span id="line950"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>You should see a screen similar to the following</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line951"></span>
<span id="line952"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part2_task22_01.png">/genomics/images/part2_task22_01.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line953"></span>
<span id="line954"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>If you are running this on datasets with large numbers of datasets with limited coverage where recombination is a factor, you can obtain increased sensitivity by passing all the BAM files to the variant caller simultaneously (hence the multiple BAM file options in samtools).</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line955"></span>
<span id="line956"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Type the following:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line957"></span>
<span id="line958"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>samtools mpileup -v -u -P Illumina --reference ~/workshop_materials/genomics_tutorial/data/reference/U00096/U00096.fna E_Coli_CGATGT_L001_filtered.sorted.rmdup.bam <span class="entity"><span>&amp;</span>gt;</span>  var.raw.vcf </span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line959"></span>
<span id="line960"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>This may take 10 minutes or so and will generate a VCF file containing the raw unfiltered variant calls for each position in the genome. Note that we are asking samtools mpileup to generate uncompressed VCF output with the -v and -u options. -P tells samtools it is dealing with Illumina data so that it can apply to the correct model to help account for mis-calls or indels.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line961"></span>
<span id="line962"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>This output by itself is not useful to us since it contains information on each position in the genome. So lets use the sister package of samtools, called bcftools to call what it thinks are the variant sites:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line963"></span>
<span id="line964"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>bcftools --help</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line965"></span>
<span id="line966"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part2_task22_02.png">/genomics/images/part2_task22_02.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line967"></span>
<span id="line968"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>bcftools call -c -v --ploidy 1 -O v -o var.called.vcf var.raw.vcf</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line969"></span>
<span id="line970"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Note that we are asking bcftools to call using assuming a ploidy of 1 and to output only the variant sites in VCF format. Using grep we can count how many sites were identified as being variant sites (i.e. sites with a potential mutation). We ask grep not to count lines beginning with a comment (#).</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line971"></span>
<span id="line972"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>grep -v -c  "^#" var.called.vcf</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line973"></span>
<span id="line974"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>You should find 320 or so sites.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line975"></span>
<span id="line976"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Now we just need to filter this a bit further to ensure we only retain regions where we have <span class="entity"><span>&amp;</span>gt;</span>90% allele frequency:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line977"></span>
<span id="line978"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>vcftools --min-alleles 2 --max-alleles 2 --non-ref-af 0.9 --vcf var.called.vcf --recode --out var.called.filt</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line979"></span>
<span id="line980"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>(the output will be called var.called.filt.recode.vcf)</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line981"></span>
<span id="line982"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Once complete, view the file using the ‘more’ command. You should see something similar to: (lines beginning with # are just comment lines explaining the output)</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line983"></span>
<span id="line984"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part2_task22_03.png">/genomics/images/part2_task22_03.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line985"></span>
<span id="line986"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>You can see the chromosome, position, reference and alternate allele as well as a quality score for the SNP. This is a VCF file (Variant Call File). This is a standard developed for the 1000 genomes project.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line987"></span>
<span id="line988"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>The full specification is given </span><span>&lt;<span class="start-tag">a</span> <span class="attribute-name">href</span>="<a class="attribute-value" href="view-source:http://samtools.github.io/hts-specs/VCFv4.2.pdf">http://samtools.github.io/hts-specs/VCFv4.2.pdf</a>"&gt;</span><span>at http://samtools.github.io/hts-specs/VCFv4.2.pdf </span><span>&lt;/<span class="end-tag">a</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line989"></span>
<span id="line990"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>The lines starting DP and INDEL contain various details concerning the variants. For haploid organisms, most of these details are not necessary.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line991"></span>
<span id="line992"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>This forms our definitive list of variants for this sample.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line993"></span>
<span id="line994"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Take a look at some of the variants we just excluded, was it justified? Remember there is no filter that can keep all the correct variants and remove all the dubious!</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line995"></span>
<span id="line996"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>You can load the VCF file to IGV:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line997"></span>
<span id="line998"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Task 23: Compare the variants found using this method to those you
<span id="line999"></span>found in the manual section</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1000"></span>
<span id="line1001"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Can you see any variants which may have been missed? Often variants within a few bp of indels are filtered out as they could be spurious SNPs thrown up by a poor alignment. This is especially the case if you use non-gapped aligners such as Bowtie.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1002"></span>
<span id="line1003"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Task 24 locating genes which are missing compared to the reference</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1004"></span>
<span id="line1005"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>We can use a command from the BEDTools package (http://bedtools.readthedocs.org/en/latest/) to identify annotated genes which are not covered by reads across their full length.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1006"></span>
<span id="line1007"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Type the following on one line:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1008"></span>
<span id="line1009"></span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">pre</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">code</span>&gt;</span><span>coverageBed -a ~/workshop_materials/genomics_tutorial/data/reference/U00096/U00096.gff -b E_Coli_CGATGT_L001_filtered.sorted.rmdup.bam <span class="entity"><span>&amp;</span>gt;</span> gene_coverage.txt
<span id="line1010"></span></span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">pre</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span>
<span id="line1011"></span>
<span id="line1012"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>This should only take a minute or so. The output contains one row per annotated gene - the last column contains the proportion of the gene that is covered but reads from our sequencing. 1.00 means the gene is 100% covered and 0.00 means no coverage.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1013"></span>
<span id="line1014"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>If we sort by this column we can see which genes are missing  (N.B. -k 13 means sort by the 13th column, replace the 13 by whatever the final column number is in
<span id="line1015"></span>your file):</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1016"></span>
<span id="line1017"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>sort -t$'\t' -g -k 13 gene_coverage.txt  | more</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1018"></span>
<span id="line1019"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>There is another region of about 10kb which is absent from out sequences - can you find it in IGV?</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1020"></span>
<span id="line1021"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part2_task24_01.png">/genomics/images/part2_task24_01.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1022"></span>
<span id="line1023"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Task 25: Determine the effect of variants</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1024"></span>
<span id="line1025"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>So far we have found a number of genes missing from this strain of </span><span>&lt;<span class="start-tag">em</span>&gt;</span><span>E.coli</span><span>&lt;/<span class="end-tag">em</span>&gt;</span><span> which obviously could have a phenotypic effect. Let’s now take a closer look at the variants. We’d like to obtain a list of genes in which these variants occur and whether they result in amino acid changes.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1026"></span>
<span id="line1027"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>To do this we’ll use a custom perl script developed by David Studholme and Konrad Paszkiewicz.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1028"></span>
<span id="line1029"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>We’ll just need the reference annotation, sequence and the VCF file containing the SNPs.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1030"></span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">pre</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">code</span>&gt;</span><span>snp_comparator.pl 10 ~/workshop_materials/genomics_tutorial/data/reference/U00096/U00096.fna ~/workshop_materials/genomics_tutorial/data/reference/U00096/U00096.gff var.called.filt.recode.vcf  <span class="entity"><span>&amp;</span>gt;</span> snp_report.txt
<span id="line1031"></span></span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">pre</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span>
<span id="line1032"></span>
<span id="line1033"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>You will see lots of warnings about ‘Use of uninitialized value $gene_name - you can ignore these.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1034"></span>
<span id="line1035"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>This program takes the information from the reference sequence and annotation, and the VCF SNP files and determines whether the variant occurs within a gene, and if so the effect of each mutation.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1036"></span>
<span id="line1037"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Once complete, view the snp_report.txt file.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1038"></span>
<span id="line1039"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part2_task25_01.png">/genomics/images/part2_task25_01.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1040"></span>
<span id="line1041"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>In later workshops we will see how we can use this program to compare results between different strains.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1042"></span>
<span id="line1043"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>You can also use tools such as SNPEff to evaluate the effect of variants </span><span>&lt;<span class="start-tag">a</span> <span class="attribute-name">href</span>="<a class="attribute-value" href="view-source:http://snpeff.sourceforge.net/index.html">http://snpeff.sourceforge.net/index.html</a>"&gt;</span><span>(http://snpeff.sourceforge.net/index.html) </span><span>&lt;/<span class="end-tag">a</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1044"></span>
<span id="line1045"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Task 26: Check each variant in IGV</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1046"></span>
<span id="line1047"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>**N.B. If a variant doesn’t seem to match what the snp_report file says, check the reverse reading frames.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1048"></span>
<span id="line1049"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>That concludes the first part of the course. You have successfully, QC’d, filtered, remapped and analysed a whole bacterial genome! Well done!</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1050"></span>
<span id="line1051"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>In the next instalment we will be looking at how to extract and assemble
<span id="line1052"></span>unmapped reads. This will enable us to look at material which may be present in the strain of interest but not in the reference sequence.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1053"></span>
<span id="line1054"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/content/genomics2017/images/2017_Workshop_on_Genomics_FINAL-62_1.png">/content/genomics2017/images/2017_Workshop_on_Genomics_FINAL-62_1.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1055"></span>
<span id="line1056"></span>  
<span id="line1057"></span>    </span><span>&lt;<span class="start-tag">h1</span> <span class="attribute-name">class</span>="<a class="attribute-value">page-title</a>"&gt;</span><span>Part 3. Unmapped Read Assembly</span><span>&lt;/<span class="end-tag">h1</span>&gt;</span><span>
<span id="line1058"></span>    </span><span>&lt;<span class="start-tag">h2</span> <span class="attribute-name">id</span>="<a class="attribute-value">background</a>"&gt;</span><span>Background</span><span>&lt;/<span class="end-tag">h2</span>&gt;</span><span>
<span id="line1059"></span>
<span id="line1060"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>In this section of the workshop we will continue the analysis of a strain of </span><span>&lt;<span class="start-tag">em</span>&gt;</span><span>E.coli</span><span>&lt;/<span class="end-tag">em</span>&gt;</span><span>. In the previous  section we cleaned our data, checked QC metrics, mapped our data and obtained a list of variants and an overview of any missing regions.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1061"></span>
<span id="line1062"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Now, we will examine those reads which did not map to the reference genome. We want to know what  these sequences represent. Are they novel genes, plasmids or just contamination?</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1063"></span>
<span id="line1064"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>To do this we will extract unmapped reads, evaluate their quality, prepare them for de novo assembly,  assemble them using SPAdes, generate assembly statistics and then produce some annotation via Pfam, BLAST and RAST.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1065"></span>
<span id="line1066"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Extraction and QC of unmapped reads</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1067"></span>
<span id="line1068"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Task 1: Extract the unmapped reads</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1069"></span>
<span id="line1070"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>First of all make sure you are in the directory</span><span>&lt;<span class="start-tag">br</span> <span>/</span>&gt;</span><span>
<span id="line1071"></span>~/workshop_materials/genomics_tutorial/data/sequencing/ecoli_exeter</span><span>&lt;<span class="start-tag">br</span> <span>/</span>&gt;</span><span>
<span id="line1072"></span>(hint: use the cd command).</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1073"></span>
<span id="line1074"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Then create a directory called unmapped_assembly in which we will do our de novo assembly and analysis.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1075"></span>
<span id="line1076"></span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">pre</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">code</span>&gt;</span><span>mkdir unmapped_assembly
<span id="line1077"></span>cd unmapped_assembly/
<span id="line1078"></span></span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">pre</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span>
<span id="line1079"></span>
<span id="line1080"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Now we will use the bam2fastq program </span><span>&lt;<span class="start-tag">a</span> <span class="attribute-name">href</span>="<a class="attribute-value" href="view-source:http://gsl.hudsonalpha.org/information/software/bam2fastq">http://gsl.hudsonalpha.org/information/software/bam2fastq</a>"&gt;</span><span>(http://gsl.hudsonalpha.org/information/software/bam2fastq) </span><span>&lt;/<span class="end-tag">a</span>&gt;</span><span>to extract from the BAM file just those reads which did NOT map to the reference genome. The bam2fastq program has a number of options, most of which are self-explanatory. Type (all on one line):</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1081"></span>
<span id="line1082"></span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">pre</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">code</span>&gt;</span><span>bam2fastq --no-aligned -o unaligned#.fastq ../remapping_to_reference/E_Coli_CGATGT_L001_filtered.sorted.rmdup.bam
<span id="line1083"></span></span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">pre</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span>
<span id="line1084"></span>
<span id="line1085"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>The –no-aligned option means only extract reads which did not align. The -o unaligned# means dump read 1 into a file called unaligned_1.fastq and read 2 into a file unaligned_2.fastq. Below we can see that the program has successfully created the two files.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1086"></span>
<span id="line1087"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part3_task2_01.png">/genomics/images/part3_task2_01.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1088"></span>
<span id="line1089"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Note that some reads were singletons (i.e. one read mapped to the reference, but the other did not). These will not be included in this analysis.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1090"></span>
<span id="line1091"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Task 2: Check the extraction looks correct</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span>
<span id="line1092"></span>Check that the number of entries in both fastq files is the same. Also check that the last few entries in the read 1 and read 2 files have the same header (i.e. that they have been correctly paired).</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1093"></span>
<span id="line1094"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Task 3: Evaluate QC of unmapped reads</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1095"></span>
<span id="line1096"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Use the fastqc program to look at the statistics and QC for the unaligned_1.fastq and unaligned_2.fastq files.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1097"></span>
<span id="line1098"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Do these look reasonably good? Remember, some reads will fail to map to the reference because they are poor quality, so the average scores will be lower than the initial fastqc report we did in the remapping workshop. The aim here is to see if it looks as though there are reads of reasonable quality which did not map.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1099"></span>
<span id="line1100"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Assuming these reads look ok, we will proceed with preparing them for de novo assembly.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1101"></span>
<span id="line1102"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>De-novo assembly</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1103"></span>
<span id="line1104"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>de-novo</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span>* is a Latin expression meaning “from the beginning,” “afresh,” “anew,” “beginning again.” when we perform a de novo assembly we try to reconstruct a genome or part of the genome from our reads without making any prior assumptions (in contrast to remapping where we compare out reads to what we think is a close reference sequence).</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1105"></span>
<span id="line1106"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>The advantage is that is that de novo assembly can reveal completely novel results, identify horizontal gene transfer events for example. The disadvantage is that it is difficult to get a good assembly from short reads and it can be prone to misleading results due to contamination and mis-assembly.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1107"></span>
<span id="line1108"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Task 4: Learn more about de novo assemblers</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1109"></span>
<span id="line1110"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>To understand more about de-novo assemblers, read the technical note at:
<span id="line1111"></span></span><span>&lt;<span class="start-tag">a</span> <span class="attribute-name">href</span>="<a class="attribute-value" href="view-source:https://www.illumina.com/Documents/products/technotes/technote_denovo_assembly_ecoli.pdf">https://www.illumina.com/Documents/products/technotes/technote_denovo_assembly_ecoli.pdf</a>"&gt;</span><span>https://www.illumina.com/Documents/products/technotes/technote_denovo_assembly_ecoli.pdf</span><span>&lt;/<span class="end-tag">a</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1112"></span>
<span id="line1113"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>N.B.</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span> You will  also learn more in the next section so don’t worry if it doesn’t all make sense immediately. You should however understand the idea of the k-mer and broadly how the assembly is built up from them.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1114"></span>
<span id="line1115"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Task 5: Generate the assembly.</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1116"></span>
<span id="line1117"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>We will be using an assembler called SPAdes </span><span>&lt;<span class="start-tag">a</span> <span class="attribute-name">href</span>="<a class="attribute-value" href="view-source:http://cab.spbu.ru/software/spades/">http://cab.spbu.ru/software/spades/</a>"&gt;</span><span>(http://cab.spbu.ru/software/spades/)</span><span>&lt;/<span class="end-tag">a</span>&gt;</span><span>. It generally performs pretty well with a variety of genomes. It can also incorporate longer reads produced from PacBio sequencers that we will use later in the course.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1118"></span>
<span id="line1119"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>One big advantage is that it is not just a pure assembler - it is a suite of programs that prepare the reads you have, assembles them and then refines the assembly.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1120"></span>
<span id="line1121"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>SPAdes runs the modules that are required for a particular dataset and it produces the assembly with a minimum of preparation and parameter selection - making it very straightforward to produce a decent assembly. As with everything in bio-informatics you should
<span id="line1122"></span>try to assess the results critically and understand the implications for further analysis.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1123"></span>
<span id="line1124"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Let’s start the assembler because it takes a few minutes to run.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1125"></span>
<span id="line1126"></span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">pre</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">code</span>&gt;</span><span>spades.py -k 21,33,55,77,99,127 --careful -o spades_assembly -1 unaligned_1.fastq -2 unaligned_2.fastq
<span id="line1127"></span></span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">pre</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span>
<span id="line1128"></span>
<span id="line1129"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part3_task5_01.png">/genomics/images/part3_task5_01.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1130"></span>
<span id="line1131"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>We are telling it to run the SPAdes assembly pipeline with a range of k-mer sizes (-k); specifying –careful tells it to run a mismatch correction algorithm to reduce the number of errors; put the output in the spades_assembly directory and the reads to assemble.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1132"></span>
<span id="line1133"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Just because SPAdes does a lot for you does not mean you should not try to understand the process.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1134"></span>
<span id="line1135"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Have a read of this:
<span id="line1136"></span></span><span>&lt;<span class="start-tag">a</span> <span class="attribute-name">href</span>="<a class="attribute-value" href="view-source:http://thegenomefactory.blogspot.co.uk/2013/08/how-spades-differs-from-velvet.html">http://thegenomefactory.blogspot.co.uk/2013/08/how-spades-differs-from-velvet.html</a>"&gt;</span><span>http://thegenomefactory.blogspot.co.uk/2013/08/how-spades-differs-from-velvet.html </span><span>&lt;/<span class="end-tag">a</span>&gt;</span><span>
<span id="line1137"></span>It is a discussion of how SPAdes differs from Velvet another widely used assembler, it explains the overall process nicely:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1138"></span>
<span id="line1139"></span></span><span>&lt;<span class="start-tag">ul</span>&gt;</span><span>
<span id="line1140"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>Read error correction based on k-mer frequencies using</span><span>&lt;<span class="start-tag">a</span> <span class="attribute-name">href</span>="<a class="attribute-value" href="view-source:http://genomebiology.com/1471-2164/14/S1/S7">http://genomebiology.com/1471-2164/14/S1/S7</a>"&gt;</span><span> BayesHammer </span><span>&lt;/<span class="end-tag">a</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line1141"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>De Bruijn graph assembly at </span><span>&lt;<span class="start-tag">em</span>&gt;</span><span>multiple</span><span>&lt;/<span class="end-tag">em</span>&gt;</span><span> k-mer sizes, not just a single fixed one.</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line1142"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>Merging of different k-mer assemblies (good for varying coverage)</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line1143"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>Scaffolding of contigs from paired end/mate pair reads</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line1144"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>Repeat resolution from paired end/mate pair data using </span><span>&lt;<span class="start-tag">a</span> <span class="attribute-name">href</span>="<a class="attribute-value" href="view-source:http://bioinf.spbau.ru/rectangles">http://bioinf.spbau.ru/rectangles</a>"&gt;</span><span>rectangle graphs</span><span>&lt;/<span class="end-tag">a</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line1145"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>Contig error correction based on aligning the original reads with </span><span>&lt;<span class="start-tag">a</span> <span class="attribute-name">href</span>="<a class="attribute-value" href="view-source:http://bio-bwa.sourceforge.net/">http://bio-bwa.sourceforge.net/</a>"&gt;</span><span>BWA</span><span>&lt;/<span class="end-tag">a</span>&gt;</span><span> back to contigs.</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line1146"></span></span><span>&lt;/<span class="end-tag">ul</span>&gt;</span><span>
<span id="line1147"></span>
<span id="line1148"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Try to understand the steps in the context of the whole picture:</span><span>&lt;<span class="start-tag">br</span> <span>/</span>&gt;</span><span>
<span id="line1149"></span>Can you explain why depth of coverage and error correction of reads becomes more important as k-mer length increases?</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1150"></span>
<span id="line1151"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>When the assembly is complete:
<span id="line1152"></span>Change to the spades_assembly directory (use cd) and look at the output.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1153"></span>
<span id="line1154"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part3_task5_02.png">/genomics/images/part3_task5_02.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1155"></span>
<span id="line1156"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Let’s take a look at some of the more important content.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1157"></span></span><span>&lt;<span class="start-tag">ul</span>&gt;</span><span>
<span id="line1158"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>params.txt</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span>  - a summary of the parameters used for assembly it is useful so
<span id="line1159"></span>you can repeat the exact analysis performed, or can remember you setting when you want to
<span id="line1160"></span>publish the genome.</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line1161"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>contigs.fasta</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span> - the final results of the assembly in fasta format.</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line1162"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>scaffolds.fasta</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span> - the final results after scaffolding (which means using paired end information to join contigs together with gaps). In this case the files are identical, probably because the sum of the lengths of our paired reads is not much smaller than our insert size (there are very few large gaps between reads).</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line1163"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>assembly_graph.fastg</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span> SPAdes assembly graph in FASTG format - this is a slightly different format that contains more information than fasta - for example it can contain alternative alleles in diploid assemblies. We don’t need it here, but see </span><span>&lt;<span class="start-tag">a</span> <span class="attribute-name">href</span>="<a class="attribute-value" href="view-source:http://fastg.sourceforge.net/FASTG_Spec_v1.00.pdf">http://fastg.sourceforge.net/FASTG_Spec_v1.00.pdf</a>"&gt;</span><span>http://fastg.sourceforge.net/FASTG_Spec_v1.00.pdf</span><span>&lt;/<span class="end-tag">a</span>&gt;</span><span> if you might be working with diploid organisms. You can use the Bandage </span><span>&lt;<span class="start-tag">a</span> <span class="attribute-name">href</span>="<a class="attribute-value" href="view-source:http://rrwick.github.io/Bandage/">http://rrwick.github.io/Bandage/</a>"&gt;</span><span>(http://rrwick.github.io/Bandage/)</span><span>&lt;/<span class="end-tag">a</span>&gt;</span><span> to view this file.</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line1164"></span></span><span>&lt;/<span class="end-tag">ul</span>&gt;</span><span>
<span id="line1165"></span>
<span id="line1166"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Task 6: Assessment of the assembly</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1167"></span>
<span id="line1168"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>We will use QUAST </span><span>&lt;<span class="start-tag">a</span> <span class="attribute-name">href</span>="<a class="attribute-value" href="view-source:http://quast.sourceforge.net/quast">http://quast.sourceforge.net/quast</a>"&gt;</span><span>(http://quast.sourceforge.net/quast)</span><span>&lt;/<span class="end-tag">a</span>&gt;</span><span> to generate some statistics on the assembly (in the spades_assembly directory)</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1169"></span>
<span id="line1170"></span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">pre</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">code</span>&gt;</span><span>cd spades_assembly
<span id="line1171"></span>quast.py --output-dir quast contigs.fasta
<span id="line1172"></span></span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">pre</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span>
<span id="line1173"></span>
<span id="line1174"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>This will create a directory called quast and create some statistics on the assembly you produced.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1175"></span>
<span id="line1176"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>cat quast/report.txt</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1177"></span>
<span id="line1178"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part3_task6_01.png">/genomics/images/part3_task6_01.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1179"></span>
<span id="line1180"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Try to interpret the information in the light of what we were trying to do. Because we are assembling unaligned reads we are not expecting a whole chromosome to pop out. We are expecting bits of our strain that does not exist in the reference we aligned against; possibly some contamination; various small contigs made up of reads that didn’t quite align to our reference.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1181"></span>
<span id="line1182"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>The N50 and L50 measures are very important in a normal assembly and we will visit them later, they are not really relevant to this assembly.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1183"></span>
<span id="line1184"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>You will notice that we have 1 contig 30-60kb long - what do you think this might be? And 12 other contigs longer than 1kb. We should find out what this is.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1185"></span>
<span id="line1186"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Now that we have assembled the reads and have a feel for how much (or in this case, how little) data we have, we can set about analysing it. By analysing, we mean identifying which genes are present, which organism they are from and whether they form part of the main
<span id="line1187"></span>chromosome or are an independent unit (e.g. plasmid).</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1188"></span>
<span id="line1189"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>We are going to take a 3-prong approach. The first will simply search the nucleotide sequences of the contigs against the NCBI non-redundant database. This will enable us to identify the species to which a given contig matches best (or most closely). The second will call open reading frames within the contigs and search those against the Swissprot database of manually curated (i.e. high quality) annotated protein sequences. Finally, we will search the open reading frames against the Pfam database of protein families </span><span>&lt;<span class="start-tag">a</span> <span class="attribute-name">href</span>="<a class="attribute-value" href="view-source:http://pfam.xfam.org/">http://pfam.xfam.org/</a>"&gt;</span><span>(http://pfam.xfam.org/).</span><span>&lt;/<span class="end-tag">a</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1190"></span>
<span id="line1191"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Why not just search the NCBI blast database? Well, remember nearly all of our biological knowledge is based on homology - if two proteins are similar they probably share an evolutionary history and may thus share functional characteristics. Metrics to define whether two sequences are homologous are notoriously difficult to define accurately. If two sequences share 90% sequence identity over their length, you can be pretty sure they are homologous. If they share 2% they probably aren’t. But what if they share 30%? This is the notorious twilight zone of 20-30% sequence identity where it is very difficult to judge whether two proteins are homologous based on sequence alone.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1192"></span>
<span id="line1193"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>To help
<span id="line1194"></span>overcome this searching more subtle signatures may help - this is where Pfam comes in. Pfam is a database which contains protein families identified by particular signatures or patterns in
<span id="line1195"></span>their protein sequence. These signatures are modeled by Hidden Markov Models (HMMs) and used to
<span id="line1196"></span>search query sequences. These can provide a high level annotation where BLAST might otherwise
<span id="line1197"></span>fail. It also has the advantage of being much faster than BLAST.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1198"></span>
<span id="line1199"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Task 7: Search contigs against NCBI non-redundant database</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1200"></span>
<span id="line1201"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Firstly we can filter out low coverage and very short contigs using a perl script:
<span id="line1202"></span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>filter_low_coverage_contigs.pl <span class="entity"><span>&amp;</span>lt;</span> contigs.fasta <span class="entity"><span>&amp;</span>gt;</span> contigs.goodcov.fasta</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1203"></span>
<span id="line1204"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>The following command executes a nucleotide BLAST search (blastn) of the sequences in the contigs.fa file against the non-redundant database. As this takes a long time to run the results have been precomputed and are available in</span><span>&lt;<span class="start-tag">br</span> <span>/</span>&gt;</span><span>
<span id="line1205"></span>~/workshop_materials/genomics_tutorial/data/sequencing/ecoli_exeter/blast_precompute/unmapped_reads/</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1206"></span>
<span id="line1207"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">code</span>&gt;</span><span>
<span id="line1208"></span>blastn -db ~/workshop_materials/genomics_tutorial/db/blast/nt -query contigs.goodcov.fasta -evalue 1e-06 -num_threads 8 -show_gis -num_alignments 10 -num_descriptions 10 -out contigs.fasta.blastn
<span id="line1209"></span></span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1210"></span>
<span id="line1211"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>There are a lot of options in this command, let’s go through them</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1212"></span></span><span>&lt;<span class="start-tag">ul</span>&gt;</span><span>
<span id="line1213"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>-db</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span> is the prepared blast database to search</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line1214"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>
<span id="line1215"></span>    </span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>evalue</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span> apply an e-value (expectation value) </span><span>&lt;<span class="start-tag">a</span> <span class="attribute-name">href</span>="<a class="attribute-value" href="view-source:http://www.ncbi.nlm.nih.gov/BLAST/tutorial/Altschul-1.html">http://www.ncbi.nlm.nih.gov/BLAST/tutorial/Altschul-1.html</a>"&gt;</span><span>(http://www.ncbi.nlm.nih.gov/BLAST/tutorial/Altschul-1.html)</span><span>&lt;/<span class="end-tag">a</span>&gt;</span><span> cutoff of 1e-06 to limit ourselves to statistically significant hits (i.e. in this case 1 in 1 million likelihood of a hit to a database of this size by a sequence of this length).</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1216"></span>  </span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line1217"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>-num_alignments</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span> and </span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>-num_descriptions</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span> flags tell blastn to only display the top 10 results  for each hit.</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line1218"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>-</span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>num_threads</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span> that it should use 8 CPU cores \</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line1219"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>-show_gis</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span> that it should include general identifier (GI) numbers in the output.</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line1220"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>-out</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span> file in which to place the output.</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line1221"></span></span><span>&lt;/<span class="end-tag">ul</span>&gt;</span><span>
<span id="line1222"></span>
<span id="line1223"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>There is lots of information on running blast from the command line at </span><span>&lt;<span class="start-tag">a</span> <span class="attribute-name">href</span>="<a class="attribute-value" href="view-source:http://www.ncbi.nlm.nih.gov/books/NBK1763/">http://www.ncbi.nlm.nih.gov/books/NBK1763/</a>"&gt;</span><span>http://www.ncbi.nlm.nih.gov/books/NBK1763/ </span><span>&lt;/<span class="end-tag">a</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1224"></span>
<span id="line1225"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">em</span>&gt;</span><span>N.B. GI (GeneInfo Identifiers) are being phased out by NCBI so future versions of Blast and NCBI databases will not support the -show_gis option and may break some down-stream tools such as KronaTools and other databases.</span><span>&lt;/<span class="end-tag">em</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1226"></span>
<span id="line1227"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Open the results file
<span id="line1228"></span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>pluma contigs.fasta.blastn</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1229"></span>
<span id="line1230"></span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">pre</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">code</span>&gt;</span><span>BLASTN 2.2.31+
<span id="line1231"></span>
<span id="line1232"></span>
<span id="line1233"></span>Reference: Zheng Zhang, Scott Schwartz, Lukas Wagner, and Webb
<span id="line1234"></span>Miller (2000), "A greedy algorithm for aligning DNA sequences", J
<span id="line1235"></span>Comput Biol 2000; 7(1-2):203-14.
<span id="line1236"></span>
<span id="line1237"></span>
<span id="line1238"></span>
<span id="line1239"></span>Database: Nucleotide collection (nt)
<span id="line1240"></span>           29,442,065 sequences; 84,823,117,434 total letters
<span id="line1241"></span>
<span id="line1242"></span>
<span id="line1243"></span>
<span id="line1244"></span>Query= NODE_1_length_67364_cov_602.091
<span id="line1245"></span>
<span id="line1246"></span>Length=67364
<span id="line1247"></span>                                                                      Score     E
<span id="line1248"></span>Sequences producing significant alignments:                          (Bits)  Value
<span id="line1249"></span>
<span id="line1250"></span>gi|664682453|gb|CP008801.1|  Escherichia coli KLY, complete genome    86309   0.0  
<span id="line1251"></span>gi|8918823|dbj|AP001918.1|  Escherichia coli K-12 plasmid F DNA, ...  86272   0.0  
<span id="line1252"></span>gi|619497957|gb|KJ170699.1|  Escherichia coli strain K-12 plasmid...  65330   0.0  
<span id="line1253"></span>gi|665821556|gb|KJ484626.1|  Escherichia coli plasmid pH2332-166,...  65302   0.0  
<span id="line1254"></span>gi|665821958|gb|KJ484628.1|  Escherichia coli plasmid pH2291-144,...  65213   0.0  
<span id="line1255"></span>gi|28629230|gb|AF550679.1|  Escherichia coli plasmid p1658/97, co...  64591   0.0  
<span id="line1256"></span>gi|4874241|gb|U01159.2|  Escherichia coli F sex factor transfer r...  61474   0.0  
<span id="line1257"></span>gi|665822931|gb|KJ484636.1|  Escherichia coli plasmid pC59-153, c...  41227   0.0  
<span id="line1258"></span>gi|301130432|gb|CP002090.1|  Salmonella enterica subsp. enterica ...  41026   0.0  
<span id="line1259"></span>gi|301130304|gb|CP002089.1|  Salmonella enterica subsp. enterica ...  41026   0.0  
<span id="line1260"></span>
<span id="line1261"></span>. . .
<span id="line1262"></span></span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">pre</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span>
<span id="line1263"></span>
<span id="line1264"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>SPAdes names the contigs by increasing size, so ‘NODE_1’ is the longest contig.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1265"></span>
<span id="line1266"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>There are a number of good hits; notice from the contig header line that the average coverage is <span class="entity"><span>&amp;</span>gt;</span>500 and the coverage of our genome was around 50 - does this give you a clue to what it is?</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1267"></span>
<span id="line1268"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Task 8: Obtain open reading frames</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1269"></span>
<span id="line1270"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>The first task is to call open reading frames within the contigs. These are designated by canonical start and stop codons and are usually identified by searching for regions free of stop codons. We will use the EMBOSS package program getorf to call these.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1271"></span>
<span id="line1272"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>We will use codon table 11 which defines the bacterial codon usage table </span><span>&lt;<span class="start-tag">a</span> <span class="attribute-name">href</span>="<a class="attribute-value" href="view-source:http://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi">http://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi</a>"&gt;</span><span>(http://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi)</span><span>&lt;/<span class="end-tag">a</span>&gt;</span><span> and state that the sequences we are dealing with are not circular (only the 67k contig is long enough!). We will also restrict the ORFs to just those sequences longer than 300 nucleotides (i.e. 100 amino acids). We will store the results in file contigs.orf.fa.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1273"></span>
<span id="line1274"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>getorf -table 11  -circular N -minsize 300 -sequence contigs.goodcov.fasta -outseq contigs.orf.fasta</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1275"></span>
<span id="line1276"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>If we look at the output file we can see that it is a FASTA formatted file containing the name of the contig on which the ORF occurs, followed by an underscore and a number (e.g. _1) to indicate the number of the ORF on that contig. The numbers in
<span id="line1277"></span>square brackets indicate the start and end position of the ORF on the contig (i.e. in nucleotide space). So the first ORF occurs on NODE 9 and is between position 934 and 1494. The third ORF occurs between positions 2400 and 2047 on the reverse strand. This is a relatively short peptide sequence and is unlikely to be a genuine peptide.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1278"></span>
<span id="line1279"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Also note that many ORFs do not start with a Methionine. This is because by default the getorf program calls ORFs between stop codons rather than start and stop codons. Primarily this is to avoid spurious ORFs due to Met residues within a protein sequence and to ensure untranslated regions are captured.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1280"></span>
<span id="line1281"></span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">pre</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">code</span>&gt;</span><span><span class="entity"><span>&amp;</span>gt;</span>NODE_9_length_3631_cov_29.6618_ID_17_1 [934 - 1494]  \
<span id="line1282"></span>TERFEVSEINSQALREAAEQAMHDDWGFDADLFHELVTPSIVLELLDERERNQQYIKRRD
<span id="line1283"></span>QENEDIALTVGKLRVELETAKSKLNEQREYYEGVISDGSKRIAKLESNEVREDGNQFLVV
<span id="line1284"></span>RHPGKTPVIKHCTGDLEEFLRQLIEQDPLVTIDIITHRYYGVGGQWVQDAGEYLHMMSDA
<span id="line1285"></span>GIRIKGE
<span id="line1286"></span><span class="entity"><span>&amp;</span>gt;</span>NODE_9_length_3631_cov_29.6618_ID_17_2 [2450 - 3529]  \
<span id="line1287"></span>RGSEMGRRRSHERRDLPPNLYIRNNGYYCYRDPRTGKEFGLGRDRRIAITEAIQANIELF
<span id="line1288"></span>SGHKHKPLTARINSDNSVTLHSWLDRYEKILASRGIKQKTLINYMSKIKAIRRGLPDAPL
<span id="line1289"></span>EDITTKEIAAMLNGYIDEGKAASAKLIRSTLSDAFREAIAEGHITTNHVAATRAAKSEVR
<span id="line1290"></span>RSRLTADEYLKIYQAAESSPCWLRLAMELAVVTGQRVGDLCEMKWSDIVDGYLYVEQSKT
<span id="line1291"></span>GVKIAIPTALHIDALGISMKETLDKCKEILGGETIIASTRREPLSSGTVSRYFMRARKAS
<span id="line1292"></span>GLSFEGDPPTFHELRSLSARLYEKQISDKFAQHLLGHKSDTMASQYRDDRGREWDKIEIK
<span id="line1293"></span><span class="entity"><span>&amp;</span>gt;</span>NODE_9_length_3631_cov_29.6618_ID_17_3 [2400 - 2047] (REVERSE
<span id="line1294"></span>SENSE)  
<span id="line1295"></span>FVEQILSSILNRRWEYPAFPNPSTNCFKASWTSLACVPLLKCQVHRKVSAITRKKKPPSG
<span id="line1296"></span>GLVFFQFFNSNIGYVCMCYLRPYHPVVVAVVDVLRFDNSVEWLSIPFSCDSEVHLSSP
<span id="line1297"></span></span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">pre</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span>
<span id="line1298"></span>
<span id="line1299"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Task 9: Search open reading frames against NCBI non-redundant database</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1300"></span>
<span id="line1301"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>The first thing we can do with these open reading frames is to search them against the NCBI non-redundant database of protein sequences to see what they may match.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1302"></span>
<span id="line1303"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>First reduce the number of orfs so that we have a manageable number - this small perl program
<span id="line1304"></span>selects 10% of the orfs.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1305"></span>
<span id="line1306"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>reduce_fasta_10x.pl <span class="entity"><span>&amp;</span>lt;</span> contigs.orf.fasta <span class="entity"><span>&amp;</span>gt;</span> contigs.orf.small.fasta</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1307"></span>
<span id="line1308"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Then you would type (all on one line). HOWEVER this could take a while, therefore the results have been precomputed in </span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>~/workshop_materials/genomics_tutorial/data/sequencing/ecoli_exeter/blast_precompute/unmapped_reads/</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1309"></span>
<span id="line1310"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>blastp -db ~/workshop_materials/genomics_tutorial/db/blast/nr -query contigs.orf.small.fasta -evalue 1e-06 -num_threads 2 -show_gis -num_alignments 10 -num_descriptions 10 -out contigs.orf.blastp</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1311"></span>
<span id="line1312"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Task 10: Review the BLAST format</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1313"></span>
<span id="line1314"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Open the results file with pluma and search for plasmid in the text. You should find a number of hits to plasmid related proteins - one example is below - can you find any others? (Remember we only checked 10% of the orfs we found). This evidence is not conclusive, but combined with the high coverage over, it is starting to look like this contig is a plasmid.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1315"></span>
<span id="line1316"></span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">pre</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">code</span>&gt;</span><span>Query= NODE_1_length_67364_cov_602.091_17 [9087 - 9569]
<span id="line1317"></span>
<span id="line1318"></span>Length=161
<span id="line1319"></span>                                                                      Score     E
<span id="line1320"></span>Sequences producing significant alignments:                          (Bits)  Value
<span id="line1321"></span>
<span id="line1322"></span>gi|553349257|gb|ESA76469.1|  Protein PsiB                             322     5e-110
<span id="line1323"></span>gi|503044320|ref|WP_013279296.1|  plasmid SOS inhibition protein B    320     2e-109
<span id="line1324"></span>gi|324006232|gb|EGB75451.1|  plasmid SOS inhibition protein (PsiB)    310     2e-105
<span id="line1325"></span>gi|300355538|gb|EFJ71408.1|  plasmid SOS inhibition protein (PsiB)    308     5e-105
<span id="line1326"></span>gi|324010272|gb|EGB79491.1|  plasmid SOS inhibition protein (PsiB)    305     2e-103
<span id="line1327"></span>gi|505582716|ref|WP_015675404.1|  recombinase                         300     1e-101
<span id="line1328"></span>gi|446768672|ref|WP_000845928.1|  recombinase                         299     2e-101
<span id="line1329"></span>gi|301073353|gb|EFK88159.1|  Protein PsiB                             300     2e-101
<span id="line1330"></span>gi|446768671|ref|WP_000845927.1|  recombinase                         297     1e-100
<span id="line1331"></span>gi|446610348|ref|WP_000687694.1|  recombinase                         297     1e-100
<span id="line1332"></span></span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">pre</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span>
<span id="line1333"></span>
<span id="line1334"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Task 11: Check that the contigs do not appear in the reference sequence</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1335"></span>
<span id="line1336"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>In theory, the unmapped reads used to generate the contigs should not assemble into something which will map against the genome. However, it is always possible (especially with more complex genomes), that this might happen. To double-check move back to the folder containing the contigs.goodcov.fasta:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1337"></span>
<span id="line1338"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>blastn -subject ~/workshop_materials/genomics_tutorial/data/reference/U00096/U00096.fna -query contigs.goodcov.fasta | more</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1339"></span>
<span id="line1340"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Here we use the BLAST+ package in a different mode to compare two sequences against each other.
<span id="line1341"></span>Unlike the previous examples where we have searched against a database of sequences, here we are doing a simple search of the contigs against the reference genome we are using. Scroll down a little…</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1342"></span>
<span id="line1343"></span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">pre</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">code</span>&gt;</span><span>Query= NODE_18_length_917_cov_10.7608
<span id="line1344"></span>
<span id="line1345"></span>Length=917
<span id="line1346"></span>
<span id="line1347"></span>Subject= U00096.3
<span id="line1348"></span>
<span id="line1349"></span>Length=4641652
<span id="line1350"></span>
<span id="line1351"></span>
<span id="line1352"></span> Score = 193 bits (104),  Expect = 3e-49
<span id="line1353"></span> Identities = 186/227 (82%), Gaps = 0/227 (0%)
<span id="line1354"></span> Strand=Plus/Minus
<span id="line1355"></span>
<span id="line1356"></span>Query  624      GCCTTTGGCGTTGCAGCAAGCGTTTCAGACGTGCTGTTGGTTGCACTGCTGAGCTGCACT  683
<span id="line1357"></span>                |||||||||||||| || |||||||||||||| |||||||| |||||||||||||| |||
<span id="line1358"></span>Sbjct  1430511  GCCTTTGGCGTTGCCGCCAGCGTTTCAGACGTACTGTTGGTCGCACTGCTGAGCTGTACT  1430452
<span id="line1359"></span>
<span id="line1360"></span>Query  684      ATCCCCTTTCTCGTTGTGTCCGCATCCTCAAGCGCGACAGCTGAAGCTATATCTTCTGCA  743
<span id="line1361"></span>                ||||||||| |||| ||    |||||||||||||| || || || || ||||| ||||| 
<span id="line1362"></span>Sbjct  1430451  ATCCCCTTTTTCGTCGTACTTGCATCCTCAAGCGCCACGGCGGATGCAATATCCTCTGCC  1430392
<span id="line1363"></span>
<span id="line1364"></span></span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">pre</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span>
<span id="line1365"></span>
<span id="line1366"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>You can see that some of the contigs that have been assembled hit the reference sequence. In the record above the evalue is 3e-49 which is massively significant; however, the evalue is calculated as the chance of a hit this close against a random sequence of the same size. Since our subject sequence is now very small and we know it is related to our strain it is not surprising that there are some hits. We are concerned about whole contigs that map closely to
<span id="line1367"></span>the reference genome.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1368"></span>
<span id="line1369"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Task 12: Run open reading frames through pfam_scan</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1370"></span>
<span id="line1371"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Pfam is a database of protein families. They are grouped together using a number of criteria based on their function. For more information read </span><span>&lt;<span class="start-tag">a</span> <span class="attribute-name">href</span>="<a class="attribute-value" href="view-source:http://en.wikipedia.org/wiki/Pfam">http://en.wikipedia.org/wiki/Pfam</a>"&gt;</span><span>http://en.wikipedia.org/wiki/Pfam.</span><span>&lt;/<span class="end-tag">a</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1372"></span>
<span id="line1373"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Pfam is grouped into several databases depending on the level of curation. Pfam-A is high-quality manual curation and consists of around 12,500 families. Pfam-B is full of automated predictions which may be informative but should not be relied upon without additional evidence. Pfam will also search for signatures of active-sites if you specify the correct flag.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1374"></span>
<span id="line1375"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Here we want to search the Pfam database of Hidden Markov Models to see which protein families are contained within this contig. You’ll notice that this runs considerably faster than BLAST. Here we search using the contigs.orf.fa file against the Pfam databases ~/genomics_tutorial/db/pfam/ and output the results to contigs.orf.pfam. We’ll use 2 CPU cores for the search and state that we want to search PfamB entries as well as active site residues.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1376"></span>
<span id="line1377"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>This step might take about 30 minutes. So you can get a coffee in the meantime.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1378"></span>
<span id="line1379"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>pfam_scan.pl -fasta contigs.orf.fasta -dir ~/workshop_materials/genomics_tutorial/db/pfam/ -outfile contigs.orf.pfam -cpu 2 -pfamB -as</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1380"></span>
<span id="line1381"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>View the output using pluma:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1382"></span>
<span id="line1383"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Search for NODE_9 (for example).</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1384"></span>
<span id="line1385"></span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">pre</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">code</span>&gt;</span><span>NODE_9_length_3631_cov_31.6458_1         7    106      7    106 PF13935.1   Ead_Ea22          Family     1   139   139    103.8     9e-30   1 No_clan  
<span id="line1386"></span>NODE_9_length_3631_cov_31.6458_1        77    113     77    126 PB009353    Pfam-B_9353       Pfam-B     1    37    82     53.4   2.8e-14  NA NA      
<span id="line1387"></span>NODE_9_length_3631_cov_31.6458_2         5     74      5     75 PF09003.5   Phage_integ_N     Domain     1    75    76     89.5     8e-26   1 CL0081   
<span id="line1388"></span>NODE_9_length_3631_cov_31.6458_2        85    162     84    162 PF02899.12  Phage_int_SAM_1   Domain     2    84    84     24.1     3e-05   1 CL0469   
<span id="line1389"></span>NODE_9_length_3631_cov_31.6458_2       183    349    182    353 PF00589.17  Phage_integrase   Family     2   169   173    115.1   2.6e-33   1 CL0382   predicted_active_site[239,312,216,346,337,315]
<span id="line1390"></span>NODE_9_length_3631_cov_31.6458_3        53    115     48    117 PB009641    Pfam-B_9641       Pfam-B    50   112   168     23.3   6.7e-05  NA NA      
<span id="line1391"></span></span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">pre</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span>
<span id="line1392"></span>
<span id="line1393"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>The 8th column shows the type of entry that was hit in the pfam database.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1394"></span>
<span id="line1395"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Go to </span><span>&lt;<span class="start-tag">a</span> <span class="attribute-name">href</span>="<a class="attribute-value" href="view-source:http://pfam.xfam.org/">http://pfam.xfam.org/</a>"&gt;</span><span>http://pfam.xfam.org/</span><span>&lt;/<span class="end-tag">a</span>&gt;</span><span> and enter the Pfam domain
<span id="line1396"></span>in the search box.</span><span>&lt;<span class="start-tag">br</span> <span>/</span>&gt;</span><span>
<span id="line1397"></span>Let’s take a look at Pfam domain Phage_integ_N</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1398"></span>
<span id="line1399"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part3_task12_02.png">/genomics/images/part3_task12_02.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1400"></span>
<span id="line1401"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>There are a lot of hits to phage domains and domains that manipulate DNA. You might expect this as these sequences have presumably been incorporated into our strain since it diverged from the reference.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1402"></span>
<span id="line1403"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Also look at domains (the most specific type of hit) from our large contig NODE_1_….. is there any evidence for it being a plasmid?</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1404"></span>
<span id="line1405"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>The Pfam-B matches do not tell you much that is useful.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1406"></span>
<span id="line1407"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Examine one or two more domains from your results file - is there anything interesting?</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1408"></span>
<span id="line1409"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Analysing the results in RAST</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1410"></span>
<span id="line1411"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>By now you should be able to see that analysing results for de novo assembled reads of any sort can be difficult and time-consuming. Bear in mind that we have only been faced with a single contig of 3kb. Quite often you may find yourself dealing with hundreds, if not thousands of contigs. Some will be a
<span id="line1412"></span>few 100kb long. Others may only be 200-300bp. How should we go about analysing these in a more efficient manner? There are a number of options here.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1413"></span>
<span id="line1414"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>For eukaryotes I would suggest looking at MAKER </span><span>&lt;<span class="start-tag">a</span> <span class="attribute-name">href</span>="<a class="attribute-value" href="view-source:http://www.yandell-lab.org/software/maker.html">http://www.yandell-lab.org/software/maker.html</a>"&gt;</span><span>(http://www.yandell-lab.org/software/maker.html)</span><span>&lt;/<span class="end-tag">a</span>&gt;</span><span>.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1415"></span>
<span id="line1416"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>For prokaryotes the situation is somewhat easier and we can use a web-based service known as RAST. This is not the only service (Xbase is another), but it is one of the most commonly used.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1417"></span>
<span id="line1418"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>RAST is a website where you upload the results of your de novo assembly and RAST will attempt to provide annotation in commonly used GFF and Genbank formats. This can be used to load up the annotation in Artemis or Apollo. Alternatively RAST has its own in-built viewer.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1419"></span>
<span id="line1420"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Task 13 (Optional) Log in to RAST</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1421"></span>
<span id="line1422"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Within your instance, go to </span><span>&lt;<span class="start-tag">a</span> <span class="attribute-name">href</span>="<a class="attribute-value" href="view-source:http://rast.nmpdr.org/">http://rast.nmpdr.org/</a>"&gt;</span><span>http://rast.nmpdr.org/</span><span>&lt;/<span class="end-tag">a</span>&gt;</span><span>
<span id="line1423"></span>Log-in with the details RAST provided to you before you started this series of workshops.
<span id="line1424"></span>If you do not have one, you may need to wait several days for your login to be issued
<span id="line1425"></span>by RAST. Please skip ahead and come back to this section.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1426"></span>
<span id="line1427"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part3_task13_01.png">/genomics/images/part3_task13_01.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1428"></span>
<span id="line1429"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Task 14 (Optional) Upload the assembled contigs and annotate using RAST</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span>
<span id="line1430"></span>Click on Your jobs-<span class="entity"><span>&amp;</span>gt;</span>Upload New Job</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1431"></span>
<span id="line1432"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part3_task14_01.png">/genomics/images/part3_task14_01.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1433"></span>
<span id="line1434"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Upload the contigs.fasta file obtained by the de novo assembly of unmapped reads. Click on ‘Use this data’ and go to step 2.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1435"></span>
<span id="line1436"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part3_task14_02.png">/genomics/images/part3_task14_02.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1437"></span>
<span id="line1438"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>We know this is an </span><span>&lt;<span class="start-tag">em</span>&gt;</span><span>E.coli</span><span>&lt;/<span class="end-tag">em</span>&gt;</span><span> genome so we can enter 562 as the Taxonomy ID and click on ‘Fill  in form based on NCBI taxonomy-ID’. If you’re dealing with a different organism, be sure to change this number. RAST will automatically split any scaffolds (i.e. contigs with bits missing in the middle - denoted by Ns). Then click ‘Use this data and go to step 3’.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1439"></span>
<span id="line1440"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part3_task14_03.png">/genomics/images/part3_task14_03.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1441"></span>
<span id="line1442"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Replicate the settings above and click on ‘Finish the upload’.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1443"></span>
<span id="line1444"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Your job may take several hours to run. In the meantime, proceed to the next workshop and
<span id="line1445"></span>come back to this later.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1446"></span>
<span id="line1447"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part3_task14_04.png">/genomics/images/part3_task14_04.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1448"></span>
<span id="line1449"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Once complete, RAST should email you a message. You can then view the results or download them
<span id="line1450"></span>in standardized formats (e.g. GFF3, Genbank, EMBL etc).</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1451"></span>
<span id="line1452"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part3_task14_05.png">/genomics/images/part3_task14_05.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1453"></span>
<span id="line1454"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>On the start page click on view details for your annotation</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1455"></span>
<span id="line1456"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part3_task14_06.png">/genomics/images/part3_task14_06.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1457"></span>
<span id="line1458"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>You will get a summary of the sequence you uploaded and you have the ability to download the annotations to your computer.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1459"></span>
<span id="line1460"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Download the GFF3 annotation and open it in a text editor</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1461"></span>
<span id="line1462"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>##gff-version 3
<span id="line1463"></span>NODE_10_length_3324_cov_22.7003_ID_19  FIG  CDS 249 1163  . 0 ID=fig|562.4461.peg.1;Name=FIG010773: NAD-dependent epimerase/dehydratase</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1464"></span>
<span id="line1465"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>note: your output may be different.
<span id="line1466"></span>Scan down the list of annotations do any themes stand out?</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1467"></span>
<span id="line1468"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Click on ‘Browse annotated genome in SEED viewer’</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1469"></span>
<span id="line1470"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part3_task14_07.png">/genomics/images/part3_task14_07.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1471"></span>
<span id="line1472"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>This gives you a hierarchical view of the subsystems.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1473"></span>
<span id="line1474"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Browse the rest of the RAST server and get a feel for the possibilities the platform may offer you.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1475"></span>
<span id="line1476"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>When you’re ready, move on to (or back to) the de novo assembly part of the workshop.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1477"></span>
<span id="line1478"></span>  
<span id="line1479"></span>    </span><span>&lt;<span class="start-tag">h1</span> <span class="attribute-name">class</span>="<a class="attribute-value">page-title</a>"&gt;</span><span>Part 4. De-novo Assembly</span><span>&lt;/<span class="end-tag">h1</span>&gt;</span><span>
<span id="line1480"></span>    </span><span>&lt;<span class="start-tag">h3</span> <span class="attribute-name">id</span>="<a class="attribute-value">introduction</a>"&gt;</span><span>Introduction:</span><span>&lt;/<span class="end-tag">h3</span>&gt;</span><span>
<span id="line1481"></span>
<span id="line1482"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>In this section of the workshop we will continue the analysis of a strain of </span><span>&lt;<span class="start-tag">em</span>&gt;</span><span>E.coli</span><span>&lt;/<span class="end-tag">em</span>&gt;</span><span>. In the previous section we extracted those reads which did not map to the reference genome and assembled them. However, it is often necessary to be able to perform a de-novo assembly of a genome. In this case, rather than doing any remapping, we will start with the filtered reads we obtained in part 3 of the workshop.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1483"></span>
<span id="line1484"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>To do this we will a program called SPAdes to try to get the best possible assembly for a given genome. We will then generate assembly statistics and then produce some annotation via Pfam and BLAST.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1485"></span>
<span id="line1486"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Task 1: Start the Assembly</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1487"></span>
<span id="line1488"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>The assembly takes so the results have
<span id="line1489"></span>been </span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>pre-computed</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span> for you and are available in the directory</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1490"></span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">pre</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">code</span>&gt;</span><span>~/workshop_materials/genomics_tutorial/data/sequencing/ecoli_exeter/denovo_assembly
<span id="line1491"></span></span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">pre</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span>
<span id="line1492"></span>
<span id="line1493"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>If you were to run the command it would be as follows:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1494"></span>
<span id="line1495"></span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">pre</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">code</span>&gt;</span><span>spades.py -o denovo_assembly_rerun -1 E_Coli_CGATGT_L001_R1_001.filtered.fastq -2 E_Coli_CGATGT_L001_R2_001.filtered.fastq
<span id="line1496"></span></span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">pre</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span>
<span id="line1497"></span>
<span id="line1498"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>This will create a directory called denovo_assembly_rerun to hold the results.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1499"></span>
<span id="line1500"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>**Assembly theory **</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1501"></span>
<span id="line1502"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>We are using SPAdes </span><span>&lt;<span class="start-tag">a</span> <span class="attribute-name">href</span>="<a class="attribute-value" href="view-source:http://cab.spbu.ru/software/spades/">http://cab.spbu.ru/software/spades/</a>"&gt;</span><span>(http://cab.spbu.ru/software/spades/)</span><span>&lt;/<span class="end-tag">a</span>&gt;</span><span> to perform our assembly. It is a de Bruijn graph based assembler, similar to other short read assemblers like velvet </span><span>&lt;<span class="start-tag">a</span> <span class="attribute-name">href</span>="<a class="attribute-value" href="view-source:https://www.ebi.ac.uk/~zerbino/velvet/">https://www.ebi.ac.uk/~zerbino/velvet/</a>"&gt;</span><span>(https://www.ebi.ac.uk/~zerbino/velvet/)</span><span>&lt;/<span class="end-tag">a</span>&gt;</span><span>. The advantage of SPAdes is that it does lot of error correction and checking before and after the assembly which improve the final result. A downside of SPAdes is that it was designed for assembling reads from a single cell and although it does a good job with DNA prepared from a community it can leave in some low coverage sequences which are likely to be artifacts.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1503"></span>
<span id="line1504"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>You can read more about the comparison here </span><span>&lt;<span class="start-tag">a</span> <span class="attribute-name">href</span>="<a class="attribute-value" href="view-source:http://thegenomefactory.blogspot.co.uk/2013/08/how-spades-differs-from-velvet.html">http://thegenomefactory.blogspot.co.uk/2013/08/how-spades-differs-from-velvet.html</a>"&gt;</span><span>http://thegenomefactory.blogspot.co.uk/2013/08/how-
<span id="line1505"></span>spades-differs-from-velvet.html</span><span>&lt;/<span class="end-tag">a</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1506"></span>
<span id="line1507"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>SPAdes is also very easy to use - apart from telling it where your input files are the only parameter that you might want to choose is the length of k-mer.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1508"></span>
<span id="line1509"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>K-mer length</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1510"></span>
<span id="line1511"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Rather than store all reads individually which would be unfeasible for Illumina type datasets, de Bruijn assemblers convert each read to a series of k-mers and stores each k-mer once, along with information about how often it occurs and which other k-mers it links to. A short k-mer length (e.g. 21) reduces the chance that data will be missed from an assembly (e.g. due to reads being shorter than the k-mer length or sequencing errors in the k-mer), but can result in shorter contigs as repeat regions cannot be resolved.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1512"></span>
<span id="line1513"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>When using the Velvet assembler it is necessary to try a large combination of parameters to ensure that you obtain the ‘best’ possible assembly for a given dataset. There is even a program called VelvetOptimiser which does it for you. However, what ‘best’ actually means in the context of genome assembly is ill-defined. For a genomic assembly you want to try to obtain the lowest number of contigs, with the longest length, with the fewest errors. However, although numbers of contigs and longest lengths are easy to evaluate, it is extremely difficult to know what is or isn’t an error when sequencing a genome for the first time.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1514"></span>
<span id="line1515"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>SPAdes allows you to choose more than one k-mer length - it then performs an assembly for each k-mer and merges the result - trying to get the best of both worlds. It actually has some pre-calculated k-mer settings based on the length of reads you have, so you don’t even have to choose that.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1516"></span>
<span id="line1517"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Let’s look at the assembly process in more detail:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1518"></span>
<span id="line1519"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Description of k-mers:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1520"></span>
<span id="line1521"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>What are they? Let’s say you have a single read:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1522"></span>
<span id="line1523"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part4_task1_01.png">/genomics/images/part4_task1_01.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1524"></span>
<span id="line1525"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>The set of k-mers obtained from this read with length 6 (i.e. 6-mers) would be obtained by taking the first six bases, then moving the window along one base, taking the next 6 bases and so-on until the end of the read. E.g:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1526"></span>
<span id="line1527"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part4_task1_02.png">/genomics/images/part4_task1_02.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1528"></span>
<span id="line1529"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>You may well ask, ‘So what? How does that help?’ For a single read, it really doesn’t help. However let’s say that you have another read which is identical except for a single base:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1530"></span>
<span id="line1531"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part4_task1_03.png">/genomics/images/part4_task1_03.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1532"></span>
<span id="line1533"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Rather than represent both reads separately, we need only store the k-mers which differ and the number of times they occur. Note the ‘bubble’ like structure which occurs when a single base-change occurs. This kind of representation of reads is called a ‘k-mer graph’ (sometimes inaccurately referred to as a de-bruijn graph).</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1534"></span>
<span id="line1535"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Now let’s see what happens when we add in a third read. This is identical to the first read except for a change at another location. This results in an extra dead-end being added to the path.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1536"></span>
<span id="line1537"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part4_task1_04.png">/genomics/images/part4_task1_04.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1538"></span>
<span id="line1539"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>The job of any k-mer based assembler is to find a path through the k-mer graph which correctly represents the genome sequence.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1540"></span>
<span id="line1541"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">em</span>&gt;</span><span>Images courtesy of Mario Caccamo</span><span>&lt;/<span class="end-tag">em</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1542"></span>
<span id="line1543"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Description of coverage cutoff:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1544"></span>
<span id="line1545"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part4_task1_05.png">/genomics/images/part4_task1_05.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1546"></span>
<span id="line1547"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>In the figure above, you can see that the coverage of various k-mers varies between 1x and 3x. The question is which parts of the graph can be trimmed or removed so that we avoid any errors. As the graph stands, we could output three different contigs as there are three possible paths through the graph. However, we might wish to apply a coverage cutoff and remove the top right part of the graph because it has only 1x coverage and is more likely to be an error than a genuine variant.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1548"></span>
<span id="line1549"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>In a real graph you would have millions of k-mers and thousands of possible paths to deal with. The best way to estimate the coverage cutoff in such cases is to look at the frequency plot of contig (node) coverage, weighted by length. In the example below you can see that contigs with a coverage below 7x or 8x occur very infrequently. As such it is probably a good idea to exclude those contigs which have coverage less than this - they are likely to be errors.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1550"></span>
<span id="line1551"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Description of expected coverage:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1552"></span>
<span id="line1553"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>In the example below you can see a stretch of DNA with many reads mapping to it. There are two repetitive regions A1 and A2 which have identical sequence. If we try to assemble the reads without any knowledge of the true DNA sequence, we will end up with an assembly that is split into two or more contigs rather than one.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1554"></span>
<span id="line1555"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part4_task1_06.png">/genomics/images/part4_task1_06.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1556"></span>
<span id="line1557"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>One contig will contain all the reads which did not fall into A1 and A2. The other will contain reads from both A1 and A2. As such the coverage of the repetitive contig will be twice as high as that of the non-repetitive contig.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1558"></span>
<span id="line1559"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>If we had 5 repeats we would expect 5x more coverage relative to the non-repetitive contig. As such, provided we know what level of coverage we expect for a given set of data, we can use this information to try and resolve the number of repeats we expect.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1560"></span>
<span id="line1561"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>A commonly used metric to describe the effectiveness of the assembly is called N50 - see
<span id="line1562"></span></span><span>&lt;<span class="start-tag">a</span> <span class="attribute-name">href</span>="<a class="attribute-value" href="view-source:http://en.wikipedia.org/wiki/N50_statistic">http://en.wikipedia.org/wiki/N50_statistic</a>"&gt;</span><span>http://en.wikipedia.org/wiki/N50_statistic </span><span>&lt;/<span class="end-tag">a</span>&gt;</span><span>for details.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1563"></span>
<span id="line1564"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Task 2: Checking the assembly</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1565"></span>
<span id="line1566"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Change into the denovo_assembly directory:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1567"></span>
<span id="line1568"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>cd denovo_assembly</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1569"></span>
<span id="line1570"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Firstly we can filter out low coverage and very short contigs using a perl script:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1571"></span>
<span id="line1572"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>filter_low_coverage_contigs.pl <span class="entity"><span>&amp;</span>lt;</span> contigs.fasta <span class="entity"><span>&amp;</span>gt;</span> contigs.goodcov.fasta</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1573"></span>
<span id="line1574"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>We will use QUAST again
<span id="line1575"></span></span><span>&lt;<span class="start-tag">a</span> <span class="attribute-name">href</span>="<a class="attribute-value" href="view-source:http://quast.sourceforge.net/quast">http://quast.sourceforge.net/quast</a>"&gt;</span><span>(http://quast.sourceforge.net/quast)</span><span>&lt;/<span class="end-tag">a</span>&gt;</span><span> to generate some statistics on the assembly.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1576"></span>
<span id="line1577"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>quast.py --output-dir quast contigs.goodcov.fasta</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1578"></span>
<span id="line1579"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>This will create a directory called quast and create some statistics on the assembly you produced.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1580"></span>
<span id="line1581"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>cat quast/report.txt</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1582"></span>
<span id="line1583"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part4_task2_01_new.jpg">/genomics/images/part4_task2_01_new.jpg</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1584"></span>
<span id="line1585"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>You can see that there are 75 contigs in the assembly - so it is still far from complete. The N50 is 159K and the N75 is 95K so most of the assembly is in quite large contigs.
<span id="line1586"></span>This is fairly normal for a short read assembly - don’t expect complete chromosomes.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1587"></span>
<span id="line1588"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Task 3: Map reads back to assembly</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1589"></span>
<span id="line1590"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>A good check at this point is to map the original readsback to the contigs.fasta file and check that all positions are covered by reads. Amazingly it is actually possible for de-novo assemblers to generate contigs to which the original reads will not map.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1591"></span>
<span id="line1592"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Here we will use BWA again to index the contigs.fasta file and remap the reads. This is almost identical to the procedure we followed during the alignment section, the only difference is that instead of aligning to the reference genome, we are aligning to our newly created reference.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1593"></span>
<span id="line1594"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Make sure you are in the following directory:
<span id="line1595"></span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>cd ~/workshop_materials/genomics_tutorial/data/sequencing/ecoli_exeter/denovo_assembly/</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1596"></span>
<span id="line1597"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Let’s create a subdirectory to keep our work separate</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1598"></span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">pre</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">code</span>&gt;</span><span>mkdir remapping_to_assembly
<span id="line1599"></span>cd remapping_to_assembly
<span id="line1600"></span>cp ../contigs.fasta .
<span id="line1601"></span></span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">pre</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span>
<span id="line1602"></span>
<span id="line1603"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Let’s start by indexing the contigs.fasta file. Type:</span><span>&lt;<span class="start-tag">br</span> <span>/</span>&gt;</span><span>
<span id="line1604"></span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>bwa index contigs.fasta</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1605"></span>
<span id="line1606"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part4_task3_01.png">/genomics/images/part4_task3_01.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1607"></span>
<span id="line1608"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Once complete we can start to align the reads back to the contigs. Type (all on one line):</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1609"></span>
<span id="line1610"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>bwa mem -t 2 contigs.fasta ../../E_Coli_CGATGT_L001_R1_001.filtered.fastq ../../E_Coli_CGATGT_L001_R2_001.filtered.fastq <span class="entity"><span>&amp;</span>gt;</span> E_Coli_CGATGT_L001_filtered.sam</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1611"></span>
<span id="line1612"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Once complete we can convert the SAM file to a BAM file:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1613"></span>
<span id="line1614"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>samtools view -bS E_Coli_CGATGT_L001_filtered.sam <span class="entity"><span>&amp;</span>gt;</span> E_Coli_CGATGT_L001_filtered.bam</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1615"></span>
<span id="line1616"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>And then we can sort the BAM file:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1617"></span>
<span id="line1618"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>samtools sort -o E_Coli_CGATGT_L001_filtered.sorted.bam E_Coli_CGATGT_L001_filtered.bam</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1619"></span>
<span id="line1620"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Once completed, we can index the BAM file:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1621"></span>
<span id="line1622"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>samtools index E_Coli_CGATGT_L001_filtered.sorted.bam</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1623"></span>
<span id="line1624"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>We can then (at last!) obtain some basic summary statistics using the samtools flagstat command:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1625"></span>
<span id="line1626"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>samtools flagstat E_Coli_CGATGT_L001_filtered.sorted.bam</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1627"></span>
<span id="line1628"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part4_task3_02_new.jpg">/genomics/images/part4_task3_02_new.jpg</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1629"></span>
<span id="line1630"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>We can see here that very few of the reads do not map back to the contigs. Importantly 98% of reads are properly paired which gives us some indication that there are not too many mis-assemblies.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1631"></span>
<span id="line1632"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Run qualimap to get some more detailed information (and some images)</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1633"></span>
<span id="line1634"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>qualimap bamqc -outdir bamqc -bam E_Coli_CGATGT_L001_filtered.sorted.bam</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1635"></span>
<span id="line1636"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>firefox bamqc/qualimapReport.html</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1637"></span>
<span id="line1638"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>In the Chromosome stats section:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1639"></span>
<span id="line1640"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part4_task3_03_new.jpg">/genomics/images/part4_task3_03_new.jpg</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1641"></span>
<span id="line1642"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>The larger of our contigs have a mean coverage of around 50 - which is what we would expect from our original alignment.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1643"></span>
<span id="line1644"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>There is one contig which has the size of 67128 and much higher coverage - this is exactly the same as the contig we found in the unmapped reads - a good indication that it is a separate sequence (remember we suspected a plasmid) and not integrated into the chromosome.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1645"></span>
<span id="line1646"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part4_task3_04_new.jpg">/genomics/images/part4_task3_04_new.jpg</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1647"></span>
<span id="line1648"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Let’s double check that by blasting these contigs against the unmapped assembly contigs from part 4:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1649"></span>
<span id="line1650"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>blastn -subject ../contigs.goodcov.fasta -query ../../unmapped_assembly/spades_assembly/contigs.fasta <span class="entity"><span>&amp;</span>gt;</span> check_plasmid.blastn</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1651"></span>
<span id="line1652"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Open the file in a text editor:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1653"></span>
<span id="line1654"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>pluma check_plasmid.blastn</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1655"></span>
<span id="line1656"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>and about 30% of the way down the file you should find: (hint use search/find)</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1657"></span>
<span id="line1658"></span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">pre</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">code</span>&gt;</span><span>Query= NODE_1_length_67364_cov_602.091
<span id="line1659"></span>
<span id="line1660"></span>Length=67364
<span id="line1661"></span>
<span id="line1662"></span>Subject= NODE_24_length_67128_cov_604.709
<span id="line1663"></span>
<span id="line1664"></span>Length=67128
<span id="line1665"></span>
<span id="line1666"></span>
<span id="line1667"></span> Score = 1.164e+05 bits (63055),  Expect = 0.0
<span id="line1668"></span> Identities = 63057/63058 (99%), Gaps = 0/63058 (0%)
<span id="line1669"></span> Strand=Plus/Minus
<span id="line1670"></span></span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">pre</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span>
<span id="line1671"></span>
<span id="line1672"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>This shows us that this contig exactly almost matches that in the unmapped assembly, strongly supporting that this is a plasmid sequence and not integrated into the chromosomes.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1673"></span>
<span id="line1674"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Task 4: View assembly in IGV</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span>
<span id="line1675"></span>Load up IGV</span><span>&lt;<span class="start-tag">br</span> <span>/</span>&gt;</span><span>
<span id="line1676"></span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>igv.sh</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1677"></span>
<span id="line1678"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Click Genomes -<span class="entity"><span>&amp;</span>gt;</span> Load Genome from File.  We are going to import the contigs we have assembled as the reference. Unlike the reference genome though, we have no annotation available. Make sure you select the contigs.goodcov.fasta file for the complete de novo assembly (not the unmapped reads assembly).</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1679"></span>
<span id="line1680"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Once loaded, click on File-<span class="entity"><span>&amp;</span>gt;</span>Load From File select the E_Coli_CGATGT_L001_filtered.sorted.bam
<span id="line1681"></span>file. Again, make sure you load the file in the </span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>remapping_to_assembly</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span> directory.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1682"></span>
<span id="line1683"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part4_task4_01.png">/genomics/images/part4_task4_01.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1684"></span>
<span id="line1685"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Once loaded, explore some of the contigs in IGV. See if you can find anything unusual in any of the contigs.
<span id="line1686"></span>**
<span id="line1687"></span>Here is one to get you started - Select </span><span>&lt;<span class="start-tag">em</span>&gt;</span><span>NODE_3</span><span>&lt;/<span class="end-tag">em</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1688"></span>
<span id="line1689"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part4_task4_02_new.jpg">/genomics/images/part4_task4_02_new.jpg</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1690"></span>
<span id="line1691"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Why does the contig start and end in repetitive sequence (indicated by the coloured / white reads = low mapping quality)? You may need to zoom in to see the details. Think about what an assembler will do if it cannot uniquely assign reads.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1692"></span>
<span id="line1693"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>If an assembler cannot resolve these repetitive regions with paired-end reads or coverage information, it will generally be unable to assemble any further sequence for that contig. Therefore it is quite common to see contigs which start and end in sequence which is repeated elsewhere.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1694"></span>
<span id="line1695"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Here is another - Select </span><span>&lt;<span class="start-tag">em</span>&gt;</span><span>NODE_48</span><span>&lt;/<span class="end-tag">em</span>&gt;</span><span>.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1696"></span>
<span id="line1697"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Right click on the reads and select view as pairs:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1698"></span>
<span id="line1699"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part4_task4_03_new.jpg">/genomics/images/part4_task4_03_new.jpg</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1700"></span>
<span id="line1701"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>What do you think is going on here?
<span id="line1702"></span>Try blasting the contig sequence using BlastX at </span><span>&lt;<span class="start-tag">a</span> <span class="attribute-name">href</span>="<a class="attribute-value" href="view-source:http://blast.ncbi.nlm.nih.gov/Blast.cgi">http://blast.ncbi.nlm.nih.gov/Blast.cgi</a>"&gt;</span><span>http://blast.ncbi.nlm.nih.gov/Blast.cgi </span><span>&lt;/<span class="end-tag">a</span>&gt;</span><span> to identify which genes the contig contains. To obtain the sequence you can right click and select ‘Copy consensus sequence’</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1703"></span>
<span id="line1704"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>You can also do the same for individual reads, but you need to un-select ‘View as pairs’ before right clicking on a read. You may lose track of the paired-end reads and find it easier to copy the read name before un-selecting ‘View as pairs’ and then and then pasting it into the ‘Select by name’ search box.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1705"></span>
<span id="line1706"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>You should find that the contig contains at least two phage genes. There appear to be at least two phages present, one which seems to be the full contig, the other with the red read-pairs seems to be missing the sequence in the middle of the contig.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1707"></span>
<span id="line1708"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Annotation of de-novo assembled contigs</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1709"></span>
<span id="line1710"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>We will now annotate the contigs using BLAST, Pfam and RAST as with the
<span id="line1711"></span>unmapped contigs.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1712"></span>
<span id="line1713"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Task 5: Obtain open reading frames</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1714"></span>
<span id="line1715"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>As before, we’ll call open reading frames within the de-novo assembly. Again, we will use codon table 11 which defines the bacterial codon usage table </span><span>&lt;<span class="start-tag">a</span> <span class="attribute-name">href</span>="<a class="attribute-value" href="view-source:http://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi">http://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi</a>"&gt;</span><span>(http://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi)</span><span>&lt;/<span class="end-tag">a</span>&gt;</span><span> and state that the sequences we are dealing with are not circular. We will also restrict the ORFs to just those sequences longer than 300 nucleotides (i.e. 100
<span id="line1716"></span>amino acids). We will store the results in file contigs.orf.fasta.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1717"></span>
<span id="line1718"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Make sure you are in the denovo_assembly/ directory:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1719"></span>
<span id="line1720"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>getorf -table 11  -circular N -minsize 300 -sequence contigs.goodcov.fasta -outseq contigs.orf.fasta</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1721"></span>
<span id="line1722"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">em</span>&gt;</span><span>The following two tasks are optional. Be warned - the BLAST searches and RAST will take several days! I recommend you start these now and then these and proceed to Task 9.</span><span>&lt;/<span class="end-tag">em</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1723"></span>
<span id="line1724"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Task 6 (Optional): Search open reading frames against NCBI non-redundant database</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1725"></span>
<span id="line1726"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>We can also search these open reading frames against the NCBI non-redundant database.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1727"></span>
<span id="line1728"></span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">pre</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">code</span>&gt;</span><span>blastp -db ~/workshop_materials/genomics_tutorial/db/blast/nr -query contigs.orf.fasta -evalue 1e-06 -num_threads 4 -show_gis -num_alignments 10 -num_descriptions 10 -out contigs.orf.fasta.blastp
<span id="line1729"></span></span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">pre</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span>
<span id="line1730"></span>
<span id="line1731"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Task 7 (Optional): Search contigs against NCBI non-redundant database</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1732"></span>
<span id="line1733"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>The following command executes a nucleotide BLAST search (blastn) of the sequences in the
<span id="line1734"></span>contigs.fasta file against the non-redundant database. Again we restrict ourselves to 10 results per hit and an e-value cutoff of 1e-06.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1735"></span>
<span id="line1736"></span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">pre</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">code</span>&gt;</span><span>blastn -db ~/workshop_materials/genomics_tutorial/db/blast/nt -query contigs.fasta -evalue 1e-06 -num_threads 4 -show_gis -num_alignments 10 -num_descriptions 10 -out contigs.fasta.blastn
<span id="line1737"></span></span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">pre</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span>
<span id="line1738"></span>
<span id="line1739"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Task 8 (Optional)</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span>
<span id="line1740"></span>Run the contigs through the RAST server and import the resulting GFF annotation into IGV (refer back to Part 3 for instructions).</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1741"></span>
<span id="line1742"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Task 9: Run open reading frames through Pfam</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1743"></span>
<span id="line1744"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>As with the unmapped reads we will search the open reading frames against the Pfam HMM database of protein families. Later on we will be able to use these results to identify Pfam domains which are unique to a particular strain.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1745"></span>
<span id="line1746"></span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">pre</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">code</span>&gt;</span><span>pfam_scan.pl -fasta contigs.orf.fasta -dir ~/workshop_materials/genomics_tutorial/db/pfam/ -outfile contigs.orf.pfam -cpu 2 -pfamB -as
<span id="line1747"></span></span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">pre</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span>
<span id="line1748"></span>
<span id="line1749"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">em</span>&gt;</span><span>This will take around 5 hours so it is recommended that you leave this running while continuing with the rest of the tutorial. If it is still running when you finish
<span id="line1750"></span>your session for today, leave your instance running overnight, but please be sure to turn it off in the morning!</span><span>&lt;/<span class="end-tag">em</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1751"></span>
<span id="line1752"></span></span><span>&lt;<span class="start-tag">h2</span> <span class="attribute-name">id</span>="<a class="attribute-value">hybrid-assembly</a>"&gt;</span><span>Hybrid Assembly</span><span>&lt;/<span class="end-tag">h2</span>&gt;</span><span>
<span id="line1753"></span>
<span id="line1754"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>You will have seen that even with good coverage and a relatively long (300bp) paired end Illumina dataset - the assembly we get is still fairly fragmented. Our </span><span>&lt;<span class="start-tag">em</span>&gt;</span><span>E.coli</span><span>&lt;/<span class="end-tag">em</span>&gt;</span><span> example assembles into 75 contigs and the largest contig is around 10% of the genome size.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1755"></span>
<span id="line1756"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Why is this?</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1757"></span>
<span id="line1758"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>One possible reason would be that regions of the original genome were not sequenced, or sequenced at too low coverage to assemble correctly. Regions of the genome will occur with different frequencies in the library that was sequenced - You can see this in
<span id="line1759"></span>the variation of coverage when you did the alignment. This can be due to inherent biases in the preparation and the random nature of the process.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1760"></span>
<span id="line1761"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>However as coverage increases the chances of not sequencing a particular region of the genome reduces and the most significant factor becomes the resolution of repeats within the assembly process. If two regions contain the same or very similar sequences the assembler cannot reliably detect that they are actually two or more distinct sequences and incorrectly ‘collapses’ the repeat into a single sequence. The assembler is now effectively missing a sequence and therefore breaks in the assembly occur.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1762"></span>
<span id="line1763"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>One resolution to this is to use a sequencing technology like PacBio or Sanger which can produce longer reads - the reads are then long enough to include the repeated sequence, plus some unique sequence, and the problem can be resolved. Unfortunately getting enough coverage using Sanger sequencing is expensive and PacBio - although relatively inexpensive has a high error rate.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1764"></span>
<span id="line1765"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>An approach becoming more and more popular is to combine technologies. For example: high quality Illumina sequencing to get the accuracy of reads combined with low quality PacBio sequencing to enable the repeats to be spanned and correctly resolved.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1766"></span>
<span id="line1767"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Our exercise will be to use Illumina and PacBio datasets to assemble a species of </span><span>&lt;<span class="start-tag">em</span>&gt;</span><span>Pseudomonas</span><span>&lt;/<span class="end-tag">em</span>&gt;</span><span>.
<span id="line1768"></span>These are subsets of data used in “Evaluation and validation of de novo
<span id="line1769"></span>and hybrid assembly techniques to derive high-quality genome sequences” Utturkar et al., 2014.
<span id="line1770"></span></span><span>&lt;<span class="start-tag">a</span> <span class="attribute-name">href</span>="<a class="attribute-value" href="view-source:http://www.ncbi.nlm.nih.gov/pubmed/24930142">http://www.ncbi.nlm.nih.gov/pubmed/24930142</a>"&gt;</span><span>(http://www.ncbi.nlm.nih.gov/pubmed/24930142)</span><span>&lt;/<span class="end-tag">a</span>&gt;</span><span>.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1771"></span>
<span id="line1772"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>This paper also contains a good explanation of the process and different approaches that are available.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1773"></span>
<span id="line1774"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Note</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span> our example is not a particularly good dataset and the improvement is quite modest.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1775"></span>
<span id="line1776"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Firstly, as always, it is important to check and understand the quality of the data you are working with:
<span id="line1777"></span>Change to the directory and run fastqc</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1778"></span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">pre</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">code</span>&gt;</span><span>cd ~/workshop_materials/genomics_tutorial/data/sequencing/pseudomonas_gm41
<span id="line1779"></span>fastqc
<span id="line1780"></span></span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">pre</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span>
<span id="line1781"></span>
<span id="line1782"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Open the files SRR1042836a.fastq SRR491287a_1.fastq and SRR491287a_2.fastq and look at the re-
<span id="line1783"></span>ports generated.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1784"></span>
<span id="line1785"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Note that the quality of the PacBio reads (SRR1042836a.fastq) is much lower than the Illumina reads with a greater than 1 chance in 10 of there being a mistake for most reads.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1786"></span>
<span id="line1787"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part4_task10_01.png">/genomics/images/part4_task10_01.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1788"></span>
<span id="line1789"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>However, importantly, the length of the PacBio reads is much longer.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1790"></span>
<span id="line1791"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part4_task10_02.png">/genomics/images/part4_task10_02.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1792"></span>
<span id="line1793"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Trim the Illumina reads as before:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1794"></span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">pre</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">code</span>&gt;</span><span>fastq-mcf ../../reference/adaptors/adaptors.fasta SRR491287a_1.fastq SRR491287a_2.fastq -o SRR491287a_1.filtered.fastq -o SRR491287a_2.filtered.fastq -q 20 -p 10 -u -x 0.01
<span id="line1795"></span></span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">pre</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span>
<span id="line1796"></span>
<span id="line1797"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>You can check the number of filtered reads using grep -c and the quality of trimmed reads
<span id="line1798"></span>with fastqc if you want.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1799"></span>
<span id="line1800"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>For this exercise we want the long reads from PacBio even though they are low quality. We are relying on the assembler to use them appropriately.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1801"></span>
<span id="line1802"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Task 11: Illumina Only Assembly</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1803"></span>
<span id="line1804"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Firstly let’s construct an assembly using only the available Illumina data.
<span id="line1805"></span>Make sure you are in the directory
<span id="line1806"></span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>cd ~/workshop_materials/genomics_tutorial/data/sequencing/pseudomonas_gm41</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1807"></span>
<span id="line1808"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>and Run:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1809"></span>
<span id="line1810"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>spades.py --threads 2 --careful -o illumina_only_assembly -1 SRR491287a_1.filtered.fastq -2 SRR491287a_2.filtered.fastq</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1811"></span>
<span id="line1812"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>This may take some time so the data has been precomputed and is available in illumina_assembly if you are impatient!</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1813"></span>
<span id="line1814"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Change to the directory:</span><span>&lt;<span class="start-tag">br</span> <span>/</span>&gt;</span><span>
<span id="line1815"></span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>cd illumina_only_assembly</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1816"></span>
<span id="line1817"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Filter out low coverage and very short contigs using a perl script:</span><span>&lt;<span class="start-tag">br</span> <span>/</span>&gt;</span><span>
<span id="line1818"></span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>filter_low_coverage_contigs.pl <span class="entity"><span>&amp;</span>lt;</span> contigs.fasta <span class="entity"><span>&amp;</span>gt;</span> contigs.goodcov.fasta</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1819"></span>
<span id="line1820"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Let’s look at the metrics for the assembly.</span><span>&lt;<span class="start-tag">br</span> <span>/</span>&gt;</span><span>
<span id="line1821"></span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>quast.py --output-dir quast contigs.goodcov.fasta</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1822"></span>
<span id="line1823"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>cat quast/report.txt</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1824"></span>
<span id="line1825"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part4_task11_01_new.jpg">/genomics/images/part4_task11_01_new.jpg</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1826"></span>
<span id="line1827"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Your results may be slightly different. This is because spades uses a random seed that changes every time</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1828"></span>
<span id="line1829"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Task 12: Create Hybrid Assembly</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1830"></span>
<span id="line1831"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Now will execute the same command, but this time include the longer PacBio reads to see the effect it has on our assembly. Change back into the directory</span><span>&lt;<span class="start-tag">br</span> <span>/</span>&gt;</span><span>
<span id="line1832"></span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>cd ~/workshop_materials/genomics_tutorial/data/sequencing/pseudomonas_gm41</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1833"></span>
<span id="line1834"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Run (This may take some time so the data has been precomputed and is available in hybrid_assembly_pre/ if you are impatient!):</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1835"></span>
<span id="line1836"></span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">pre</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">code</span>&gt;</span><span>spades.py --threads 2 --careful -o hybrid_assembly --pacbio SRR1042836a.fastq -1 SRR491287a_1.filtered.fastq -2 SRR491287a_2.filtered.fastq
<span id="line1837"></span></span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">pre</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span>
<span id="line1838"></span>
<span id="line1839"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Change to the directory: </span><span>&lt;<span class="start-tag">br</span> <span>/</span>&gt;</span><span>
<span id="line1840"></span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>cd hybrid_assembly</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1841"></span>
<span id="line1842"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Filter out low coverage and very short contigs using the perl script:</span><span>&lt;<span class="start-tag">br</span> <span>/</span>&gt;</span><span>
<span id="line1843"></span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>filter_low_coverage_contigs.pl <span class="entity"><span>&amp;</span>lt;</span> contigs.fasta <span class="entity"><span>&amp;</span>gt;</span> contigs.goodcov.fasta</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1844"></span>
<span id="line1845"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Let’s look at the metrics for the assembly - this time we will compare it with the illumina only assembly:</span><span>&lt;<span class="start-tag">br</span> <span>/</span>&gt;</span><span>
<span id="line1846"></span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>quast.py --output-dir quast contigs.goodcov.fasta ../illumina_only_assembly/contigs.goodcov.fasta</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1847"></span>
<span id="line1848"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>cat quast/report.txt</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1849"></span>
<span id="line1850"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part4_task12_01_new.jpg">/genomics/images/part4_task12_01_new.jpg</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1851"></span>
<span id="line1852"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>firefox quast/report.html</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1853"></span>
<span id="line1854"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>It seems that using the longer reads has improved the completeness of our assembly - reducing the number of contigs. In truth, this is a very modest improvement, for bacteria it is difficult to find a dataset where the long reads help, without making the short reads redundant and assembling eukaryotes would take too long in the context of the course.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1855"></span>
<span id="line1856"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Task 13: Align reads back to reference</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span>
<span id="line1857"></span>Let’s realign our original reads back to the assembly and see what we have - refer to previous notes if you are unsure of the steps.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1858"></span>
<span id="line1859"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Start in the hybrid assembly directory</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1860"></span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">pre</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">code</span>&gt;</span><span>cd ~/workshop_materials/genomics_tutorial/data/sequencing/pseudomonas_gm41/hybrid_assembly
<span id="line1861"></span>mkdir remapping_to_assembly
<span id="line1862"></span>cd remapping_to_assembly
<span id="line1863"></span>cp ../contigs.fasta .
<span id="line1864"></span>bwa index contigs.fasta
<span id="line1865"></span></span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">pre</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span>
<span id="line1866"></span>
<span id="line1867"></span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">pre</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">code</span>&gt;</span><span>bwa mem -t 2 contigs.fasta ../../SRR491287a_1.filtered.fastq ../../SRR491287a_2.filtered.fastq <span class="entity"><span>&amp;</span>gt;</span> gm41.illumina.sam
<span id="line1868"></span>samtools view -bS gm41.illumina.sam <span class="entity"><span>&amp;</span>gt;</span> gm41.illumina.bam
<span id="line1869"></span>samtools sort -o gm41.illumina.sorted.bam gm41.illumina.bam  
<span id="line1870"></span>samtools index gm41.illumina.sorted.bam
<span id="line1871"></span>samtools flagstat gm41.illumina.sorted.bam
<span id="line1872"></span></span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">pre</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span>
<span id="line1873"></span>
<span id="line1874"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>We can also map the PacBio reads, but we need to tell bwa we are using PacBio reads</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1875"></span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">pre</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">code</span>&gt;</span><span>bwa mem -t 2 -x pacbio contigs.fasta ../../SRR1042836a.fastq <span class="entity"><span>&amp;</span>gt;</span> gm41.pacbio.sam
<span id="line1876"></span>samtools view -bS gm41.pacbio.sam <span class="entity"><span>&amp;</span>gt;</span> gm41.pacbio.bam
<span id="line1877"></span>samtools sort -o gm41.pacbio.sorted.bam  gm41.pacbio.bam  
<span id="line1878"></span>samtools index gm41.pacbio.sorted.bam
<span id="line1879"></span>samtools flagstat gm41.pacbio.sorted.bam
<span id="line1880"></span></span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">pre</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span>
<span id="line1881"></span>
<span id="line1882"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part4_task13_01_new.jpg">/genomics/images/part4_task13_01_new.jpg</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1883"></span>
<span id="line1884"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>You will notice that not such a high proportion of PacBio reads map back to the assembly.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1885"></span>
<span id="line1886"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Now start igv:</span><span>&lt;<span class="start-tag">br</span> <span>/</span>&gt;</span><span>
<span id="line1887"></span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>igv.sh</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1888"></span>
<span id="line1889"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Load your assembled genome - Click on genome - load from file</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1890"></span>
<span id="line1891"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Make sure you get the assembly from the hybrid_assembly (igv remembers the previous directory which may contain similar files.)</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1892"></span>
<span id="line1893"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Now load your 2 alignment files:</span><span>&lt;<span class="start-tag">br</span> <span>/</span>&gt;</span><span>
<span id="line1894"></span>click on load from File and then select gm41.pacbio.sorted.bam and gm41.illumina.sorted.bam</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1895"></span>
<span id="line1896"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>On the toolbar select - “Show Details on Click”</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1897"></span>
<span id="line1898"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part4_task13_02.jpg">/genomics/images/part4_task13_02.jpg</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1899"></span>
<span id="line1900"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Find a region that has decent coverage of both reads and zoom in. (Region shown here: NODE_43_length_17566_cov_24.3317:8,699-8,867)</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1901"></span>
<span id="line1902"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part4_task13_03_new.jpg">/genomics/images/part4_task13_03_new.jpg</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1903"></span>
<span id="line1904"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>You can see that the PacBio reads are much longer, but the error rate particularly insertions and deletions is much higher than for the Illumina reads.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1905"></span>
<span id="line1906"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Explore a few other contigs to see if you can find something that looks like an error or mis-assembly. Remember the assembly process is difficult and far from perfect.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1907"></span>
<span id="line1908"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Summary</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1909"></span>
<span id="line1910"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>You have seen that de-novo assembly of short reads is a challenging problem. Even for small genomes, the resulting assembly is fragmented into contigs and far from complete.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1911"></span>
<span id="line1912"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Incorporating longer reads to produce a hybrid assembly can be used to reduce the fragmentation of the genome. We have only used a single (perhaps the simplest) technique to incorporate long reads.
<span id="line1913"></span>You can read more about hybrid assembly techniques here: </span><span>&lt;<span class="start-tag">a</span> <span class="attribute-name">href</span>="<a class="attribute-value" href="view-source:http://www.ncbi.nlm.nih.gov/pubmed/24930142">http://www.ncbi.nlm.nih.gov/pubmed/24930142</a>"&gt;</span><span>http://www.ncbi.nlm.nih.gov/pubmed/24930142 </span><span>&lt;/<span class="end-tag">a</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1914"></span>
<span id="line1915"></span>  
<span id="line1916"></span>    </span><span>&lt;<span class="start-tag">h1</span> <span class="attribute-name">class</span>="<a class="attribute-value">page-title</a>"&gt;</span><span>Part 5. Comparative Analysis</span><span>&lt;/<span class="end-tag">h1</span>&gt;</span><span>
<span id="line1917"></span>    </span><span>&lt;<span class="start-tag">h3</span> <span class="attribute-name">id</span>="<a class="attribute-value">introduction</a>"&gt;</span><span>Introduction:</span><span>&lt;/<span class="end-tag">h3</span>&gt;</span><span>
<span id="line1918"></span>
<span id="line1919"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/content/genomics2017/images/2017_Workshop_on_Genomics_FINAL-102_1.png">/content/genomics2017/images/2017_Workshop_on_Genomics_FINAL-102_1.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1920"></span>
<span id="line1921"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>In the previous sections you have
<span id="line1922"></span>been taken through the steps required to:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1923"></span>
<span id="line1924"></span></span><span>&lt;<span class="start-tag">ol</span>&gt;</span><span>
<span id="line1925"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>QC and filter Illumina data</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line1926"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>Remap Illumina short-read data to a reference sequence</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line1927"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>View the results in IGV</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line1928"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>Identify SNPs and Indels in an automated fashion using samtools and bcftools</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line1929"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>Determine whether SNPs result in synonymous or non-synonymous changes in the corresponding amino acid</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line1930"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>Extract unmapped reads</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line1931"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>Assemble unmapped reads and obtain assembly statistics</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line1932"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>Annotate unmapped reads using Pfam, RAST and/or BLAST</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line1933"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>Assemble a bacterial genome de-novo using SPAdes</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line1934"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>Obtain assembly statistics</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line1935"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>Annotate as per the unmapped reads (where computationally feasible).</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line1936"></span></span><span>&lt;/<span class="end-tag">ol</span>&gt;</span><span>
<span id="line1937"></span>
<span id="line1938"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Now we want you to do the same on a set of </span><span>&lt;<span class="start-tag">em</span>&gt;</span><span>Vibrio parahaemolyticus</span><span>&lt;/<span class="end-tag">em</span>&gt;</span><span> strains which have been isolated and sequenced. There are six strains available depending on how much time is available and enthusiasm you have - choose a number of strains (at least 2) as we want to run some comparisons.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1939"></span>
<span id="line1940"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>The strains can be found in:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1941"></span>
<span id="line1942"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>cd ~/workshop_materials/genomics_tutorial/data/sequencing/Vibrio_parahaemolyticus</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1943"></span>
<span id="line1944"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part5_intro_01.png">/genomics/images/part5_intro_01.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1945"></span>
<span id="line1946"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Under each Sample directory is a subdirectory called raw_illumina_reads which contains the fastq files.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1947"></span>
<span id="line1948"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>For remapping, the reference can be found in the folder:</span><span>&lt;<span class="start-tag">br</span> <span>/</span>&gt;</span><span>
<span id="line1949"></span>~/workshop_materials/genomics_tutorial/data/reference/Vibrio_parahaemolyticus_RIMD_2210633_uid57969
<span id="line1950"></span>(Remember, you will need to create an index first).</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1951"></span>
<span id="line1952"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>For each strain, make a list of:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1953"></span>
<span id="line1954"></span></span><span>&lt;<span class="start-tag">ul</span>&gt;</span><span>
<span id="line1955"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>SNPs, Indels and their effects (from the remapping)</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line1956"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>Missing genes (from the remapping)</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line1957"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>Novel plasmids and/or genes (Pfam domains are the easiest way to do this via denovo assembly of unmapped reads - when performing the assembly - don’t specify the k-mers SPAdes will choose appropriate ones.)</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line1958"></span></span><span>&lt;/<span class="end-tag">ul</span>&gt;</span><span>
<span id="line1959"></span>
<span id="line1960"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Once completed, see if you can predict what the phenotype of these bacteria might be. Then proceed to the final part of the tutorial where we will compare the results from all of these strains.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1961"></span>
<span id="line1962"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>N.B.</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span> It is recommended that you follow the same directory naming convention we have followed here (i.e. one for remapping to the reference, another for assembly of unmapped reads and a final one for the denovo assembly).</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1963"></span>
<span id="line1964"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>These tasks may take you several days. However, remember that all of the basic procedures are detailed in the previous sections - only the input FASTQ files will have changed. Feel free to refer to these previous tasks to remind yourself of the commands and parameters. By all means feel free to play around with different parameters if you wish, although remember that the results may differ from those you see here.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1965"></span>
<span id="line1966"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Just to give you some guidance:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1967"></span>
<span id="line1968"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>You should find that strain Sample_T0347070 yields many more SNPs than other strains.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1969"></span>
<span id="line1970"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>You may find that some scripts and programs run more slowly because of these extra differences and larger datasets.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1971"></span>
<span id="line1972"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Also, if you find the de novo assembly process causes your session to end, the chances are that SPAdes has caused your instance to run out of memory. If this happens, increase the minimum k-mer size in the spades.py command line.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1973"></span>
<span id="line1974"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Here we will use a script to compare the variants called in each sample. Ensure you are in the</span><span>&lt;<span class="start-tag">br</span> <span>/</span>&gt;</span><span>
<span id="line1975"></span>~/workshop_materials/genomics_tutorial/data/sequencing/Vibrio_parahaemolyticus directory</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1976"></span>
<span id="line1977"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>First of all, let’s make a directory to store the results of the comparison:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1978"></span>
<span id="line1979"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>mkdir snp_comparison</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1980"></span>
<span id="line1981"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>We need a copy of all of the vcf4 files we created here. This is a quick way to do it - paste this in as one command</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1982"></span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">pre</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">code</span>&gt;</span><span>for sample in Sample*
<span id="line1983"></span>do
<span id="line1984"></span>cp -v $sample/remapping_to_reference/out.snps.vcf4 snp_comparison/\$sample.out.snps.vcf4
<span id="line1985"></span>done
<span id="line1986"></span></span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">pre</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span>
<span id="line1987"></span>
<span id="line1988"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>cd snp_comparison</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1989"></span>
<span id="line1990"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Note that the copy commands may require modification depending on where you have saved the variant call results.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1991"></span>
<span id="line1992"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Our directory contents should look something like:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1993"></span>
<span id="line1994"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part5_intro_02.png">/genomics/images/part5_intro_02.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1995"></span>
<span id="line1996"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>We’ll now set up some variables so we don’t have to type long path names</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1997"></span>
<span id="line1998"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>ref=~/workshop_materials/genomics_tutorial/data/reference/Vibrio_parahaemolyticus_RIMD_2210633_uid57969/Vibrio_parahaemolyticus_RIMD_2210633_uid57969.fasta</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line1999"></span>
<span id="line2000"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>gff=~/workshop_materials/genomics_tutorial/data/reference/Vibrio_parahaemolyticus_RIMD_2210633_uid57969/Vibrio_parahaemolyticus_RIMD_2210633_uid57969.gff</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2001"></span>
<span id="line2002"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>samples=`ls *.vcf`</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2003"></span>
<span id="line2004"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>We can now use $ref instead of the long path to our reference and $gff for the feature file.
<span id="line2005"></span>e.g.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2006"></span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">pre</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">code</span>&gt;</span><span>head $ref
<span id="line2007"></span>echo $samples
<span id="line2008"></span></span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">pre</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span>
<span id="line2009"></span>
<span id="line2010"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>When we are happy our variables are correct then run:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2011"></span>
<span id="line2012"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>snp_comparator.pl 10  $ref  $gff  $samples  <span class="entity"><span>&amp;</span>gt;</span> snp_comparison.txt</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2013"></span>
<span id="line2014"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Looking at the snp_comparison.txt file (either in a text editor, or in a spreadsheet):</span><span>&lt;<span class="start-tag">br</span> <span>/</span>&gt;</span><span>
<span id="line2015"></span>If you have chosen different samples - you will get different results of course.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2016"></span>
<span id="line2017"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Here we can see the chromosome ID, the position in bp, the reference base and the base at each position as well as the gene (if any) the variant occurs in as well as the effect (silent, non-silent or indel).</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2018"></span>
<span id="line2019"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Obtaining a phylogeny based on synonymous SNPs only:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2020"></span>
<span id="line2021"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>How are the strains related on the basis of these variants? We can ask a number of questions, but if we are looking at the long-term evolutionary history of the strains we should only look at synonymous (i.e. silent) mutations as these should not confer a significant selective advantage to any strain. Using the data snp_comparison.txt file, we can form ‘pseudo-sequences’ using the script snp2tree_fullsequence.pl</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2022"></span>
<span id="line2023"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>These are concatenated bases consisting of only those positions which are silent across all strains. It is essentially the same as turning each column of each strain in the snp_comparison.txt file into a FASTA entry.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2024"></span>
<span id="line2025"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>snp2tree_fullsequence.pl snp_comparison.txt <span class="entity"><span>&amp;</span>gt;</span> synonymous_tree.fasta</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2026"></span>
<span id="line2027"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Examine the contents of the tree.fasta file. We can then treat this file as an alignment (since each base in each sequence is at the same position on the chromosome) and pass it to a phylogeny program called FastTree. FastTree will take an input alignment and output a Newick formatted tree </span><span>&lt;<span class="start-tag">a</span> <span class="attribute-name">href</span>="<a class="attribute-value" href="view-source:http://en.wikipedia.org/wiki/Newick_format">http://en.wikipedia.org/wiki/Newick_format</a>"&gt;</span><span>(http://en.wikipedia.org/wiki/Newick_format).</span><span>&lt;/<span class="end-tag">a</span>&gt;</span><span> \</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2028"></span>
<span id="line2029"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>FastTree -nt -gtr <span class="entity"><span>&amp;</span>lt;</span> synonymous_tree.fasta <span class="entity"><span>&amp;</span>gt;</span> synonymous_tree.newick</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2030"></span>
<span id="line2031"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Now we can visually view this tree by using an online tool.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2032"></span>
<span id="line2033"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>firefox http://www.trex.uqam.ca/index.php?action=newick</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2034"></span>
<span id="line2035"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part5_04.png">/genomics/images/part5_04.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2036"></span>
<span id="line2037"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Either paste the contents of the .newick file into the window or select ‘Sequences file’ and load the file through the browser. Then select ‘View Tree’</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2038"></span>
<span id="line2039"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Advanced task:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2040"></span>
<span id="line2041"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Copy the snp2tree_fullsequence.pl script to this directory: (~/workshop_materials/genomics_tutorial/data/sequencing/Vibrio_parahaemolyticus/snp_comparison) and modify it so that it selects positions containing only non-silent mutations (not indels as these modify the alignment). Generate a new alignment and compare the resulting tree against the silent mutations.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2042"></span>
<span id="line2043"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>You will need to have some experience of programming in the Perl language to do this.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2044"></span>
<span id="line2045"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/content/genomics2017/images/2017_Workshop_on_Genomics_FINAL-107_1.png">/content/genomics2017/images/2017_Workshop_on_Genomics_FINAL-107_1.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2046"></span>
<span id="line2047"></span>
<span id="line2048"></span>  
<span id="line2049"></span>    </span><span>&lt;<span class="start-tag">h1</span> <span class="attribute-name">class</span>="<a class="attribute-value">page-title</a>"&gt;</span><span>Part 6. Long Read Assembly</span><span>&lt;/<span class="end-tag">h1</span>&gt;</span><span>
<span id="line2050"></span>    </span><span>&lt;<span class="start-tag">h3</span> <span class="attribute-name">id</span>="<a class="attribute-value">introduction</a>"&gt;</span><span>Introduction:</span><span>&lt;/<span class="end-tag">h3</span>&gt;</span><span>
<span id="line2051"></span>
<span id="line2052"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>The latest third-generation sequencing platforms from Oxford Nanopore and Pacific Biosciences enable sequencing of </span><span>&lt;<span class="start-tag">em</span>&gt;</span><span>single DNA molecules.</span><span>&lt;/<span class="end-tag">em</span>&gt;</span><span> Remember second generation sequencing platforms (e.g. Illumina) rely on sequencing a group of ~1000 mono-clonal molecules formed via PCR-like reactions. This has advantages in terms of reducing the quantity of DNA input required to enable sequencing, usually enables lower per-base costs, but has many shortcomings.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2053"></span>
<span id="line2054"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>In the case of Illumina, although the sequencer is capable of generating large quantities of data (terabases in the case of a HiSeq run), the amplification step introduces GC-biases and also
<span id="line2055"></span>places a maximum limit on the size of the fragments which can be sequenced since the amplification becomes inefficient at larger fragments sizes and sequencing becomes de-phased within individual mono-clonal clusters due to polymerase errors. In the case of genome assembly, this introduces fragmentation of the assembly in areas of low coverage or repetitive regions which cannot be spanned.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2056"></span>
<span id="line2057"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Third generation sequencing of single molecules has the following features:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2058"></span>
<span id="line2059"></span></span><span>&lt;<span class="start-tag">ul</span>&gt;</span><span>
<span id="line2060"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>Ability to sequence native genomic DNA without limitations on size (100s kilobases are possible)</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line2061"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>Higher single pass per-base error rates than second-generation sequencers</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line2062"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>Higher per-base cost  than second-generation sequencing</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line2063"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>Full length cDNA or amplicon sequencing</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line2064"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>Some platforms such as the MinION from Oxford Nanopore hold the potential to enable portable sequencing and sequencing of other types of nucleic acid (e.g. RNA).</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line2065"></span></span><span>&lt;/<span class="end-tag">ul</span>&gt;</span><span>
<span id="line2066"></span>
<span id="line2067"></span></span><span>&lt;<span class="start-tag">h3</span> <span class="attribute-name">id</span>="<a class="attribute-value">pacbio-sequencing</a>"&gt;</span><span>PacBio sequencing</span><span>&lt;/<span class="end-tag">h3</span>&gt;</span><span>
<span id="line2068"></span>
<span id="line2069"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>You can watch a video outlining PacBio sequencing at </span><span>&lt;<span class="start-tag">a</span> <span class="attribute-name">href</span>="<a class="attribute-value" href="view-source:http://sequencing.exeter.ac.uk/pacific-biosciences-overview/">http://sequencing.exeter.ac.uk/pacific-biosciences-overview/</a>"&gt;</span><span>http://sequencing.exeter.ac.uk/pacific-
<span id="line2070"></span>biosciences-overview/.</span><span>&lt;/<span class="end-tag">a</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2071"></span>
<span id="line2072"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Pacific Biosciences have two sequencing platforms - RSII and Sequel. The RSII is an older platform capable of generating around 750Mbases of data in 4-6 hours. The Sequel platform is the latest platform and in theory can generate up to 7G bases of data in a similar time frame. However the chemistry is still being refined and at present 4-5 Gbases is the maximum which should be expected.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2073"></span>
<span id="line2074"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>The Sequel platform is PacBio’s focus for development at the moment, with optics included on the chip rather than the instrument, PacBio aims to increase the number of Zero Mode Waveguides (ZMWs) to increase throughput without significant instrument modifications.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2075"></span>
<span id="line2076"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>PacBio SMRTcells (RSII left - Sequel right)
<span id="line2077"></span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part6_pacbio_01.jpg">/genomics/images/part6_pacbio_01.jpg</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2078"></span>
<span id="line2079"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>PacBio Yield and read length distributions (RSII left - Sequel right)
<span id="line2080"></span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part6_pacbio_02.png">/genomics/images/part6_pacbio_02.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2081"></span>
<span id="line2082"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>A typical genomic library preparation workflow is similar to that of Illumina sequencing with the difference that much higher molecular weight DNA is used and dumb-bell shaped adaptors are attached:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2083"></span>
<span id="line2084"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part6_pacbio_03.jpg">/genomics/images/part6_pacbio_03.jpg</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2085"></span>
<span id="line2086"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>An important note regarding PacBio data quality</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2087"></span>
<span id="line2088"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>PacBio libraries are circular. Because of this, a single polymerase can sequence the same piece of DNA several times. However, a balance exists between the DNA template length, polymerase life time (also known as polymerase read length) and read quality with PacBio data. The figure below illustrates this.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2089"></span>
<span id="line2090"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>A polymerase which is able to read for 10kb, could read a single 10kb template or, for example, it could read a 2kb template 5 times. The 2kb fragment, having been read 5 times, would be of much higher quality than the 10kb fragment.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2091"></span>
<span id="line2092"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part6_pacbio_04.jpg">/genomics/images/part6_pacbio_04.jpg</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2093"></span>
<span id="line2094"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Terminology which is important to understand with PacBio data is highlighted in above. A polymerase read refers to the complete set of basecalls associated with the polymerase sequencing the forward strand, adaptor, and reverse strand. Internally, we remove the adaptor sequences and provide the subreads (just the forward and reverse sequences of the template). These can be further analysed to produce circular consensus reads (CCS) if the enzyme has made more than 1 pass of the molecule.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2095"></span>
<span id="line2096"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Most assembly programs require the subreads if utilizing PacBio data.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2097"></span>
<span id="line2098"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>MinION sequencing technology:</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2099"></span>
<span id="line2100"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>This is a radically new sequencing technology based upon monitoring electrical current fluctuations associated with the translocation of single DNA molecules through nanopores embedded in a membrane. You can read more about the technology in detail </span><span>&lt;<span class="start-tag">a</span> <span class="attribute-name">href</span>="<a class="attribute-value" href="view-source:https://nanoporetech.com/how-it-works">https://nanoporetech.com/how-it-works</a>"&gt;</span><span>https://nanoporetech.com/how-it-works</span><span>&lt;/<span class="end-tag">a</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2101"></span>
<span id="line2102"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>This is a remarkable technology which has the potential to supplant short-read sequencing, if the error rates can be brought down sufficiently. It has the advantage of being portable and of generating long fragment lengths.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2103"></span>
<span id="line2104"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>A typical long read assembly pipeline:</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2105"></span></span><span>&lt;<span class="start-tag">ul</span>&gt;</span><span>
<span id="line2106"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>Generate and evaluate QC metrics</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line2107"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>Perform an initial assembly with Minimap/Miniasm to check whether we have sufficient data</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line2108"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>Check for any potential contamination in the Miniasm assemblies using Blast and Krona</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line2109"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>Correct long reads via consensus alignment and assemble</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line2110"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>Polish the assembly with Illumina or other short-read data</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line2111"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>If necessary, attempt to circularize the genome correctly</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line2112"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>Compare assemblies against the reference using Quast</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line2113"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>Generate gene predictions and/or functional annotation</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line2114"></span></span><span>&lt;/<span class="end-tag">ul</span>&gt;</span><span>
<span id="line2115"></span>
<span id="line2116"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>We have provided you with data generated from the same strain of </span><span>&lt;<span class="start-tag">em</span>&gt;</span><span>E.coli</span><span>&lt;/<span class="end-tag">em</span>&gt;</span><span> K12 MG1655 across three different platforms - Oxford Nanopore MinION, PacBio RSII and PacBio Sequel. The MinION data was generated and published by Nick Loman and Josh Quick whilst the PacBio data was generated at Exeter from the same material.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2117"></span>
<span id="line2118"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>We have provided you with data from both 1D and 2D libraries. Remember these are two different types of library preparation. A 1D library enables just the template strand of the DNA molecule to be read, whilst a 2D library enables both the template and complement strand to be sequenced. This means that a 2D library will be sequenced. (Recently </span><span>&lt;<span class="start-tag">em</span>&gt;</span><span>1D squared</span><span>&lt;/<span class="end-tag">em</span>&gt;</span><span> has replaced the </span><span>&lt;<span class="start-tag">em</span>&gt;</span><span>2D</span><span>&lt;/<span class="end-tag">em</span>&gt;</span><span> libraries)</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2119"></span>
<span id="line2120"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>The datasets can be found in the directory:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2121"></span>
<span id="line2122"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>~/genomics_tutorial/long_read_tutorial/raw_data</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2123"></span>
<span id="line2124"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Note</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span> that we have four datasets in total - Sequel, RSII, MinION 1D and MinION 2D the MinIon dataset consists of two fasta files. One is a 1D dataset and another is a 2D dataset. Note that typically MinION datasets are generated in a binary formatted file on a per-read basis which contain the current-levels and other data. However, for simplicity we have converted these into fasta files.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2125"></span>
<span id="line2126"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>We’ll  follow a similar analysis protocol for all 4 datasets (some are pre-generated).</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2127"></span>
<span id="line2128"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Task 1 Generate basic statistics for each of these datasets</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2129"></span>
<span id="line2130"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>You can use the fasta_summary.pl script to do this for both fasta-formatted MinION datasets. It might take a while to compute, therefore we have pre-computed the results for you:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2131"></span>
<span id="line2132"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>fasta_summary.pl -i E_coli_K12_1D_R9.2.pass.fasta -o E_coli_K12_1D_summary -t read</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2133"></span>
<span id="line2134"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>fasta_summary.pl -i E_coli_K12_2D_R9.0.pass.fasta -o E_coli_K12_2D_summary -t read</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2135"></span>
<span id="line2136"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>~/workshop_materials/genomics_tutorial/long_read_tutorial/pre_computed_results/canu_assemblies/fasta_summaries</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2137"></span>
<span id="line2138"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>For the PacBio data you can use the fastqc program as per the Illumina datasets you have looked at previously.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2139"></span>
<span id="line2140"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>The fasta_summary.pl script will generate several output files:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2141"></span>
<span id="line2142"></span></span><span>&lt;<span class="start-tag">ul</span>&gt;</span><span>
<span id="line2143"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>stats.txt</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span> file contains text-data on the number of bases, longest read and other useful statistics.</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line2144"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>histograms_bins.dat.png</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span>  file is a picture file with a histogram of read lengths (you can use firefox to see the file).</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line2145"></span></span><span>&lt;/<span class="end-tag">ul</span>&gt;</span><span>
<span id="line2146"></span>
<span id="line2147"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Task 2: Compare the datasets</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2148"></span>
<span id="line2149"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Note that unlike Illumina sequencers, the read lengths produced by these platforms are highly variable. The MinION datasets for example vary between 100bp and 140,000bp.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2150"></span>
<span id="line2151"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part6_task2_01.png">/genomics/images/part6_task2_01.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2152"></span>
<span id="line2153"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Note the relatively poor quality scores in the fastqc report for the PacBio sequences. These reflect the 10-12% raw single-pass per-base error rate of the polymerase. Similar error rates are observed in the MinION datasets. Note that </span><span>&lt;<span class="start-tag">em</span>&gt;</span><span>unlike</span><span>&lt;/<span class="end-tag">em</span>&gt;</span><span> Illumina datasets, this error rate does not appreciably increase over the length of the read (although at very short or very long read lengths there are far fewer reads present which increase the variance estimates).</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2154"></span>
<span id="line2155"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part6_task2_02.png">/genomics/images/part6_task2_02.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2156"></span>
<span id="line2157"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Task 3: Evaluate the likelihood of obtaining a successful assembly using Minimap/Miniasm</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2158"></span>
<span id="line2159"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>This very useful assembler tries to assemble long-reads </span><span>&lt;<span class="start-tag">em</span>&gt;</span><span>without</span><span>&lt;/<span class="end-tag">em</span>&gt;</span><span> trying to correct the reads. Although you would not want to use such an assembly without further correction, it provides a useful yardstick to determine whether a more computationally intensive assembly is likely to yield good results. You can read more here </span><span>&lt;<span class="start-tag">a</span> <span class="attribute-name">href</span>="<a class="attribute-value" href="view-source:https://github.com/lh3/miniasm">https://github.com/lh3/miniasm</a>"&gt;</span><span>https://github.com/lh3/miniasm.</span><span>&lt;/<span class="end-tag">a</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2160"></span>
<span id="line2161"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Because of limitations on the memory available to us, we’ll need to subset the data first, selecting a random 100,000 reads to use (note that this is not the best approach since we should really pick the longest 100,000 reads).</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2162"></span>
<span id="line2163"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>The syntax can be a bit confusing, so here is an example using the RSII dataset to get you started. Each assembly should take about 5-10 mins.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2164"></span>
<span id="line2165"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Create a new working directory in ~/workshop_materials/genomics_tutorial/long_read_tutorial called miniasm_assemblies</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2166"></span>
<span id="line2167"></span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">pre</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">code</span>&gt;</span><span>cd ~/workshop_materials/long_read_tutorial
<span id="line2168"></span>mkdir miniasm_assemblies
<span id="line2169"></span></span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">pre</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span>
<span id="line2170"></span>
<span id="line2171"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part6_task3_01.png">/genomics/images/part6_task3_01.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2172"></span>
<span id="line2173"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>We’ll use the </span><span>&lt;<span class="start-tag">em</span>&gt;</span><span>seqtk</span><span>&lt;/<span class="end-tag">em</span>&gt;</span><span> package from the same author as samtools to randomly select 100,000 reads from the raw dataset:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2174"></span>
<span id="line2175"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>seqtk sample ~/workshop_materials/genomics_tutorial/long_read_tutorial/raw_data/RSII_Ecoli_K12_subreads.fastq 100000 <span class="entity"><span>&amp;</span>gt;</span> RSII_Ecoli_K12_subreads.subsampled.fastq</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2176"></span>
<span id="line2177"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Now we can ask minimap to calculate all the overlaps between those 100,000 reads and output the results in a compressed gzip file. Note that you will have to put the input reads (RSII_Ecoli_K12_subreads.subsampled.fastq) in  twice - since we are asking minimap to calculate the overlaps between </span><span>&lt;<span class="start-tag">em</span>&gt;</span><span>all</span><span>&lt;/<span class="end-tag">em</span>&gt;</span><span> reads in the dataset:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2178"></span>
<span id="line2179"></span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">pre</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">code</span>&gt;</span><span>minimap -Sw5 -L100 -m0 -t 2 RSII_Ecoli_K12_subreads.subsampled.fastq RSII_Ecoli_K12_subreads.subsampled.fastq | gzip -1 <span class="entity"><span>&amp;</span>gt;</span> overlaps_RSII.paf.gz
<span id="line2180"></span></span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">pre</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span>
<span id="line2181"></span>
<span id="line2182"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Once complete, we can ask miniasm to create an assembly graph and find an assembly path through it:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2183"></span>
<span id="line2184"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>miniasm -f RSII_Ecoli_K12_subreads.fastq overlaps_RSII.paf.gz <span class="entity"><span>&amp;</span>gt;</span> overlaps_RSII.gfa</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2185"></span>
<span id="line2186"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>awk '/^S/{print "\<span class="entity"><span>&amp;</span>gt;</span>"\$2"\\n"\$3}' overlaps_RSII.gfa | fold <span class="entity"><span>&amp;</span>gt;</span> miniasm.PacBio_RSII.contigs.fasta</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2187"></span>
<span id="line2188"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Repeat this for the Sequel and MinION datasets and then use the QUAST package (quast.py) to compare the assemblies against the reference genome. You should use the same E.coli reference as you used for the Illumina assemblies.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2189"></span>
<span id="line2190"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>You should be able to open the report.html file in firefox or other web-browser to compare the assemblies. It should look somewhat similar to:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2191"></span>
<span id="line2192"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part6_task3_02.png">/genomics/images/part6_task3_02.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2193"></span>
<span id="line2194"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Note that your results will differ since you will have a different subset of reads. Minimap/miniasm does not correct reads prior to assembly. This means that the alignments to the reference are likely to be error prone. Note that even without this correction, the assembler is able to reconstruct the genome in (more or less) a single contig of 4.7Mb with just 100,000 reads! This bodes well for a more computationally intensive assembly which first corrects reads and highlights the power of long reads for de-novo assembly.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2195"></span>
<span id="line2196"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>If you wish you can subset the data with the </span><span>&lt;<span class="start-tag">em</span>&gt;</span><span>seqtk sample</span><span>&lt;/<span class="end-tag">em</span>&gt;</span><span> and compare how the datasets assemble at different levels of coverage.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2197"></span>
<span id="line2198"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Task 4: Generate a corrected assembly with Canu (results pre-computed)</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2199"></span>
<span id="line2200"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Canu is derived from the original Celera assembler used to assemble the human genome from Sanger data. It has been optimized for long-read PacBio and Nanopore data. You can read more at </span><span>&lt;<span class="start-tag">a</span> <span class="attribute-name">href</span>="<a class="attribute-value" href="view-source:http://canu.readthedocs.io/en/stable/tutorial.html">http://canu.readthedocs.io/en/stable/tutorial.html</a>"&gt;</span><span>http://canu.readthedocs.io/en/stable/tutorial.html.</span><span>&lt;/<span class="end-tag">a</span>&gt;</span><span> A number of long read assemblers are available (including Nanopore-specific assemblers - a good review paper on this is:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2201"></span>
<span id="line2202"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">em</span>&gt;</span><span>Chu J, Mohamadi H, Warren RL, Yang C, Bi-Rol I. Innovations and challenges in detecting long read overlaps: an evaluation of the state-of- the-art. Bioinformatics. 2016</span><span>&lt;/<span class="end-tag">em</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2203"></span>
<span id="line2204"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>For now we will just use Canu as it is relatively straightforward to use. Here is an example command to assemble the RSII E.coli data. We have pre-computed the results for you for the MinION 2D, RSII and Sequel data.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2205"></span>
<span id="line2206"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>~/workshop_materials/genomics_tutorial/long_read_tutorial/pre_computed_results/canu_assemblies</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2207"></span>
<span id="line2208"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>To save time, and provide a fairer comparison between platforms we will not use the MinION 1D data.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2209"></span>
<span id="line2210"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>canu -p canu_RSII -d canu_RSII genomeSize=4.7m  useGrid=False  -pacbio-raw RSII_Ecoli_K12_subreads.fastq</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2211"></span>
<span id="line2212"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Note that Canu will:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2213"></span></span><span>&lt;<span class="start-tag">ul</span>&gt;</span><span>
<span id="line2214"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>Correct reads by performing all-against-all alignments and taking a consensus</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line2215"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>Overlap and trim reads</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line2216"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>Generate an assembly</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line2217"></span></span><span>&lt;/<span class="end-tag">ul</span>&gt;</span><span>
<span id="line2218"></span>
<span id="line2219"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Task 5: Polish Canu assemblies with Illumina data using Pilon (results
<span id="line2220"></span>pre-computed)</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2221"></span>
<span id="line2222"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Oxford Nanopore and MinION data suffer a tendency to introduce insertions or deletions into a sequence (even after read correction). This means that to obtain the highest per-base quality it is desirable to polish assemblies by aligning short reads using BWA and then using a tool such as Pilon </span><span>&lt;<span class="start-tag">a</span> <span class="attribute-name">href</span>="<a class="attribute-value" href="view-source:https://github.com/broadinstitute/pilon/wiki">https://github.com/broadinstitute/pilon/wiki</a>"&gt;</span><span>(https://github.com/broadinstitute/pilon/wiki)</span><span>&lt;/<span class="end-tag">a</span>&gt;</span><span> to polish the assemblies.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2223"></span>
<span id="line2224"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>We have supplied you with pre-computed results. We’ll  use just the RSII data to illustrate an example set of commands. First let’s align the Illumina reads from the short-read section of the workshop using BWA.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2225"></span>
<span id="line2226"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>We need to create a reference for the PacBio RSII contigs:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2227"></span>
<span id="line2228"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>bwa index RSII_canu_contigs.fasta</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2229"></span>
<span id="line2230"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Now we need to align the Illumina reads against the contigs:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2231"></span>
<span id="line2232"></span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">pre</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">code</span>&gt;</span><span>bwa mem -x pacbio -t 2 RSII_canu_contigs.fasta ../../../data/sequencing/ecoli_exeter/E_Coli_CGATGT_L001_R1_001.fastq ../.
<span id="line2233"></span>./../data/sequencing/ecoli_exeter/E_Coli_CGATGT_L001_R2_001.fastq   |   samtools sort -@ 2 -O
<span id="line2234"></span>bam -o RSII_canu_contigs_illumina_aligned.bam
<span id="line2235"></span></span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">pre</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span>
<span id="line2236"></span>
<span id="line2237"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>samtools index RSII_canu_contigs_illumina_aligned.bam</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2238"></span>
<span id="line2239"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Now that we have aligned the Illumina data against the contigs we can run Pilon to correct the contigs where they differ from the Illumina reads.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2240"></span>
<span id="line2241"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>pilon --genome RSII_canu_contigs.fasta --frags RSII_canu_contigs_illumina_aligned.bam --changes
<span id="line2242"></span>--outdir RSII_canu_pilon_polished</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2243"></span>
<span id="line2244"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>You can view the corrected contigs in RSII_canu_pilon_polished/pilon.fasta and the see a list of the changes which have been made in RSII_canu_pilon_polished/pilon.changes. Note that most changes correct indels and the much higher number of corrections made for the MinION assembly vs the PacBio assemblies  (46558 for MinION 2D vs 387 for RSII and 2247 for Sequel).</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2245"></span>
<span id="line2246"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>We need to interpret these polished results with care. Remember we are using Illumina sequencing which contains all sorts of biases of its own thanks to amplification biases introduced by PCR and other artefacts. As such whilst we might be correcting some errors, we could be introducing Illumina biases into these assemblies. We’re also using Illumina reads which are not quite identical to the reference or the PacBio material which is also not ideal (but often the case
<span id="line2247"></span>in the real world!). We’ll  see the effect of this in the final task.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2248"></span>
<span id="line2249"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Task 6: Use Blast and Krona to confirm species present in the assembly (results pre-
<span id="line2250"></span>computed)</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2251"></span>
<span id="line2252"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>We can use BLAST to identify taxonomic hits to ensure that we have the correct species present and filter out any contigs resulting from control spike-in DNA or other contaminants. You can also use other tools to do this such as Kraken or Centrifuge.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2253"></span>
<span id="line2254"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>As the blast searches take some time, the results have been pre-computed for you in</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2255"></span>
<span id="line2256"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>~/workshop_materials/genomics_tutorial/long_read_tutorial/pre_computed_results/krona_plot/</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2257"></span>
<span id="line2258"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>but we have included the commands used below:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2259"></span>
<span id="line2260"></span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">pre</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">code</span>&gt;</span><span>blastn -db ../../../db/blast/nt -query RSII_canu_pilon_polished.fasta  -outfmt 7 -evalue 1e-06 -out RSII_canu_pilon_polished.fasta.blastn.outfmt7 -num_threads 2
<span id="line2261"></span></span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">pre</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span>
<span id="line2262"></span>
<span id="line2263"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Once this has completed, we can import the results into Krona which is a neat little visualization tool for BLAST results.
<span id="line2264"></span>**
<span id="line2265"></span></span><span>&lt;<span class="start-tag">code</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span>ktImportBLAST -i RSII_canu_pilon_polished.fasta.blastn.outfmt7 -o krona_blast_results.html</span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2266"></span>
<span id="line2267"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>These results can be visualized in a web-browser such as firefox.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2268"></span>
<span id="line2269"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Note that the MinION data contains Lambda virus DNA which is used as a spike-in to some MinION runs. You can identify which contigs hit species of interest by clicking on the area of the pie chart you are interested in and then clicking on the ‘Count’ in the top right corner. This wil  give you the contig names which match to the virus as opposed to the bacteria.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2270"></span>
<span id="line2271"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">img</span> <span class="attribute-name">src</span>="<a class="attribute-value" href="view-source:https://biomedicalhub.github.io/genomics/images/part6_task6_01.png">/genomics/images/part6_task6_01.png</a>" <span class="attribute-name">alt</span>="<a class="attribute-value"></a>" <span>/</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2272"></span>
<span id="line2273"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Task 7: Circularise assemblies using Circlator (results pre-computed)</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2274"></span>
<span id="line2275"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Note that bacterial genomes are circular. As such we may end up mis-assembling the genome because it is circular. The circlator package </span><span>&lt;<span class="start-tag">a</span> <span class="attribute-name">href</span>="<a class="attribute-value" href="view-source:https://github.com/sanger-pathogens/circlator/wiki/Minimus2-circularization-pipeline">https://github.com/sanger-pathogens/circlator/wiki/Minimus2-circularization-pipeline</a>"&gt;</span><span>(https://github.com/sanger/pathogens/circlator/wiki/Minimus2-circularization-pipeline)</span><span>&lt;/<span class="end-tag">a</span>&gt;</span><span> attempts to correct this.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2276"></span>
<span id="line2277"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>In the circularized_results directory you will find the results of the circularization pipeline.</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2278"></span>
<span id="line2279"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span></span><span>&lt;<span class="start-tag">strong</span>&gt;</span><span>Task 8: Compare polished assemblies using Quast</span><span>&lt;/<span class="end-tag">strong</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2280"></span>
<span id="line2281"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Use quast.py to generate a report for the original Canu results (i.e. pre Illumina correction and
<span id="line2282"></span>circularization) and these corrected and circularized results. E.g (all on one line):</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2283"></span>
<span id="line2284"></span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlighter-rouge</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">div</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">pre</span> <span class="attribute-name">class</span>="<a class="attribute-value">highlight</a>"&gt;</span><span></span><span>&lt;<span class="start-tag">code</span>&gt;</span><span>quast.py -R ~/workshop_materials/genomics_tutorial/data/reference/U00096/U00096.fna MinION_2D_canu_pilon_polished.circularise.fasta
<span id="line2285"></span>~/workshop_materials/genomics_tutorial/long_read_tutorial/pre_computed_results/canu_assemblies/MinION_2D.contigs.fasta RSII_canu_pilon_polished_circularise.fasta ~/workshop_materials/genomics_tutorial/long_read_tutorial/pre_computed_results/canu_assemblies/
<span id="line2286"></span>RSII_canu_contigs.fasta Sequel_canu_pilon_polished.circularise.fasta
<span id="line2287"></span>~/workshop_materials/genomics_tutorial/long_read_tutorial/pre_computed_results/canu_assemblies/Sequel_canu_contigs.fasta
<span id="line2288"></span></span><span>&lt;/<span class="end-tag">code</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">pre</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span>
<span id="line2289"></span>
<span id="line2290"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Look at the effect of the polishing step for each technology. Have a think about the following:</span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2291"></span></span><span>&lt;<span class="start-tag">ul</span>&gt;</span><span>
<span id="line2292"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>The PacBio reads have a higher mismatch error after polishing but a lower indel rate? (think about the possible effects of PCR amplification on GC-bias)</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line2293"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>What about the total number of contigs for each platform?</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line2294"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>How does the mis-match rate to the reference genome compare between the MinION and the PacBio platforms?</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line2295"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>The Sequel platform used very early chemistry - how does it perform compared to the other platforms?</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line2296"></span>  </span><span>&lt;<span class="start-tag">li</span>&gt;</span><span>Which platform would you select if you wanted to use the assembly as a reference to perform variant calling against?</span><span>&lt;/<span class="end-tag">li</span>&gt;</span><span>
<span id="line2297"></span></span><span>&lt;/<span class="end-tag">ul</span>&gt;</span><span>
<span id="line2298"></span>
<span id="line2299"></span></span><span>&lt;<span class="start-tag">h2</span> <span class="attribute-name">id</span>="<a class="attribute-value">concluding-remarks</a>"&gt;</span><span>Concluding remarks:</span><span>&lt;/<span class="end-tag">h2</span>&gt;</span><span>
<span id="line2300"></span>
<span id="line2301"></span></span><span>&lt;<span class="start-tag">p</span>&gt;</span><span>Well done! If you have reached this far, you deserve a round of applause. You have completed some of the most common tasks in genomics. You can use the same machine and the same scripts to perform analysis of any dataset! If you need to transfer data to/from the instance a tutorial can be found at </span><span>&lt;<span class="start-tag">a</span> <span class="attribute-name">href</span>="<a class="attribute-value" href="view-source:http://www.siteground.com/tutorials/ssh/ssh_winscp.htm">http://www.siteground.com/tutorials/ssh/ssh_winscp.htm</a>"&gt;</span><span>http://www.siteground.com/tutorials/ssh/ssh_winscp.htm</span><span>&lt;/<span class="end-tag">a</span>&gt;</span><span></span><span>&lt;/<span class="end-tag">p</span>&gt;</span><span>
<span id="line2302"></span>
<span id="line2303"></span>  
<span id="line2304"></span></span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span>
<span id="line2305"></span>
<span id="line2306"></span>    </span><span>&lt;/<span class="end-tag">div</span>&gt;</span><span>
<span id="line2307"></span>
<span id="line2308"></span>  </span><span>&lt;/<span class="end-tag">body</span>&gt;</span><span>
<span id="line2309"></span></span><span>&lt;/<span class="end-tag">html</span>&gt;</span><span>
<span id="line2310"></span></span></pre></body></html>